<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Guide to TwoSampleMR • TwoSampleMR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Guide to TwoSampleMR">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">TwoSampleMR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/guide.html">Guide to TwoSampleMR</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Guide to TwoSampleMR</h1>
            
      
      
      <div class="hidden name"><code>guide.Rmd</code></div>

    </div>

    
    
<hr>
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>Two sample Mendelian randomisation (2SMR) is a method to estimate the causal effect of an exposure on an outcome using only summary statistics from genome wide association studies (GWAS). Though conceptually straightforward, there are a number of steps that are required to perform the analysis properly, and they can be cumbersome. The TwoSampleMR package aims to make this easy by combining three important components</p>
<ul>
<li>data management and harmonisation</li>
<li>the statistical routines to estimate the causal effects</li>
<li>connection to a large repository of the actual GWAS summary statistics needed to perform the analyses.</li>
</ul>
<p>The general principles <span class="citation">(G. Davey Smith and Ebrahim 2003; George Davey Smith and Hemani 2014)</span>, and statistical methods <span class="citation">(Pierce and Burgess 2013; Bowden, Davey Smith, and Burgess 2015)</span> can be found elsewhere, here we will just outline how to use the R package.</p>
<p>This package uses the <a href="https://github.com/mrcieu/ieugwasr">ieugwasr</a> package to connect to the database of thousands of complete GWAS summary data.</p>
<hr>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>To install directly from the GitHub repository do the following:</p>
<pre class="sourceCode r" id="cb1"><code class="sourceCode r"><div class="sourceLine" id="cb1-1" data-line-number="1">
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(devtools)</div>
<div class="sourceLine" id="cb1-2" data-line-number="2">
<span class="kw"><a href="https://rdrr.io/pkg/devtools/man/reexports.html">install_github</a></span>(<span class="st">"MRCIEU/TwoSampleMR"</span>)</div></code></pre>
<p>If you don’t have the <code>devtools</code> package install it from CRAN using <code><a href="https://rdrr.io/r/utils/install.packages.html">install.packages("devtools")</a></code>.</p>
<hr>
</div>
<div id="overview" class="section level2">
<h2 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h2>
<p>The workflow for performing MR is as follows:</p>
<ol style="list-style-type: decimal">
<li>Select instruments for the exposure (perform LD clumping if necessary)</li>
<li>Extract the instruments from the MR Base GWAS database for the outcomes of interest</li>
<li>Harmonise the effect sizes for the instruments on the exposures and the outcomes to be each for the same reference allele</li>
<li>Perform MR analysis, sensitivity analyses, create plots, compile reports</li>
</ol>
<p>A diagramatic overview is shown here:</p>
<p><img src="figure/twosamplemr_schematic_long-01.png" width="500"></p>
<p>A basic analysis, e.g. the causal effect of body mass index on coronary heart disease, looks like this:</p>
<pre class="sourceCode r" id="cb2"><code class="sourceCode r"><div class="sourceLine" id="cb2-1" data-line-number="1"></div>
<div class="sourceLine" id="cb2-2" data-line-number="2">
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(TwoSampleMR)</div>
<div class="sourceLine" id="cb2-3" data-line-number="3"></div>
<div class="sourceLine" id="cb2-4" data-line-number="4"><span class="co"># List available GWASs</span></div>
<div class="sourceLine" id="cb2-5" data-line-number="5">ao &lt;-<span class="st"> </span><span class="kw"><a href="../reference/available_outcomes.html">available_outcomes</a></span>()</div>
<div class="sourceLine" id="cb2-6" data-line-number="6"></div>
<div class="sourceLine" id="cb2-7" data-line-number="7"><span class="co"># Get instruments</span></div>
<div class="sourceLine" id="cb2-8" data-line-number="8">exposure_dat &lt;-<span class="st"> </span><span class="kw"><a href="../reference/read_exposure_data.html">read_exposure_data</a></span>(<span class="st">"path_to_bmi_instruments.txt"</span>)</div>
<div class="sourceLine" id="cb2-9" data-line-number="9"></div>
<div class="sourceLine" id="cb2-10" data-line-number="10"><span class="co"># Get effects of instruments on outcome</span></div>
<div class="sourceLine" id="cb2-11" data-line-number="11">outcome_dat &lt;-<span class="st"> </span><span class="kw"><a href="../reference/extract_outcome_data.html">extract_outcome_data</a></span>(<span class="dt">snps=</span>exposure_dat<span class="op">$</span>SNP, <span class="dt">outcomes=</span><span class="st">"7"</span>)</div>
<div class="sourceLine" id="cb2-12" data-line-number="12"></div>
<div class="sourceLine" id="cb2-13" data-line-number="13"><span class="co"># Harmonise the exposure and outcome data</span></div>
<div class="sourceLine" id="cb2-14" data-line-number="14">dat &lt;-<span class="st"> </span><span class="kw"><a href="../reference/harmonise_data.html">harmonise_data</a></span>(exposure_dat, outcome_dat)</div>
<div class="sourceLine" id="cb2-15" data-line-number="15"></div>
<div class="sourceLine" id="cb2-16" data-line-number="16"><span class="co"># Perform MR</span></div>
<div class="sourceLine" id="cb2-17" data-line-number="17">res &lt;-<span class="st"> </span><span class="kw"><a href="../reference/mr.html">mr</a></span>(dat)</div></code></pre>
<p>Each step is documented in detail below.</p>
</div>
<div id="authentication" class="section level2">
<h2 class="hasAnchor">
<a href="#authentication" class="anchor"></a>Authentication</h2>
<p>**The authentication method has changed due to the GoogleAuthR method changing. The main differences are that:</p>
<ol style="list-style-type: decimal">
<li>By default you will not be asked to authenticate and will only have access to public data</li>
<li>If you do need to authenticate in order to access private datasets there is no longer a single file called <code>mrbase.oauth</code>, rather, there is a directory called <code>ieugwasr_oauth</code>.</li>
</ol>
<p>Detailed information is given here: <a href="https://github.com/MRCIEU/ieugwasr/blob/master/README.md#authentication" class="uri">https://github.com/MRCIEU/ieugwasr/blob/master/README.md#authentication</a></p>
<hr>
</div>
</div>
<div id="exposure-data" class="section level1">
<h1 class="hasAnchor">
<a href="#exposure-data" class="anchor"></a>Exposure data</h1>
<p>A data frame of the instruments for an exposure is required. Each line has the information for one SNP for one exposure. The minimum information required for MR analysis is the following:</p>
<ul>
<li>
<code>SNP</code> - rs ID</li>
<li>
<code>beta</code> - The effect size. If the trait is binary then log(OR) should be used</li>
<li>
<code>se</code> - The standard error of the effect size</li>
<li>
<code>effect_allele</code> - The allele of the SNP which has the effect marked in <code>beta</code>
</li>
</ul>
<p>Other information that is useful for MR can also be provided:</p>
<ul>
<li>
<code>other_allele</code> - The non-effect allele</li>
<li>
<code>eaf</code> - The effect allele frequency</li>
<li>
<code>Phenotype</code> - The name of the phenotype for which the SNP has an effect</li>
</ul>
<p>You can also provide the following extra information:</p>
<ul>
<li>
<code>samplesize</code> - Sample size for estimating the effect size</li>
<li>
<code>ncase</code> - Number of cases</li>
<li>
<code>ncontrol</code> - Number of controls</li>
<li>
<code>pval</code> - The P-value for the SNP’s association with the exposure</li>
<li>
<code>units</code> - The units in which the effects are presented</li>
<li>
<code>gene</code> - The gene or other annotation for the the SNP</li>
</ul>
<div id="reading-in-from-a-file" class="section level2">
<h2 class="hasAnchor">
<a href="#reading-in-from-a-file" class="anchor"></a>Reading in from a file</h2>
<p>The data can be read in from a text file using the <code>read_exposure_data</code> function. The file must have a header with column names corresponding to the columns described above.</p>
<div id="example-1-the-default-column-names-are-used" class="section level3">
<h3 class="hasAnchor">
<a href="#example-1-the-default-column-names-are-used" class="anchor"></a>Example 1: The default column names are used</h3>
<p>An example of a text file with the default column names is provided as part of the package, the first few rows look like this:</p>
<pre><code>Phenotype SNP beta se effect_allele other_allele eaf pval units gene samplesize
BMI rs10767664 0.19 0.0306122448979592 A T 0.78 5e-26 kg/m2 BDNF 225238
BMI rs13078807 0.1 0.0204081632653061 G A 0.2 4e-11 kg/m2 CADM2 221431
BMI rs1514175 0.07 0.0204081632653061 A G 0.43 8e-14 kg/m2 TNNI3K 207641
BMI rs1558902 0.39 0.0204081632653061 A T 0.42 5e-120 kg/m2 FTO 222476
BMI rs10968576 0.11 0.0204081632653061 G A 0.31 3e-13 kg/m2 LRRN6C 247166
BMI rs2241423 0.13 0.0204081632653061 G A 0.78 1e-18 kg/m2 LBXCOR1 227886</code></pre>
<p>The exact path to the file will be different on everyone’s computer, but it can be located like this:</p>
<pre class="sourceCode r" id="cb4"><code class="sourceCode r"><div class="sourceLine" id="cb4-1" data-line-number="1">bmi_file &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata/bmi.txt"</span>, <span class="dt">package=</span><span class="st">"TwoSampleMR"</span>)</div></code></pre>
<p>You can read the data in like this:</p>
<pre class="sourceCode r" id="cb5"><code class="sourceCode r"><div class="sourceLine" id="cb5-1" data-line-number="1">bmi_exp_dat &lt;-<span class="st"> </span><span class="kw"><a href="../reference/read_exposure_data.html">read_exposure_data</a></span>(bmi_file)</div>
<div class="sourceLine" id="cb5-2" data-line-number="2">
<span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(bmi_exp_dat)</div>
<div class="sourceLine" id="cb5-3" data-line-number="3"><span class="co">#&gt;          SNP beta.exposure se.exposure effect_allele.exposure</span></div>
<div class="sourceLine" id="cb5-4" data-line-number="4"><span class="co">#&gt; 1 rs10767664          0.19  0.03061224                      A</span></div>
<div class="sourceLine" id="cb5-5" data-line-number="5"><span class="co">#&gt; 2 rs13078807          0.10  0.02040816                      G</span></div>
<div class="sourceLine" id="cb5-6" data-line-number="6"><span class="co">#&gt; 3  rs1514175          0.07  0.02040816                      A</span></div>
<div class="sourceLine" id="cb5-7" data-line-number="7"><span class="co">#&gt; 4  rs1558902          0.39  0.02040816                      A</span></div>
<div class="sourceLine" id="cb5-8" data-line-number="8"><span class="co">#&gt; 5 rs10968576          0.11  0.02040816                      G</span></div>
<div class="sourceLine" id="cb5-9" data-line-number="9"><span class="co">#&gt; 6  rs2241423          0.13  0.02040816                      G</span></div>
<div class="sourceLine" id="cb5-10" data-line-number="10"><span class="co">#&gt;   other_allele.exposure eaf.exposure pval.exposure units.exposure</span></div>
<div class="sourceLine" id="cb5-11" data-line-number="11"><span class="co">#&gt; 1                     T         0.78         5e-26          kg/m2</span></div>
<div class="sourceLine" id="cb5-12" data-line-number="12"><span class="co">#&gt; 2                     A         0.20         4e-11          kg/m2</span></div>
<div class="sourceLine" id="cb5-13" data-line-number="13"><span class="co">#&gt; 3                     G         0.43         8e-14          kg/m2</span></div>
<div class="sourceLine" id="cb5-14" data-line-number="14"><span class="co">#&gt; 4                     T         0.42        5e-120          kg/m2</span></div>
<div class="sourceLine" id="cb5-15" data-line-number="15"><span class="co">#&gt; 5                     A         0.31         3e-13          kg/m2</span></div>
<div class="sourceLine" id="cb5-16" data-line-number="16"><span class="co">#&gt; 6                     A         0.78         1e-18          kg/m2</span></div>
<div class="sourceLine" id="cb5-17" data-line-number="17"><span class="co">#&gt;   gene.exposure samplesize.exposure exposure mr_keep.exposure</span></div>
<div class="sourceLine" id="cb5-18" data-line-number="18"><span class="co">#&gt; 1          BDNF              225238      BMI             TRUE</span></div>
<div class="sourceLine" id="cb5-19" data-line-number="19"><span class="co">#&gt; 2         CADM2              221431      BMI             TRUE</span></div>
<div class="sourceLine" id="cb5-20" data-line-number="20"><span class="co">#&gt; 3        TNNI3K              207641      BMI             TRUE</span></div>
<div class="sourceLine" id="cb5-21" data-line-number="21"><span class="co">#&gt; 4           FTO              222476      BMI             TRUE</span></div>
<div class="sourceLine" id="cb5-22" data-line-number="22"><span class="co">#&gt; 5        LRRN6C              247166      BMI             TRUE</span></div>
<div class="sourceLine" id="cb5-23" data-line-number="23"><span class="co">#&gt; 6       LBXCOR1              227886      BMI             TRUE</span></div>
<div class="sourceLine" id="cb5-24" data-line-number="24"><span class="co">#&gt;   pval_origin.exposure units.exposure_dat id.exposure data_source.exposure</span></div>
<div class="sourceLine" id="cb5-25" data-line-number="25"><span class="co">#&gt; 1             reported              kg/m2      vs2usl             textfile</span></div>
<div class="sourceLine" id="cb5-26" data-line-number="26"><span class="co">#&gt; 2             reported              kg/m2      vs2usl             textfile</span></div>
<div class="sourceLine" id="cb5-27" data-line-number="27"><span class="co">#&gt; 3             reported              kg/m2      vs2usl             textfile</span></div>
<div class="sourceLine" id="cb5-28" data-line-number="28"><span class="co">#&gt; 4             reported              kg/m2      vs2usl             textfile</span></div>
<div class="sourceLine" id="cb5-29" data-line-number="29"><span class="co">#&gt; 5             reported              kg/m2      vs2usl             textfile</span></div>
<div class="sourceLine" id="cb5-30" data-line-number="30"><span class="co">#&gt; 6             reported              kg/m2      vs2usl             textfile</span></div></code></pre>
<p>The output from this function is a new data frame with standardised column names:</p>
<ul>
<li><code>SNP</code></li>
<li><code>exposure</code></li>
<li><code>beta.exposure</code></li>
<li><code>se.exposure</code></li>
<li><code>effect_allele.exposure</code></li>
<li><code>other_allele.exposure</code></li>
<li><code>eaf.exposure</code></li>
<li><code>mr_keep.exposure</code></li>
<li><code>pval.exposure</code></li>
<li><code>pval_origin.exposure</code></li>
<li><code>id.exposure</code></li>
<li><code>data_source.exposure</code></li>
<li><code>units.exposure</code></li>
<li><code>gene.exposure</code></li>
<li><code>samplesize.exposure</code></li>
</ul>
<p>The function attempts to match the columns to the ones it expects. It also checks that the data type is as expected.</p>
<p>If the required data for MR to be performed is not present (SNP name, effect size, standard error, effect allele) for a particular SNP, then the column <code>mr_keep.exposure</code> will be “FALSE”.</p>
</div>
<div id="example-2-the-text-file-has-non-default-column-names" class="section level3">
<h3 class="hasAnchor">
<a href="#example-2-the-text-file-has-non-default-column-names" class="anchor"></a>Example 2: The text file has non-default column names</h3>
<p>If the text file does not have default column names, this can still be read in as follows. Here are the first few rows of an example:</p>
<pre><code>rsid,effect,SE,a1,a2,a1_freq,p-value,Units,Gene,n
rs10767664,0.19,0.030612245,A,T,0.78,5.00E-26,kg/m2,BDNF,225238
rs13078807,0.1,0.020408163,G,A,0.2,4.00E-11,kg/m2,CADM2,221431
rs1514175,0.07,0.020408163,A,G,0.43,8.00E-14,kg/m2,TNNI3K,207641
rs1558902,0.39,0.020408163,A,T,0.42,5.00E-120,kg/m2,FTO,222476</code></pre>
<p>Note that this is a CSV file, with commas separating fields. The file is located here:</p>
<pre class="sourceCode r" id="cb7"><code class="sourceCode r"><div class="sourceLine" id="cb7-1" data-line-number="1"></div>
<div class="sourceLine" id="cb7-2" data-line-number="2">bmi2_file &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata/bmi.csv"</span>, <span class="dt">package=</span><span class="st">"TwoSampleMR"</span>)</div></code></pre>
<p>To read in this data:</p>
<pre class="sourceCode r" id="cb8"><code class="sourceCode r"><div class="sourceLine" id="cb8-1" data-line-number="1">bmi_exp_dat &lt;-<span class="st"> </span><span class="kw"><a href="../reference/read_exposure_data.html">read_exposure_data</a></span>(</div>
<div class="sourceLine" id="cb8-2" data-line-number="2">    <span class="dt">filename =</span> bmi2_file,</div>
<div class="sourceLine" id="cb8-3" data-line-number="3">    <span class="dt">sep =</span> <span class="st">","</span>,</div>
<div class="sourceLine" id="cb8-4" data-line-number="4">    <span class="dt">snp_col =</span> <span class="st">"rsid"</span>,</div>
<div class="sourceLine" id="cb8-5" data-line-number="5">    <span class="dt">beta_col =</span> <span class="st">"effect"</span>,</div>
<div class="sourceLine" id="cb8-6" data-line-number="6">    <span class="dt">se_col =</span> <span class="st">"SE"</span>,</div>
<div class="sourceLine" id="cb8-7" data-line-number="7">    <span class="dt">effect_allele_col =</span> <span class="st">"a1"</span>,</div>
<div class="sourceLine" id="cb8-8" data-line-number="8">    <span class="dt">other_allele_col =</span> <span class="st">"a2"</span>,</div>
<div class="sourceLine" id="cb8-9" data-line-number="9">    <span class="dt">eaf_col =</span> <span class="st">"a1_freq"</span>,</div>
<div class="sourceLine" id="cb8-10" data-line-number="10">    <span class="dt">pval_col =</span> <span class="st">"p-value"</span>,</div>
<div class="sourceLine" id="cb8-11" data-line-number="11">    <span class="dt">units_col =</span> <span class="st">"Units"</span>,</div>
<div class="sourceLine" id="cb8-12" data-line-number="12">    <span class="dt">gene_col =</span> <span class="st">"Gene"</span>,</div>
<div class="sourceLine" id="cb8-13" data-line-number="13">    <span class="dt">samplesize_col =</span> <span class="st">"n"</span>
</div>
<div class="sourceLine" id="cb8-14" data-line-number="14">)</div>
<div class="sourceLine" id="cb8-15" data-line-number="15"><span class="co">#&gt; No phenotype name specified, defaulting to 'exposure'.</span></div>
<div class="sourceLine" id="cb8-16" data-line-number="16">
<span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(bmi_exp_dat)</div>
<div class="sourceLine" id="cb8-17" data-line-number="17"><span class="co">#&gt;          SNP beta.exposure se.exposure effect_allele.exposure</span></div>
<div class="sourceLine" id="cb8-18" data-line-number="18"><span class="co">#&gt; 1 rs10767664          0.19  0.03061224                      A</span></div>
<div class="sourceLine" id="cb8-19" data-line-number="19"><span class="co">#&gt; 2 rs13078807          0.10  0.02040816                      G</span></div>
<div class="sourceLine" id="cb8-20" data-line-number="20"><span class="co">#&gt; 3  rs1514175          0.07  0.02040816                      A</span></div>
<div class="sourceLine" id="cb8-21" data-line-number="21"><span class="co">#&gt; 4  rs1558902          0.39  0.02040816                      A</span></div>
<div class="sourceLine" id="cb8-22" data-line-number="22"><span class="co">#&gt; 5 rs10968576          0.11  0.02040816                      G</span></div>
<div class="sourceLine" id="cb8-23" data-line-number="23"><span class="co">#&gt; 6  rs2241423          0.13  0.02040816                      G</span></div>
<div class="sourceLine" id="cb8-24" data-line-number="24"><span class="co">#&gt;   other_allele.exposure eaf.exposure pval.exposure units.exposure</span></div>
<div class="sourceLine" id="cb8-25" data-line-number="25"><span class="co">#&gt; 1                     T         0.78         5e-26          kg/m2</span></div>
<div class="sourceLine" id="cb8-26" data-line-number="26"><span class="co">#&gt; 2                     A         0.20         4e-11          kg/m2</span></div>
<div class="sourceLine" id="cb8-27" data-line-number="27"><span class="co">#&gt; 3                     G         0.43         8e-14          kg/m2</span></div>
<div class="sourceLine" id="cb8-28" data-line-number="28"><span class="co">#&gt; 4                     T         0.42        5e-120          kg/m2</span></div>
<div class="sourceLine" id="cb8-29" data-line-number="29"><span class="co">#&gt; 5                     A         0.31         3e-13          kg/m2</span></div>
<div class="sourceLine" id="cb8-30" data-line-number="30"><span class="co">#&gt; 6                     A         0.78         1e-18          kg/m2</span></div>
<div class="sourceLine" id="cb8-31" data-line-number="31"><span class="co">#&gt;   gene.exposure samplesize.exposure exposure mr_keep.exposure</span></div>
<div class="sourceLine" id="cb8-32" data-line-number="32"><span class="co">#&gt; 1          BDNF              225238 exposure             TRUE</span></div>
<div class="sourceLine" id="cb8-33" data-line-number="33"><span class="co">#&gt; 2         CADM2              221431 exposure             TRUE</span></div>
<div class="sourceLine" id="cb8-34" data-line-number="34"><span class="co">#&gt; 3        TNNI3K              207641 exposure             TRUE</span></div>
<div class="sourceLine" id="cb8-35" data-line-number="35"><span class="co">#&gt; 4           FTO              222476 exposure             TRUE</span></div>
<div class="sourceLine" id="cb8-36" data-line-number="36"><span class="co">#&gt; 5        LRRN6C              247166 exposure             TRUE</span></div>
<div class="sourceLine" id="cb8-37" data-line-number="37"><span class="co">#&gt; 6       LBXCOR1              227886 exposure             TRUE</span></div>
<div class="sourceLine" id="cb8-38" data-line-number="38"><span class="co">#&gt;   pval_origin.exposure units.exposure_dat id.exposure data_source.exposure</span></div>
<div class="sourceLine" id="cb8-39" data-line-number="39"><span class="co">#&gt; 1             reported              kg/m2      TrGPJg             textfile</span></div>
<div class="sourceLine" id="cb8-40" data-line-number="40"><span class="co">#&gt; 2             reported              kg/m2      TrGPJg             textfile</span></div>
<div class="sourceLine" id="cb8-41" data-line-number="41"><span class="co">#&gt; 3             reported              kg/m2      TrGPJg             textfile</span></div>
<div class="sourceLine" id="cb8-42" data-line-number="42"><span class="co">#&gt; 4             reported              kg/m2      TrGPJg             textfile</span></div>
<div class="sourceLine" id="cb8-43" data-line-number="43"><span class="co">#&gt; 5             reported              kg/m2      TrGPJg             textfile</span></div>
<div class="sourceLine" id="cb8-44" data-line-number="44"><span class="co">#&gt; 6             reported              kg/m2      TrGPJg             textfile</span></div></code></pre>
<p>If the <code>Phenotype</code> column is not provided (as is the case in this example) then it will assume that the phenotype’s name is simply “exposure”. This is entered in the <code>exposure</code> column. It can be renamed manually:</p>
<pre class="sourceCode r" id="cb9"><code class="sourceCode r"><div class="sourceLine" id="cb9-1" data-line-number="1">bmi_exp_dat<span class="op">$</span>exposure &lt;-<span class="st"> "BMI"</span>
</div></code></pre>
</div>
</div>
<div id="using-an-existing-data-frame" class="section level2">
<h2 class="hasAnchor">
<a href="#using-an-existing-data-frame" class="anchor"></a>Using an existing data frame</h2>
<p>If the data already exists as a data frame in R then it can be converted into the correct format using the <code>format_data</code> function. For example, here is some randomly created data:</p>
<pre class="sourceCode r" id="cb10"><code class="sourceCode r"><div class="sourceLine" id="cb10-1" data-line-number="1">random_df &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(</div>
<div class="sourceLine" id="cb10-2" data-line-number="2">    <span class="dt">SNP =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"rs1"</span>, <span class="st">"rs2"</span>),</div>
<div class="sourceLine" id="cb10-3" data-line-number="3">    <span class="dt">beta =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dv">1</span>, <span class="dv">2</span>),</div>
<div class="sourceLine" id="cb10-4" data-line-number="4">    <span class="dt">se =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dv">1</span>, <span class="dv">2</span>),</div>
<div class="sourceLine" id="cb10-5" data-line-number="5">    <span class="dt">effect_allele =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"A"</span>, <span class="st">"T"</span>)</div>
<div class="sourceLine" id="cb10-6" data-line-number="6">)</div>
<div class="sourceLine" id="cb10-7" data-line-number="7">random_df</div>
<div class="sourceLine" id="cb10-8" data-line-number="8"><span class="co">#&gt;   SNP beta se effect_allele</span></div>
<div class="sourceLine" id="cb10-9" data-line-number="9"><span class="co">#&gt; 1 rs1    1  1             A</span></div>
<div class="sourceLine" id="cb10-10" data-line-number="10"><span class="co">#&gt; 2 rs2    2  2             T</span></div></code></pre>
<p>This can be formatted like so:</p>
<pre class="sourceCode r" id="cb11"><code class="sourceCode r"><div class="sourceLine" id="cb11-1" data-line-number="1">random_exp_dat &lt;-<span class="st"> </span><span class="kw"><a href="../reference/format_data.html">format_data</a></span>(random_df, <span class="dt">type=</span><span class="st">"exposure"</span>)</div>
<div class="sourceLine" id="cb11-2" data-line-number="2"><span class="co">#&gt; No phenotype name specified, defaulting to 'exposure'.</span></div>
<div class="sourceLine" id="cb11-3" data-line-number="3"><span class="co">#&gt; Warning in format_data(random_df, type = "exposure"): The following columns are not present but are helpful for harmonisation</span></div>
<div class="sourceLine" id="cb11-4" data-line-number="4"><span class="co">#&gt; other_alleleeaf</span></div>
<div class="sourceLine" id="cb11-5" data-line-number="5"><span class="co">#&gt; Warning in format_data(random_df, type = "exposure"): effect_allele column</span></div>
<div class="sourceLine" id="cb11-6" data-line-number="6"><span class="co">#&gt; is not character data. Coercing...</span></div>
<div class="sourceLine" id="cb11-7" data-line-number="7"><span class="co">#&gt; Inferring p-values</span></div>
<div class="sourceLine" id="cb11-8" data-line-number="8">random_exp_dat</div>
<div class="sourceLine" id="cb11-9" data-line-number="9"><span class="co">#&gt;   SNP beta.exposure se.exposure effect_allele.exposure exposure</span></div>
<div class="sourceLine" id="cb11-10" data-line-number="10"><span class="co">#&gt; 1 rs1             1           1                      A exposure</span></div>
<div class="sourceLine" id="cb11-11" data-line-number="11"><span class="co">#&gt; 2 rs2             2           2                      T exposure</span></div>
<div class="sourceLine" id="cb11-12" data-line-number="12"><span class="co">#&gt;   mr_keep.exposure pval.exposure pval_origin.exposure id.exposure</span></div>
<div class="sourceLine" id="cb11-13" data-line-number="13"><span class="co">#&gt; 1             TRUE     0.1586553             inferred      4Hnz2e</span></div>
<div class="sourceLine" id="cb11-14" data-line-number="14"><span class="co">#&gt; 2             TRUE     0.1586553             inferred      4Hnz2e</span></div>
<div class="sourceLine" id="cb11-15" data-line-number="15"><span class="co">#&gt;   other_allele.exposure eaf.exposure</span></div>
<div class="sourceLine" id="cb11-16" data-line-number="16"><span class="co">#&gt; 1                    NA           NA</span></div>
<div class="sourceLine" id="cb11-17" data-line-number="17"><span class="co">#&gt; 2                    NA           NA</span></div></code></pre>
</div>
<div id="obtaining-instruments-from-existing-catalogues" class="section level2">
<h2 class="hasAnchor">
<a href="#obtaining-instruments-from-existing-catalogues" class="anchor"></a>Obtaining instruments from existing catalogues</h2>
<p>A number of sources of instruments have already been curated and are available for use in MR Base. They are provided as data objects in the <code>MRInstruments</code> package. To install:</p>
<pre class="sourceCode r" id="cb12"><code class="sourceCode r"><div class="sourceLine" id="cb12-1" data-line-number="1">devtools<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/devtools/man/reexports.html">install_github</a></span>(<span class="st">"MRCIEU/MRInstruments"</span>)</div></code></pre>
<p>This package contains a number of data.frames, each of which is a repository of SNP-trait associations. How to access the data frames is detailed below:</p>
<div id="gwas-catalog" class="section level3">
<h3 class="hasAnchor">
<a href="#gwas-catalog" class="anchor"></a>GWAS catalog</h3>
<p>The NHGRI-EBI GWAS catalog contains a catalog of significant associations obtained from GWASs. This version of the data is filtered and harmonised to contain associations that have the required data to perform MR, to ensure that the units used to report effect sizes from a particular study are all the same, and other data cleaning operations.</p>
<p>To use the GWAS catalog:</p>
<pre class="sourceCode r" id="cb13"><code class="sourceCode r"><div class="sourceLine" id="cb13-1" data-line-number="1">
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(MRInstruments)</div>
<div class="sourceLine" id="cb13-2" data-line-number="2">
<span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(gwas_catalog)</div>
<div class="sourceLine" id="cb13-3" data-line-number="3">
<span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(gwas_catalog)</div>
<div class="sourceLine" id="cb13-4" data-line-number="4"><span class="co">#&gt;                       Phenotype_simple</span></div>
<div class="sourceLine" id="cb13-5" data-line-number="5"><span class="co">#&gt; 1                    Eosinophil counts</span></div>
<div class="sourceLine" id="cb13-6" data-line-number="6"><span class="co">#&gt; 2 Eosinophil percentage of white cells</span></div>
<div class="sourceLine" id="cb13-7" data-line-number="7"><span class="co">#&gt; 3             Post bronchodilator FEV1</span></div>
<div class="sourceLine" id="cb13-8" data-line-number="8"><span class="co">#&gt; 4        Cisplatin-induced ototoxicity</span></div>
<div class="sourceLine" id="cb13-9" data-line-number="9"><span class="co">#&gt; 5                      Crohn's disease</span></div>
<div class="sourceLine" id="cb13-10" data-line-number="10"><span class="co">#&gt; 6                    IgG glycosylation</span></div>
<div class="sourceLine" id="cb13-11" data-line-number="11"><span class="co">#&gt;                                       MAPPED_TRAIT_EFO</span></div>
<div class="sourceLine" id="cb13-12" data-line-number="12"><span class="co">#&gt; 1                                     eosinophil count</span></div>
<div class="sourceLine" id="cb13-13" data-line-number="13"><span class="co">#&gt; 2                  eosinophil percentage of leukocytes</span></div>
<div class="sourceLine" id="cb13-14" data-line-number="14"><span class="co">#&gt; 3 forced expiratory volume, response to bronchodilator</span></div>
<div class="sourceLine" id="cb13-15" data-line-number="15"><span class="co">#&gt; 4                   ototoxicity, response to cisplatin</span></div>
<div class="sourceLine" id="cb13-16" data-line-number="16"><span class="co">#&gt; 5                                      Crohn's disease</span></div>
<div class="sourceLine" id="cb13-17" data-line-number="17"><span class="co">#&gt; 6                  serum IgG glycosylation measurement</span></div>
<div class="sourceLine" id="cb13-18" data-line-number="18"><span class="co">#&gt;                                                              MAPPED_TRAIT_EFO_URI</span></div>
<div class="sourceLine" id="cb13-19" data-line-number="19"><span class="co">#&gt; 1                                            http://www.ebi.ac.uk/efo/EFO_0004842</span></div>
<div class="sourceLine" id="cb13-20" data-line-number="20"><span class="co">#&gt; 2                                            http://www.ebi.ac.uk/efo/EFO_0007991</span></div>
<div class="sourceLine" id="cb13-21" data-line-number="21"><span class="co">#&gt; 3 http://www.ebi.ac.uk/efo/EFO_0004314, http://purl.obolibrary.org/obo/GO_0097366</span></div>
<div class="sourceLine" id="cb13-22" data-line-number="22"><span class="co">#&gt; 4 http://www.ebi.ac.uk/efo/EFO_0006951, http://purl.obolibrary.org/obo/GO_0072718</span></div>
<div class="sourceLine" id="cb13-23" data-line-number="23"><span class="co">#&gt; 5                                            http://www.ebi.ac.uk/efo/EFO_0000384</span></div>
<div class="sourceLine" id="cb13-24" data-line-number="24"><span class="co">#&gt; 6                                            http://www.ebi.ac.uk/efo/EFO_0005193</span></div>
<div class="sourceLine" id="cb13-25" data-line-number="25"><span class="co">#&gt;                                                                                                                                                Initial_sample_description</span></div>
<div class="sourceLine" id="cb13-26" data-line-number="26"><span class="co">#&gt; 1                                                                                                                                   172,275 European ancestry individuals</span></div>
<div class="sourceLine" id="cb13-27" data-line-number="27"><span class="co">#&gt; 2                                                                                                                                   172,378 European ancestry individuals</span></div>
<div class="sourceLine" id="cb13-28" data-line-number="28"><span class="co">#&gt; 3 10,094 European ancestry current and former smoker individuals, 3,260 African American current and former smoker individuals, 178 current and former smoker individuals</span></div>
<div class="sourceLine" id="cb13-29" data-line-number="29"><span class="co">#&gt; 4                                                                                                            511 European ancestry testicular cancer survivor individuals</span></div>
<div class="sourceLine" id="cb13-30" data-line-number="30"><span class="co">#&gt; 5                                                                                                         1,748 European ancestry cases, 2,938 European ancestry controls</span></div>
<div class="sourceLine" id="cb13-31" data-line-number="31"><span class="co">#&gt; 6                                                                                                                                     2,247 European ancestry individuals</span></div>
<div class="sourceLine" id="cb13-32" data-line-number="32"><span class="co">#&gt;   Replication_sample_description STUDY.ACCESSION</span></div>
<div class="sourceLine" id="cb13-33" data-line-number="33"><span class="co">#&gt; 1                           &lt;NA&gt;      GCST004606</span></div>
<div class="sourceLine" id="cb13-34" data-line-number="34"><span class="co">#&gt; 2                           &lt;NA&gt;      GCST004600</span></div>
<div class="sourceLine" id="cb13-35" data-line-number="35"><span class="co">#&gt; 3                           &lt;NA&gt;      GCST003262</span></div>
<div class="sourceLine" id="cb13-36" data-line-number="36"><span class="co">#&gt; 4                   238 children      GCST003898</span></div>
<div class="sourceLine" id="cb13-37" data-line-number="37"><span class="co">#&gt; 5              (see Parkes 2007)      GCST000042</span></div>
<div class="sourceLine" id="cb13-38" data-line-number="38"><span class="co">#&gt; 6                           &lt;NA&gt;      GCST001848</span></div>
<div class="sourceLine" id="cb13-39" data-line-number="39"><span class="co">#&gt;                              Phenotype Phenotype_info PubmedID     Author</span></div>
<div class="sourceLine" id="cb13-40" data-line-number="40"><span class="co">#&gt; 1                    Eosinophil counts                27863252   Astle WJ</span></div>
<div class="sourceLine" id="cb13-41" data-line-number="41"><span class="co">#&gt; 2 Eosinophil percentage of white cells                27863252   Astle WJ</span></div>
<div class="sourceLine" id="cb13-42" data-line-number="42"><span class="co">#&gt; 3             Post bronchodilator FEV1                26634245    Lutz SM</span></div>
<div class="sourceLine" id="cb13-43" data-line-number="43"><span class="co">#&gt; 4        Cisplatin-induced ototoxicity                28039263 Wheeler HE</span></div>
<div class="sourceLine" id="cb13-44" data-line-number="44"><span class="co">#&gt; 5                      Crohn's disease                17554300      WTCCC</span></div>
<div class="sourceLine" id="cb13-45" data-line-number="45"><span class="co">#&gt; 6                    IgG glycosylation        (IGP23) 23382691     Lauc G</span></div>
<div class="sourceLine" id="cb13-46" data-line-number="46"><span class="co">#&gt;   Year        SNP chr bp_ens_GRCh38   Region       gene</span></div>
<div class="sourceLine" id="cb13-47" data-line-number="47"><span class="co">#&gt; 1 2016  rs1000005  21      33060745 21q22.11 AP000282.2</span></div>
<div class="sourceLine" id="cb13-48" data-line-number="48"><span class="co">#&gt; 2 2016  rs1000005  21      33060745 21q22.11 AP000282.2</span></div>
<div class="sourceLine" id="cb13-49" data-line-number="49"><span class="co">#&gt; 3 2015 rs10000225   4     144312789  4q31.21 Intergenic</span></div>
<div class="sourceLine" id="cb13-50" data-line-number="50"><span class="co">#&gt; 4 2016  rs1000083   5      97197765     5q15 intergenic</span></div>
<div class="sourceLine" id="cb13-51" data-line-number="51"><span class="co">#&gt; 5 2007  rs1000113   5     150860514   5q33.1       IRGM</span></div>
<div class="sourceLine" id="cb13-52" data-line-number="52"><span class="co">#&gt; 6 2013 rs10002492   4      12578222  4p15.33         NR</span></div>
<div class="sourceLine" id="cb13-53" data-line-number="53"><span class="co">#&gt;               Gene_ens effect_allele other_allele        beta          se</span></div>
<div class="sourceLine" id="cb13-54" data-line-number="54"><span class="co">#&gt; 1 AP000282.2,LINC00945             C            G -0.02481715 0.003571429</span></div>
<div class="sourceLine" id="cb13-55" data-line-number="55"><span class="co">#&gt; 2 AP000282.2,LINC00945             C            G -0.02652552 0.003826531</span></div>
<div class="sourceLine" id="cb13-56" data-line-number="56"><span class="co">#&gt; 3           Intergenic             A            T -0.04400000 0.009420188</span></div>
<div class="sourceLine" id="cb13-57" data-line-number="57"><span class="co">#&gt; 4           intergenic             T            C -4.55000000 0.996760478</span></div>
<div class="sourceLine" id="cb13-58" data-line-number="58"><span class="co">#&gt; 5                 IRGM             T            C  0.43178242 0.083879940</span></div>
<div class="sourceLine" id="cb13-59" data-line-number="59"><span class="co">#&gt; 6                   NR             G            A -0.18370000 0.040816327</span></div>
<div class="sourceLine" id="cb13-60" data-line-number="60"><span class="co">#&gt;    pval            units       eaf date_added_to_MRBASE</span></div>
<div class="sourceLine" id="cb13-61" data-line-number="61"><span class="co">#&gt; 1 7e-12    unit decrease 0.5894000           2018-02-14</span></div>
<div class="sourceLine" id="cb13-62" data-line-number="62"><span class="co">#&gt; 2 2e-13    unit decrease 0.5894000           2018-02-14</span></div>
<div class="sourceLine" id="cb13-63" data-line-number="63"><span class="co">#&gt; 3 3e-06 NR unit decrease 0.3500000           2018-02-14</span></div>
<div class="sourceLine" id="cb13-64" data-line-number="64"><span class="co">#&gt; 4 5e-06 z score decrease        NA           2018-02-14</span></div>
<div class="sourceLine" id="cb13-65" data-line-number="65"><span class="co">#&gt; 5 3e-07             &lt;NA&gt; 0.0700000           2018-02-14</span></div>
<div class="sourceLine" id="cb13-66" data-line-number="66"><span class="co">#&gt; 6 8e-06    unit decrease 0.1672272           2018-02-14</span></div></code></pre>
<p>For example, to obtain instruments for body mass index using the Speliotes et al 2010 study:</p>
<pre class="sourceCode r" id="cb14"><code class="sourceCode r"><div class="sourceLine" id="cb14-1" data-line-number="1">bmi_gwas &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/subset.html">subset</a></span>(gwas_catalog, <span class="kw"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span>(<span class="st">"Speliotes"</span>, Author) <span class="op">&amp;</span><span class="st"> </span>Phenotype <span class="op">==</span><span class="st"> "Body mass index"</span>)</div>
<div class="sourceLine" id="cb14-2" data-line-number="2">bmi_exp_dat &lt;-<span class="st"> </span><span class="kw"><a href="../reference/format_data.html">format_data</a></span>(bmi_gwas)</div></code></pre>
</div>
<div id="metabolites" class="section level3">
<h3 class="hasAnchor">
<a href="#metabolites" class="anchor"></a>Metabolites</h3>
<p>Independent top hits from GWASs on 121 metabolites in whole blood are stored in the <code>metab_qtls</code> data object. Use <code><a href="https://rdrr.io/pkg/MRInstruments/man/metab_qtls.html">?metab_qtls</a></code> to get more information.</p>
<pre class="sourceCode r" id="cb15"><code class="sourceCode r"><div class="sourceLine" id="cb15-1" data-line-number="1">
<span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(metab_qtls)</div>
<div class="sourceLine" id="cb15-2" data-line-number="2">
<span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(metab_qtls)</div>
<div class="sourceLine" id="cb15-3" data-line-number="3"><span class="co">#&gt;   phenotype chromosome  position       SNP effect_allele other_allele</span></div>
<div class="sourceLine" id="cb15-4" data-line-number="4"><span class="co">#&gt; 1     AcAce          8   9181395 rs2169387             G            A</span></div>
<div class="sourceLine" id="cb15-5" data-line-number="5"><span class="co">#&gt; 2     AcAce         11 116648917  rs964184             C            G</span></div>
<div class="sourceLine" id="cb15-6" data-line-number="6"><span class="co">#&gt; 3       Ace          6  12042473 rs6933521             C            T</span></div>
<div class="sourceLine" id="cb15-7" data-line-number="7"><span class="co">#&gt; 4       Ala          2  27730940 rs1260326             C            T</span></div>
<div class="sourceLine" id="cb15-8" data-line-number="8"><span class="co">#&gt; 5       Ala          2  65220910 rs2160387             C            T</span></div>
<div class="sourceLine" id="cb15-9" data-line-number="9"><span class="co">#&gt; 6       Ala         12  47201814 rs4554975             G            A</span></div>
<div class="sourceLine" id="cb15-10" data-line-number="10"><span class="co">#&gt;        eaf      beta       se     pval n_studies     n</span></div>
<div class="sourceLine" id="cb15-11" data-line-number="11"><span class="co">#&gt; 1 0.870251  0.085630 0.015451 3.61e-08        11 19257</span></div>
<div class="sourceLine" id="cb15-12" data-line-number="12"><span class="co">#&gt; 2 0.857715 -0.096027 0.014624 6.71e-11        11 19261</span></div>
<div class="sourceLine" id="cb15-13" data-line-number="13"><span class="co">#&gt; 3 0.120256 -0.091667 0.015885 8.10e-09        14 24742</span></div>
<div class="sourceLine" id="cb15-14" data-line-number="14"><span class="co">#&gt; 4 0.638817 -0.104582 0.009940 7.40e-26        13 22569</span></div>
<div class="sourceLine" id="cb15-15" data-line-number="15"><span class="co">#&gt; 5 0.403170 -0.071001 0.009603 1.49e-13        14 24793</span></div>
<div class="sourceLine" id="cb15-16" data-line-number="16"><span class="co">#&gt; 6 0.644059 -0.069135 0.009598 6.12e-13        14 24792</span></div></code></pre>
<p>For example, to obtain instruments for the Alanine:</p>
<pre class="sourceCode r" id="cb16"><code class="sourceCode r"><div class="sourceLine" id="cb16-1" data-line-number="1">ala_exp_dat &lt;-<span class="st"> </span><span class="kw"><a href="../reference/format_metab_qtls.html">format_metab_qtls</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/subset.html">subset</a></span>(metab_qtls, phenotype<span class="op">==</span><span class="st">"Ala"</span>))</div></code></pre>
</div>
<div id="proteins" class="section level3">
<h3 class="hasAnchor">
<a href="#proteins" class="anchor"></a>Proteins</h3>
<p>Independent top hits from GWASs on 47 protein levels in whole blood are stored in the <code>proteomic_qtls</code> data object. Use <code><a href="https://rdrr.io/pkg/MRInstruments/man/proteomic_qtls.html">?proteomic_qtls</a></code> to get more information.</p>
<pre class="sourceCode r" id="cb17"><code class="sourceCode r"><div class="sourceLine" id="cb17-1" data-line-number="1">
<span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(proteomic_qtls)</div>
<div class="sourceLine" id="cb17-2" data-line-number="2">
<span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(proteomic_qtls)</div>
<div class="sourceLine" id="cb17-3" data-line-number="3"><span class="co">#&gt;   analyte chr  position        SNP  gene location annotation other_allele</span></div>
<div class="sourceLine" id="cb17-4" data-line-number="4"><span class="co">#&gt; 1   CFHR1   1 196698945 rs12144939   CFH      cis   missense            T</span></div>
<div class="sourceLine" id="cb17-5" data-line-number="5"><span class="co">#&gt; 2    IL6r   1 154425456 rs12126142  IL6R      cis   missense            A</span></div>
<div class="sourceLine" id="cb17-6" data-line-number="6"><span class="co">#&gt; 3   ApoA4  11 116677723  rs1263167 APOA4      cis intergenic            G</span></div>
<div class="sourceLine" id="cb17-7" data-line-number="7"><span class="co">#&gt; 4    SELE   9 136149399   rs507666   ABO    trans   intronic            A</span></div>
<div class="sourceLine" id="cb17-8" data-line-number="8"><span class="co">#&gt; 5 FetuinA   3 186335941  rs2070633  AHSG      cis   missense            T</span></div>
<div class="sourceLine" id="cb17-9" data-line-number="9"><span class="co">#&gt; 6     ACE  17  61566031     rs4343   ACE      cis synonymous            A</span></div>
<div class="sourceLine" id="cb17-10" data-line-number="10"><span class="co">#&gt;   effect_allele   eaf   maf      pval   beta         se</span></div>
<div class="sourceLine" id="cb17-11" data-line-number="11"><span class="co">#&gt; 1             G 0.643 0.357 8.99e-143 -1.108 0.04355258</span></div>
<div class="sourceLine" id="cb17-12" data-line-number="12"><span class="co">#&gt; 2             G 0.608 0.392 1.81e-106  0.850 0.03878364</span></div>
<div class="sourceLine" id="cb17-13" data-line-number="13"><span class="co">#&gt; 3             A 0.803 0.197  2.64e-54 -0.919 0.05922332</span></div>
<div class="sourceLine" id="cb17-14" data-line-number="14"><span class="co">#&gt; 4             G 0.809 0.191  1.01e-52 -0.882 0.05771545</span></div>
<div class="sourceLine" id="cb17-15" data-line-number="15"><span class="co">#&gt; 5             C 0.676 0.324  2.88e-44 -0.629 0.04506925</span></div>
<div class="sourceLine" id="cb17-16" data-line-number="16"><span class="co">#&gt; 6             G 0.508 0.492  6.66e-44  0.493 0.03547679</span></div></code></pre>
<p>For example, to obtain instruments for the ApoH protein:</p>
<pre class="sourceCode r" id="cb18"><code class="sourceCode r"><div class="sourceLine" id="cb18-1" data-line-number="1">apoh_exp_dat &lt;-<span class="st"> </span><span class="kw"><a href="../reference/format_proteomic_qtls.html">format_proteomic_qtls</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/subset.html">subset</a></span>(proteomic_qtls, analyte<span class="op">==</span><span class="st">"ApoH"</span>))</div>
<div class="sourceLine" id="cb18-2" data-line-number="2"><span class="co">#&gt; Warning in format_data(proteomic_qtls_subset, type = type, phenotype_col =</span></div>
<div class="sourceLine" id="cb18-3" data-line-number="3"><span class="co">#&gt; "analyte"): effect_allele column is not character data. Coercing...</span></div>
<div class="sourceLine" id="cb18-4" data-line-number="4"><span class="co">#&gt; Warning in format_data(proteomic_qtls_subset, type = type, phenotype_col =</span></div>
<div class="sourceLine" id="cb18-5" data-line-number="5"><span class="co">#&gt; "analyte"): other_allele column is not character data. Coercing...</span></div></code></pre>
</div>
<div id="gene-expression-levels" class="section level3">
<h3 class="hasAnchor">
<a href="#gene-expression-levels" class="anchor"></a>Gene expression levels</h3>
<p>Independent top hits from GWASs on 32432 gene identifiers and in 44 tissues are available from the GTEX study in <code>gtex_eqtl</code>. Use <code><a href="https://rdrr.io/pkg/MRInstruments/man/gtex_eqtl.html">?gtex_eqtl</a></code> to get more information.</p>
<pre class="sourceCode r" id="cb19"><code class="sourceCode r"><div class="sourceLine" id="cb19-1" data-line-number="1">
<span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(gtex_eqtl)</div>
<div class="sourceLine" id="cb19-2" data-line-number="2">
<span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(gtex_eqtl)</div>
<div class="sourceLine" id="cb19-3" data-line-number="3"><span class="co">#&gt;                 tissue     gene_name gene_start         SNP snp_position</span></div>
<div class="sourceLine" id="cb19-4" data-line-number="4"><span class="co">#&gt; 1 Adipose Subcutaneous RP4-669L17.10   1:317720   rs2519065     1:787151</span></div>
<div class="sourceLine" id="cb19-5" data-line-number="5"><span class="co">#&gt; 2 Adipose Subcutaneous RP11-206L10.1   1:661611  rs11804171     1:723819</span></div>
<div class="sourceLine" id="cb19-6" data-line-number="6"><span class="co">#&gt; 3 Adipose Subcutaneous RP11-206L10.3   1:677193 rs149110718     1:759227</span></div>
<div class="sourceLine" id="cb19-7" data-line-number="7"><span class="co">#&gt; 4 Adipose Subcutaneous RP11-206L10.2   1:700306 rs148649543     1:752796</span></div>
<div class="sourceLine" id="cb19-8" data-line-number="8"><span class="co">#&gt; 5 Adipose Subcutaneous RP11-206L10.9   1:714150  rs12184279     1:717485</span></div>
<div class="sourceLine" id="cb19-9" data-line-number="9"><span class="co">#&gt; 6 Adipose Subcutaneous RP11-206L10.8   1:736259  rs10454454     1:754954</span></div>
<div class="sourceLine" id="cb19-10" data-line-number="10"><span class="co">#&gt;   effect_allele other_allele      beta        se        pval   n</span></div>
<div class="sourceLine" id="cb19-11" data-line-number="11"><span class="co">#&gt; 1             A            G  0.551788 0.0747180 2.14627e-12 298</span></div>
<div class="sourceLine" id="cb19-12" data-line-number="12"><span class="co">#&gt; 2             A            T -0.917475 0.1150060 4.99967e-14 298</span></div>
<div class="sourceLine" id="cb19-13" data-line-number="13"><span class="co">#&gt; 3             T            C  0.807571 0.1776530 8.44694e-06 298</span></div>
<div class="sourceLine" id="cb19-14" data-line-number="14"><span class="co">#&gt; 4             T            C  0.745393 0.0958531 1.82660e-13 298</span></div>
<div class="sourceLine" id="cb19-15" data-line-number="15"><span class="co">#&gt; 5             A            C  1.927250 0.2247390 9.55098e-16 298</span></div>
<div class="sourceLine" id="cb19-16" data-line-number="16"><span class="co">#&gt; 6             A            G  1.000400 0.1787470 5.61079e-08 298</span></div></code></pre>
<p>For example, to obtain instruments for the IRAK1BP1 gene expression levels in subcutaneous adipose tissue:</p>
<pre class="sourceCode r" id="cb20"><code class="sourceCode r"><div class="sourceLine" id="cb20-1" data-line-number="1">irak1bp1_exp_dat &lt;-<span class="st"> </span><span class="kw"><a href="../reference/format_gtex_eqtl.html">format_gtex_eqtl</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/subset.html">subset</a></span>(gtex_eqtl, gene_name<span class="op">==</span><span class="st">"IRAK1BP1"</span> <span class="op">&amp;</span><span class="st"> </span>tissue <span class="op">==</span><span class="st"> "Adipose Subcutaneous"</span>))</div>
<div class="sourceLine" id="cb20-2" data-line-number="2"><span class="co">#&gt; Warning in format_data(gtex_eqtl_subset, type = type, phenotype_col = type, : The following columns are not present but are helpful for harmonisation</span></div>
<div class="sourceLine" id="cb20-3" data-line-number="3"><span class="co">#&gt; eaf</span></div>
<div class="sourceLine" id="cb20-4" data-line-number="4"><span class="co">#&gt; Inferring p-values</span></div></code></pre>
</div>
<div id="dna-methylation-levels" class="section level3">
<h3 class="hasAnchor">
<a href="#dna-methylation-levels" class="anchor"></a>DNA methylation levels</h3>
<p>Independent top hits from GWASs on 0 DNA methylation levels in whole blood across 5 time points are available from the ARIES study in <code>aries_mqtl</code>. Use <code><a href="https://rdrr.io/pkg/MRInstruments/man/aries_mqtl.html">?aries_mqtl</a></code> to get more information.</p>
<pre class="sourceCode r" id="cb21"><code class="sourceCode r"><div class="sourceLine" id="cb21-1" data-line-number="1">
<span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(aries_mqtl)</div>
<div class="sourceLine" id="cb21-2" data-line-number="2">
<span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(aries_mqtl)</div>
<div class="sourceLine" id="cb21-3" data-line-number="3"><span class="co">#&gt;          SNP timepoint        cpg    beta        pval         se snp_chr</span></div>
<div class="sourceLine" id="cb21-4" data-line-number="4"><span class="co">#&gt; 1 esv2656832         1 cg21826606  0.3459 1.60408e-26 0.03265336       1</span></div>
<div class="sourceLine" id="cb21-5" data-line-number="5"><span class="co">#&gt; 2 esv2658098         1 cg22681495 -0.6263 1.55765e-66 0.03643240      15</span></div>
<div class="sourceLine" id="cb21-6" data-line-number="6"><span class="co">#&gt; 3 esv2660043         1 cg24276624 -0.5772 3.16370e-26 0.05481823      11</span></div>
<div class="sourceLine" id="cb21-7" data-line-number="7"><span class="co">#&gt; 4 esv2660043         1 cg11157765 -0.5423 1.33928e-22 0.05583777      11</span></div>
<div class="sourceLine" id="cb21-8" data-line-number="8"><span class="co">#&gt; 5 esv2660673         1 cg05832925 -0.5919 2.88011e-50 0.03982467      11</span></div>
<div class="sourceLine" id="cb21-9" data-line-number="9"><span class="co">#&gt; 6 esv2660769         1 cg05859533 -0.6224 1.49085e-58 0.03868158      16</span></div>
<div class="sourceLine" id="cb21-10" data-line-number="10"><span class="co">#&gt;    snp_pos effect_allele other_allele    eaf   sex   age    units</span></div>
<div class="sourceLine" id="cb21-11" data-line-number="11"><span class="co">#&gt; 1 25591901             I            R 0.3974 mixed Birth SD units</span></div>
<div class="sourceLine" id="cb21-12" data-line-number="12"><span class="co">#&gt; 2 86057007             D            R 0.2076 mixed Birth SD units</span></div>
<div class="sourceLine" id="cb21-13" data-line-number="13"><span class="co">#&gt; 3 69982552             D            R 0.1450 mixed Birth SD units</span></div>
<div class="sourceLine" id="cb21-14" data-line-number="14"><span class="co">#&gt; 4 69982552             D            R 0.1450 mixed Birth SD units</span></div>
<div class="sourceLine" id="cb21-15" data-line-number="15"><span class="co">#&gt; 5 74024905             D            R 0.1671 mixed Birth SD units</span></div>
<div class="sourceLine" id="cb21-16" data-line-number="16"><span class="co">#&gt; 6 57725395             D            R 0.2136 mixed Birth SD units</span></div>
<div class="sourceLine" id="cb21-17" data-line-number="17"><span class="co">#&gt;   island_location cpg_chr  cpg_pos    gene gene_location cis_trans</span></div>
<div class="sourceLine" id="cb21-18" data-line-number="18"><span class="co">#&gt; 1         N_Shore       1 25593055                             cis</span></div>
<div class="sourceLine" id="cb21-19" data-line-number="19"><span class="co">#&gt; 2                      15 86058755  AKAP13          Body       cis</span></div>
<div class="sourceLine" id="cb21-20" data-line-number="20"><span class="co">#&gt; 3                      11 69982941    ANO1          Body       cis</span></div>
<div class="sourceLine" id="cb21-21" data-line-number="21"><span class="co">#&gt; 4                      11 69982996    ANO1          Body       cis</span></div>
<div class="sourceLine" id="cb21-22" data-line-number="22"><span class="co">#&gt; 5         S_Shelf      11 74026371                             cis</span></div>
<div class="sourceLine" id="cb21-23" data-line-number="23"><span class="co">#&gt; 6                      16 57727230 CCDC135       TSS1500       cis</span></div></code></pre>
<p>For example, to obtain instruments for cg25212131 CpG DNA methylation levels in at birth:</p>
<pre class="sourceCode r" id="cb22"><code class="sourceCode r"><div class="sourceLine" id="cb22-1" data-line-number="1">cg25212131_exp_dat &lt;-<span class="st"> </span><span class="kw"><a href="../reference/format_aries_mqtl.html">format_aries_mqtl</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/subset.html">subset</a></span>(aries_mqtl, cpg<span class="op">==</span><span class="st">"cg25212131"</span> <span class="op">&amp;</span><span class="st"> </span>age <span class="op">==</span><span class="st"> "Birth"</span>))</div></code></pre>
</div>
<div id="mr-base-gwas-database" class="section level3">
<h3 class="hasAnchor">
<a href="#mr-base-gwas-database" class="anchor"></a>MR Base GWAS database</h3>
<p>The MR Base GWAS database contains the entire summary statistics for hundreds of GWASs. You can use this database to define the instruments for a particular exposure. You can also use this database to obtain the effects for constructing polygenic risk scores using different p-value thresholds.</p>
<p>For example, to obtain details about the available GWASs do the following:</p>
<pre class="sourceCode r" id="cb23"><code class="sourceCode r"><div class="sourceLine" id="cb23-1" data-line-number="1">ao &lt;-<span class="st"> </span><span class="kw"><a href="../reference/available_outcomes.html">available_outcomes</a></span>()</div>
<div class="sourceLine" id="cb23-2" data-line-number="2"><span class="co">#&gt; Warning: 'available_outcomes' is deprecated.</span></div>
<div class="sourceLine" id="cb23-3" data-line-number="3"><span class="co">#&gt; Use 'ieugwasr::gwasinfo()' instead.</span></div>
<div class="sourceLine" id="cb23-4" data-line-number="4"><span class="co">#&gt; See help("Deprecated")</span></div>
<div class="sourceLine" id="cb23-5" data-line-number="5"><span class="co">#&gt; API: release: http://gwasapi.mrcieu.ac.uk/</span></div>
<div class="sourceLine" id="cb23-6" data-line-number="6">
<span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(ao)</div>
<div class="sourceLine" id="cb23-7" data-line-number="7"><span class="co">#&gt; # A tibble: 6 x 19</span></div>
<div class="sourceLine" id="cb23-8" data-line-number="8"><span class="co">#&gt;   id    trait sample_size   nsnp  year consortium author note  ncase    mr</span></div>
<div class="sourceLine" id="cb23-9" data-line-number="9"><span class="co">#&gt;   &lt;chr&gt; &lt;chr&gt;       &lt;int&gt;  &lt;int&gt; &lt;int&gt; &lt;chr&gt;      &lt;chr&gt;  &lt;chr&gt; &lt;int&gt; &lt;int&gt;</span></div>
<div class="sourceLine" id="cb23-10" data-line-number="10"><span class="co">#&gt; 1 UKB-… Diag…      463010 9.85e6  2018 MRC-IEU    Ben E… 4120…  1870     1</span></div>
<div class="sourceLine" id="cb23-11" data-line-number="11"><span class="co">#&gt; 2 UKB-… Diag…      463010 9.85e6  2018 MRC-IEU    Ben E… 4120…    15     1</span></div>
<div class="sourceLine" id="cb23-12" data-line-number="12"><span class="co">#&gt; 3 UKB-… Diag…      463010 9.85e6  2018 MRC-IEU    Ben E… 4120…   445     1</span></div>
<div class="sourceLine" id="cb23-13" data-line-number="13"><span class="co">#&gt; 4 UKB-… Diag…      463010 9.85e6  2018 MRC-IEU    Ben E… 4120…    28     1</span></div>
<div class="sourceLine" id="cb23-14" data-line-number="14"><span class="co">#&gt; 5 UKB-… Diag…      463010 9.85e6  2018 MRC-IEU    Ben E… 4120…    15     1</span></div>
<div class="sourceLine" id="cb23-15" data-line-number="15"><span class="co">#&gt; 6 UKB-… Diag…      463010 9.85e6  2018 MRC-IEU    Ben E… 4120…  1380     1</span></div>
<div class="sourceLine" id="cb23-16" data-line-number="16"><span class="co">#&gt; # … with 9 more variables: sex &lt;chr&gt;, pmid &lt;int&gt;, priority &lt;int&gt;,</span></div>
<div class="sourceLine" id="cb23-17" data-line-number="17"><span class="co">#&gt; #   population &lt;chr&gt;, unit &lt;chr&gt;, ncontrol &lt;int&gt;, category &lt;chr&gt;,</span></div>
<div class="sourceLine" id="cb23-18" data-line-number="18"><span class="co">#&gt; #   subcategory &lt;chr&gt;, sd &lt;dbl&gt;</span></div></code></pre>
<p>For information about authentication see the <a href="#Authentication">Authentication</a> section.</p>
<p>The <code>available_outcomes</code> function returns a table of all the available studies in the database. Each study has a unique ID. e.g.</p>
<pre class="sourceCode r" id="cb24"><code class="sourceCode r"><div class="sourceLine" id="cb24-1" data-line-number="1">
<span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/subset.html">subset</a></span>(ao, <span class="dt">select=</span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(trait, id)))</div>
<div class="sourceLine" id="cb24-2" data-line-number="2"><span class="co">#&gt; # A tibble: 6 x 2</span></div>
<div class="sourceLine" id="cb24-3" data-line-number="3"><span class="co">#&gt;   trait                                                          id        </span></div>
<div class="sourceLine" id="cb24-4" data-line-number="4"><span class="co">#&gt;   &lt;chr&gt;                                                          &lt;chr&gt;     </span></div>
<div class="sourceLine" id="cb24-5" data-line-number="5"><span class="co">#&gt; 1 Diagnoses - main ICD10: D12.6 Colon, unspecified               UKB-b:192…</span></div>
<div class="sourceLine" id="cb24-6" data-line-number="6"><span class="co">#&gt; 2 Diagnoses - secondary ICD10: S63.0 Dislocation of wrist        UKB-b:1398</span></div>
<div class="sourceLine" id="cb24-7" data-line-number="7"><span class="co">#&gt; 3 Diagnoses - secondary ICD10: L90.5 Scar conditions and fibros… UKB-b:3692</span></div>
<div class="sourceLine" id="cb24-8" data-line-number="8"><span class="co">#&gt; 4 Diagnoses - secondary ICD10: I11.0 Hypertensive heart disease… UKB-b:138…</span></div>
<div class="sourceLine" id="cb24-9" data-line-number="9"><span class="co">#&gt; 5 Diagnoses - secondary ICD10: L01.0 Impetigo [any organism] [a… UKB-b:2176</span></div>
<div class="sourceLine" id="cb24-10" data-line-number="10"><span class="co">#&gt; 6 Diagnoses - main ICD10: C67.9 Bladder, unspecified             UKB-b:5421</span></div></code></pre>
<p>To extract instruments for a particular trait using a particular study, for example to obtain SNPs for body mass index using the Locke et al 2015 GIANT study, you specify the study ID as follows:</p>
<pre class="sourceCode r" id="cb25"><code class="sourceCode r"><div class="sourceLine" id="cb25-1" data-line-number="1">bmi2014_exp_dat &lt;-<span class="st"> </span><span class="kw"><a href="../reference/extract_instruments.html">extract_instruments</a></span>(<span class="dt">outcomes=</span><span class="dv">2</span>)</div>
<div class="sourceLine" id="cb25-2" data-line-number="2"><span class="co">#&gt; Warning: 'extract_instruments' is deprecated.</span></div>
<div class="sourceLine" id="cb25-3" data-line-number="3"><span class="co">#&gt; Use 'ieugwasr::tophits()' instead.</span></div>
<div class="sourceLine" id="cb25-4" data-line-number="4"><span class="co">#&gt; See help("Deprecated")</span></div>
<div class="sourceLine" id="cb25-5" data-line-number="5"><span class="co">#&gt; Requesting default values. Extracting from pre-clumped data</span></div>
<div class="sourceLine" id="cb25-6" data-line-number="6"><span class="co">#&gt; 1 out of 1 requested outcomes have pre-calculated instruments.</span></div></code></pre>
<p>This returns a set of LD clumped SNPs that are GWAS significant for BMI. You can specify various parameters for this function:</p>
<ul>
<li>
<code>p1</code> = P-value threshold for keeping a SNP</li>
<li>
<code>clump</code> = Whether or not to return independent SNPs only (default=TRUE)</li>
<li>
<code>r2</code> = The maximum LD R-square allowed between returned SNPs</li>
<li>
<code>kb</code> = The distance in which to search for LD R-square values</li>
</ul>
<p>By changing changing the <code>p1</code> parameter it is possible to obtain SNP effects for constructing polygenic risk scores.</p>
</div>
</div>
<div id="clumping" class="section level2">
<h2 class="hasAnchor">
<a href="#clumping" class="anchor"></a>Clumping</h2>
<p>For standard two sample MR it is important to ensure that the instruments for the exposure are independent. Once instruments have been identified for an exposure variable, MR Base can be used to perform clumping.</p>
<p>The European samples from the 1000 genomes project are used to estimate LD between SNPs. You can provide a list of SNP IDs, the SNPs will be extracted from 1000 genomes data, LD calculated between them, and amongst those SNPs that have LD R-square above the specified threshold only the SNP with the lowest P-value will be retained. To do this, use the following command:</p>
<pre class="sourceCode r" id="cb26"><code class="sourceCode r"><div class="sourceLine" id="cb26-1" data-line-number="1">bmi_exp_dat &lt;-<span class="st"> </span><span class="kw"><a href="../reference/clump_data.html">clump_data</a></span>(bmi_exp_dat)</div>
<div class="sourceLine" id="cb26-2" data-line-number="2"><span class="co">#&gt; Warning: 'clump_data' is deprecated.</span></div>
<div class="sourceLine" id="cb26-3" data-line-number="3"><span class="co">#&gt; Use 'ieugwasr::ld_clump()' instead.</span></div>
<div class="sourceLine" id="cb26-4" data-line-number="4"><span class="co">#&gt; See help("Deprecated")</span></div>
<div class="sourceLine" id="cb26-5" data-line-number="5"><span class="co">#&gt; Warning: since v0.4.2 the default r2 value for clumping has changed from 0.01 to 0.001</span></div>
<div class="sourceLine" id="cb26-6" data-line-number="6"><span class="co">#&gt; Clumping TrGPJg, 30 variants</span></div>
<div class="sourceLine" id="cb26-7" data-line-number="7"><span class="co">#&gt; Removing 3 of 30 variants due to LD with other variants or absence from LD reference panel</span></div></code></pre>
<p>The <code>clump_data</code> command takes any data frame that has been formatted to be an exposure data type of data frame. Note that for the instruments in the R/MRInstruments package the SNPs are already LD clumped.</p>
<hr>
</div>
</div>
<div id="outcome-data" class="section level1">
<h1 class="hasAnchor">
<a href="#outcome-data" class="anchor"></a>Outcome data</h1>
<p>Once instruments for the exposure trait have been specified, those SNPs need to be extracted from the outcome trait.</p>
<div id="available-studies-in-mr-base" class="section level2">
<h2 class="hasAnchor">
<a href="#available-studies-in-mr-base" class="anchor"></a>Available studies in MR Base</h2>
<p>MR Base contains complete GWAS summary statistics from a large number of studies. To obtain details about the available GWASs do the following:</p>
<pre class="sourceCode r" id="cb27"><code class="sourceCode r"><div class="sourceLine" id="cb27-1" data-line-number="1">ao &lt;-<span class="st"> </span><span class="kw"><a href="../reference/available_outcomes.html">available_outcomes</a></span>()</div>
<div class="sourceLine" id="cb27-2" data-line-number="2"><span class="co">#&gt; Warning: 'available_outcomes' is deprecated.</span></div>
<div class="sourceLine" id="cb27-3" data-line-number="3"><span class="co">#&gt; Use 'ieugwasr::gwasinfo()' instead.</span></div>
<div class="sourceLine" id="cb27-4" data-line-number="4"><span class="co">#&gt; See help("Deprecated")</span></div>
<div class="sourceLine" id="cb27-5" data-line-number="5">
<span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(ao)</div>
<div class="sourceLine" id="cb27-6" data-line-number="6"><span class="co">#&gt; # A tibble: 6 x 19</span></div>
<div class="sourceLine" id="cb27-7" data-line-number="7"><span class="co">#&gt;   id    trait sample_size   nsnp  year consortium author note  ncase    mr</span></div>
<div class="sourceLine" id="cb27-8" data-line-number="8"><span class="co">#&gt;   &lt;chr&gt; &lt;chr&gt;       &lt;int&gt;  &lt;int&gt; &lt;int&gt; &lt;chr&gt;      &lt;chr&gt;  &lt;chr&gt; &lt;int&gt; &lt;int&gt;</span></div>
<div class="sourceLine" id="cb27-9" data-line-number="9"><span class="co">#&gt; 1 UKB-… Diag…      463010 9.85e6  2018 MRC-IEU    Ben E… 4120…  1870     1</span></div>
<div class="sourceLine" id="cb27-10" data-line-number="10"><span class="co">#&gt; 2 UKB-… Diag…      463010 9.85e6  2018 MRC-IEU    Ben E… 4120…    15     1</span></div>
<div class="sourceLine" id="cb27-11" data-line-number="11"><span class="co">#&gt; 3 UKB-… Diag…      463010 9.85e6  2018 MRC-IEU    Ben E… 4120…   445     1</span></div>
<div class="sourceLine" id="cb27-12" data-line-number="12"><span class="co">#&gt; 4 UKB-… Diag…      463010 9.85e6  2018 MRC-IEU    Ben E… 4120…    28     1</span></div>
<div class="sourceLine" id="cb27-13" data-line-number="13"><span class="co">#&gt; 5 UKB-… Diag…      463010 9.85e6  2018 MRC-IEU    Ben E… 4120…    15     1</span></div>
<div class="sourceLine" id="cb27-14" data-line-number="14"><span class="co">#&gt; 6 UKB-… Diag…      463010 9.85e6  2018 MRC-IEU    Ben E… 4120…  1380     1</span></div>
<div class="sourceLine" id="cb27-15" data-line-number="15"><span class="co">#&gt; # … with 9 more variables: sex &lt;chr&gt;, pmid &lt;int&gt;, priority &lt;int&gt;,</span></div>
<div class="sourceLine" id="cb27-16" data-line-number="16"><span class="co">#&gt; #   population &lt;chr&gt;, unit &lt;chr&gt;, ncontrol &lt;int&gt;, category &lt;chr&gt;,</span></div>
<div class="sourceLine" id="cb27-17" data-line-number="17"><span class="co">#&gt; #   subcategory &lt;chr&gt;, sd &lt;dbl&gt;</span></div></code></pre>
<p>For information about authentication see the <a href="#Authentication">Authentication</a> section.</p>
<p>The <code>available_outcomes</code> function returns a table of all the available studies in the database. Each study has a unique ID. e.g.</p>
<pre class="sourceCode r" id="cb28"><code class="sourceCode r"><div class="sourceLine" id="cb28-1" data-line-number="1">
<span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/subset.html">subset</a></span>(ao, <span class="dt">select=</span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(trait, id)))</div>
<div class="sourceLine" id="cb28-2" data-line-number="2"><span class="co">#&gt; # A tibble: 6 x 2</span></div>
<div class="sourceLine" id="cb28-3" data-line-number="3"><span class="co">#&gt;   trait                                                          id        </span></div>
<div class="sourceLine" id="cb28-4" data-line-number="4"><span class="co">#&gt;   &lt;chr&gt;                                                          &lt;chr&gt;     </span></div>
<div class="sourceLine" id="cb28-5" data-line-number="5"><span class="co">#&gt; 1 Diagnoses - main ICD10: D12.6 Colon, unspecified               UKB-b:192…</span></div>
<div class="sourceLine" id="cb28-6" data-line-number="6"><span class="co">#&gt; 2 Diagnoses - secondary ICD10: S63.0 Dislocation of wrist        UKB-b:1398</span></div>
<div class="sourceLine" id="cb28-7" data-line-number="7"><span class="co">#&gt; 3 Diagnoses - secondary ICD10: L90.5 Scar conditions and fibros… UKB-b:3692</span></div>
<div class="sourceLine" id="cb28-8" data-line-number="8"><span class="co">#&gt; 4 Diagnoses - secondary ICD10: I11.0 Hypertensive heart disease… UKB-b:138…</span></div>
<div class="sourceLine" id="cb28-9" data-line-number="9"><span class="co">#&gt; 5 Diagnoses - secondary ICD10: L01.0 Impetigo [any organism] [a… UKB-b:2176</span></div>
<div class="sourceLine" id="cb28-10" data-line-number="10"><span class="co">#&gt; 6 Diagnoses - main ICD10: C67.9 Bladder, unspecified             UKB-b:5421</span></div></code></pre>
</div>
<div id="extracting-particular-snps-from-particular-studies" class="section level2">
<h2 class="hasAnchor">
<a href="#extracting-particular-snps-from-particular-studies" class="anchor"></a>Extracting particular SNPs from particular studies</h2>
<p>If we want to perform MR of BMI against coronary heart disease, we need to identify the SNPs that influence the BMI, and then extract those SNPs from a GWAS on coronary heart disease.</p>
<p>We have already extracted the 27 SNP effects for BMI:</p>
<pre class="sourceCode r" id="cb29"><code class="sourceCode r"><div class="sourceLine" id="cb29-1" data-line-number="1">
<span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(bmi_exp_dat)</div>
<div class="sourceLine" id="cb29-2" data-line-number="2"><span class="co">#&gt;          SNP beta.exposure se.exposure effect_allele.exposure</span></div>
<div class="sourceLine" id="cb29-3" data-line-number="3"><span class="co">#&gt; 1 rs10767664          0.19  0.03061224                      A</span></div>
<div class="sourceLine" id="cb29-4" data-line-number="4"><span class="co">#&gt; 2 rs13078807          0.10  0.02040816                      G</span></div>
<div class="sourceLine" id="cb29-5" data-line-number="5"><span class="co">#&gt; 4  rs1558902          0.39  0.02040816                      A</span></div>
<div class="sourceLine" id="cb29-6" data-line-number="6"><span class="co">#&gt; 5 rs10968576          0.11  0.02040816                      G</span></div>
<div class="sourceLine" id="cb29-7" data-line-number="7"><span class="co">#&gt; 6  rs2241423          0.13  0.02040816                      G</span></div>
<div class="sourceLine" id="cb29-8" data-line-number="8"><span class="co">#&gt; 7  rs3817334          0.06  0.02040816                      T</span></div>
<div class="sourceLine" id="cb29-9" data-line-number="9"><span class="co">#&gt;   other_allele.exposure eaf.exposure pval.exposure units.exposure</span></div>
<div class="sourceLine" id="cb29-10" data-line-number="10"><span class="co">#&gt; 1                     T         0.78         5e-26          kg/m2</span></div>
<div class="sourceLine" id="cb29-11" data-line-number="11"><span class="co">#&gt; 2                     A         0.20         4e-11          kg/m2</span></div>
<div class="sourceLine" id="cb29-12" data-line-number="12"><span class="co">#&gt; 4                     T         0.42        5e-120          kg/m2</span></div>
<div class="sourceLine" id="cb29-13" data-line-number="13"><span class="co">#&gt; 5                     A         0.31         3e-13          kg/m2</span></div>
<div class="sourceLine" id="cb29-14" data-line-number="14"><span class="co">#&gt; 6                     A         0.78         1e-18          kg/m2</span></div>
<div class="sourceLine" id="cb29-15" data-line-number="15"><span class="co">#&gt; 7                     C         0.41         2e-12          kg/m2</span></div>
<div class="sourceLine" id="cb29-16" data-line-number="16"><span class="co">#&gt;   gene.exposure samplesize.exposure exposure mr_keep.exposure</span></div>
<div class="sourceLine" id="cb29-17" data-line-number="17"><span class="co">#&gt; 1          BDNF              225238      BMI             TRUE</span></div>
<div class="sourceLine" id="cb29-18" data-line-number="18"><span class="co">#&gt; 2         CADM2              221431      BMI             TRUE</span></div>
<div class="sourceLine" id="cb29-19" data-line-number="19"><span class="co">#&gt; 4           FTO              222476      BMI             TRUE</span></div>
<div class="sourceLine" id="cb29-20" data-line-number="20"><span class="co">#&gt; 5        LRRN6C              247166      BMI             TRUE</span></div>
<div class="sourceLine" id="cb29-21" data-line-number="21"><span class="co">#&gt; 6       LBXCOR1              227886      BMI             TRUE</span></div>
<div class="sourceLine" id="cb29-22" data-line-number="22"><span class="co">#&gt; 7        CUGBP1              209051      BMI             TRUE</span></div>
<div class="sourceLine" id="cb29-23" data-line-number="23"><span class="co">#&gt;   pval_origin.exposure units.exposure_dat id.exposure data_source.exposure</span></div>
<div class="sourceLine" id="cb29-24" data-line-number="24"><span class="co">#&gt; 1             reported              kg/m2      TrGPJg             textfile</span></div>
<div class="sourceLine" id="cb29-25" data-line-number="25"><span class="co">#&gt; 2             reported              kg/m2      TrGPJg             textfile</span></div>
<div class="sourceLine" id="cb29-26" data-line-number="26"><span class="co">#&gt; 4             reported              kg/m2      TrGPJg             textfile</span></div>
<div class="sourceLine" id="cb29-27" data-line-number="27"><span class="co">#&gt; 5             reported              kg/m2      TrGPJg             textfile</span></div>
<div class="sourceLine" id="cb29-28" data-line-number="28"><span class="co">#&gt; 6             reported              kg/m2      TrGPJg             textfile</span></div>
<div class="sourceLine" id="cb29-29" data-line-number="29"><span class="co">#&gt; 7             reported              kg/m2      TrGPJg             textfile</span></div></code></pre>
<p>We now need to find a suitable GWAS for coronary heart disease. We can search the available studies:</p>
<pre class="sourceCode r" id="cb30"><code class="sourceCode r"><div class="sourceLine" id="cb30-1" data-line-number="1">ao[<span class="kw"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span>(<span class="st">"heart disease"</span>, ao<span class="op">$</span>trait), ]</div>
<div class="sourceLine" id="cb30-2" data-line-number="2"><span class="co">#&gt; # A tibble: 29 x 19</span></div>
<div class="sourceLine" id="cb30-3" data-line-number="3"><span class="co">#&gt;    id    trait sample_size   nsnp  year consortium author note  ncase    mr</span></div>
<div class="sourceLine" id="cb30-4" data-line-number="4"><span class="co">#&gt;    &lt;chr&gt; &lt;chr&gt;       &lt;int&gt;  &lt;int&gt; &lt;int&gt; &lt;chr&gt;      &lt;chr&gt;  &lt;chr&gt; &lt;int&gt; &lt;int&gt;</span></div>
<div class="sourceLine" id="cb30-5" data-line-number="5"><span class="co">#&gt;  1 UKB-… Diag…      463010 9.85e6  2018 MRC-IEU    Ben E… 4120…    28     1</span></div>
<div class="sourceLine" id="cb30-6" data-line-number="6"><span class="co">#&gt;  2 7     Coro…      184305 9.46e6  2015 CARDIoGRA… Nikpay &lt;NA&gt;  60801     1</span></div>
<div class="sourceLine" id="cb30-7" data-line-number="7"><span class="co">#&gt;  3 UKB-… Diag…      463010 9.85e6  2018 MRC-IEU    Ben E… 4120…    63     1</span></div>
<div class="sourceLine" id="cb30-8" data-line-number="8"><span class="co">#&gt;  4 UKB-… Diag…      337199 1.09e7  2017 Neale Lab  Neale  Data…  8755     1</span></div>
<div class="sourceLine" id="cb30-9" data-line-number="9"><span class="co">#&gt;  5 UKB-… Diag…      463010 9.85e6  2018 MRC-IEU    Ben E… 4120…    20     1</span></div>
<div class="sourceLine" id="cb30-10" data-line-number="10"><span class="co">#&gt;  6 UKB-… Diag…      463010 9.85e6  2018 MRC-IEU    Ben E… 4120…    28     1</span></div>
<div class="sourceLine" id="cb30-11" data-line-number="11"><span class="co">#&gt;  7 UKB-… Diag…      463010 9.85e6  2018 MRC-IEU    Ben E… 4120…  5738     1</span></div>
<div class="sourceLine" id="cb30-12" data-line-number="12"><span class="co">#&gt;  8 UKB-… Diag…      463010 9.85e6  2018 MRC-IEU    Ben E… 4120…  1195     1</span></div>
<div class="sourceLine" id="cb30-13" data-line-number="13"><span class="co">#&gt;  9 UKB-… Diag…      463010 9.85e6  2018 MRC-IEU    Ben E… 4120…   323     1</span></div>
<div class="sourceLine" id="cb30-14" data-line-number="14"><span class="co">#&gt; 10 UKB-… Diag…      463010 9.85e6  2018 MRC-IEU    Ben E… 4120…   269     1</span></div>
<div class="sourceLine" id="cb30-15" data-line-number="15"><span class="co">#&gt; # … with 19 more rows, and 9 more variables: sex &lt;chr&gt;, pmid &lt;int&gt;,</span></div>
<div class="sourceLine" id="cb30-16" data-line-number="16"><span class="co">#&gt; #   priority &lt;int&gt;, population &lt;chr&gt;, unit &lt;chr&gt;, ncontrol &lt;int&gt;,</span></div>
<div class="sourceLine" id="cb30-17" data-line-number="17"><span class="co">#&gt; #   category &lt;chr&gt;, subcategory &lt;chr&gt;, sd &lt;dbl&gt;</span></div></code></pre>
<p>The most recent CARDIOGRAM GWAS is ID number 7. We can extract the BMI SNPs from this GWAS as follows:</p>
<pre class="sourceCode r" id="cb31"><code class="sourceCode r"><div class="sourceLine" id="cb31-1" data-line-number="1">chd_out_dat &lt;-<span class="st"> </span><span class="kw"><a href="../reference/extract_outcome_data.html">extract_outcome_data</a></span>(</div>
<div class="sourceLine" id="cb31-2" data-line-number="2">    <span class="dt">snps =</span> bmi_exp_dat<span class="op">$</span>SNP,</div>
<div class="sourceLine" id="cb31-3" data-line-number="3">    <span class="dt">outcomes =</span> <span class="dv">7</span>
</div>
<div class="sourceLine" id="cb31-4" data-line-number="4">)</div>
<div class="sourceLine" id="cb31-5" data-line-number="5"><span class="co">#&gt; Warning: 'extract_outcome_data' is deprecated.</span></div>
<div class="sourceLine" id="cb31-6" data-line-number="6"><span class="co">#&gt; Use 'ieugwasr::associations()' instead.</span></div>
<div class="sourceLine" id="cb31-7" data-line-number="7"><span class="co">#&gt; See help("Deprecated")</span></div>
<div class="sourceLine" id="cb31-8" data-line-number="8"><span class="co">#&gt; Extracting data for 27 SNP(s) from 1 GWAS(s)</span></div></code></pre>
<p>The <code>extract_outcome_data</code> is a flexible function. The <code>snps</code> argument only requires an array of rsIDs, and the <code>outcomes</code> variable can be a vector of outcomes. e.g. <code>chd_out_dat(c("rs234", "rs17097147"), c(2, 7))</code> will extract the two SNPs from each of the outcomes 2 and 7.</p>
</div>
<div id="ld-proxies" class="section level2">
<h2 class="hasAnchor">
<a href="#ld-proxies" class="anchor"></a>LD proxies</h2>
<p>By default if a particular requested SNP is not present in the outcome GWAS then a SNP (proxy) that is in LD with the requested SNP (target) will be searched for instead. LD proxies are defined using 1000 genomes European sample data. The effect of the proxy SNP on the outcome is returned, along with the proxy SNP, the effect allele of the proxy SNP, and the corresponding allele (in phase) for the target SNP.</p>
<p>The parameters for handling LD proxies are as follows:</p>
<ul>
<li>
<code>proxies</code> = TRUE or FALSE (TRUE by default)</li>
<li>
<code>rsq</code> = numeric value of minimum rsq to find a proxy. Default is 0.8, minimum is 0.6</li>
<li>
<code>palindromes</code> = Allow palindromic SNPs? Default is 1 (yes)</li>
<li>
<code>maf_threshold</code> = If palindromes allowed then what is the maximum minor allele frequency of palindromes allowed? Default is 0.3.</li>
</ul>
</div>
<div id="using-local-gwas-summary-data" class="section level2">
<h2 class="hasAnchor">
<a href="#using-local-gwas-summary-data" class="anchor"></a>Using local GWAS summary data</h2>
<p>If you have GWAS summary data that is not present in MR Base, this can still be to perform analysis, though the LD proxy functionality is not currently available.</p>
<p>Supposing there is a GWAS summary file called “gwas_summary.csv” with e.g. 2 million rows and it looks like this:</p>
<pre><code>rsid,effect,SE,a1,a2,a1_freq,p-value,Units,Gene,n
rs10767664,0.19,0.030612245,A,T,0.78,5.00E-26,kg/m2,BDNF,225238
rs13078807,0.1,0.020408163,G,A,0.2,4.00E-11,kg/m2,CADM2,221431
rs1514175,0.07,0.020408163,A,G,0.43,8.00E-14,kg/m2,TNNI3K,207641
rs1558902,0.39,0.020408163,A,T,0.42,5.00E-120,kg/m2,FTO,222476
...
...</code></pre>
<p>To extract the exposure SNPs from this data, we would use the following command:</p>
<pre class="sourceCode r" id="cb33"><code class="sourceCode r"><div class="sourceLine" id="cb33-1" data-line-number="1">outcome_dat &lt;-<span class="st"> </span><span class="kw"><a href="../reference/read_outcome_data.html">read_outcome_data</a></span>(</div>
<div class="sourceLine" id="cb33-2" data-line-number="2">    <span class="dt">snps =</span> bmi_exp_dat<span class="op">$</span>SNP,</div>
<div class="sourceLine" id="cb33-3" data-line-number="3">    <span class="dt">filename =</span> <span class="st">"gwas_summary.csv"</span>,</div>
<div class="sourceLine" id="cb33-4" data-line-number="4">    <span class="dt">sep =</span> <span class="st">","</span>,</div>
<div class="sourceLine" id="cb33-5" data-line-number="5">    <span class="dt">snp_col =</span> <span class="st">"rsid"</span>,</div>
<div class="sourceLine" id="cb33-6" data-line-number="6">    <span class="dt">beta_col =</span> <span class="st">"effect"</span>,</div>
<div class="sourceLine" id="cb33-7" data-line-number="7">    <span class="dt">se_col =</span> <span class="st">"SE"</span>,</div>
<div class="sourceLine" id="cb33-8" data-line-number="8">    <span class="dt">effect_allele_col =</span> <span class="st">"a1"</span>,</div>
<div class="sourceLine" id="cb33-9" data-line-number="9">    <span class="dt">other_allele_col =</span> <span class="st">"a2"</span>,</div>
<div class="sourceLine" id="cb33-10" data-line-number="10">    <span class="dt">eaf_col =</span> <span class="st">"a1_freq"</span>,</div>
<div class="sourceLine" id="cb33-11" data-line-number="11">    <span class="dt">pval_col =</span> <span class="st">"p-value"</span>,</div>
<div class="sourceLine" id="cb33-12" data-line-number="12">    <span class="dt">units_col =</span> <span class="st">"Units"</span>,</div>
<div class="sourceLine" id="cb33-13" data-line-number="13">    <span class="dt">gene_col =</span> <span class="st">"Gene"</span>,</div>
<div class="sourceLine" id="cb33-14" data-line-number="14">    <span class="dt">samplesize_col =</span> <span class="st">"n"</span>
</div>
<div class="sourceLine" id="cb33-15" data-line-number="15">)</div></code></pre>
<p>This returns an outcome data frame with only the SNPs that were requested (if those SNPs were present in the “gwas_summary.csv” file).</p>
</div>
<div id="outcome-data-format" class="section level2">
<h2 class="hasAnchor">
<a href="#outcome-data-format" class="anchor"></a>Outcome data format</h2>
<p>The <code>extract_outcome_data</code> function returns a table of SNP effects for the requested SNPs on the requested outcomes. The format of the data is similar to the exposure data format, except the main columns are as follows:</p>
<ul>
<li><code>SNP</code></li>
<li><code>beta.outcome</code></li>
<li><code>se.outcome</code></li>
<li><code>samplesize.outcome</code></li>
<li><code>ncase.outcome</code></li>
<li><code>ncontrol.outcome</code></li>
<li><code>pval.outcome</code></li>
<li><code>eaf.outcome</code></li>
<li><code>effect_allele.outcom</code></li>
<li><code>other_allele.outcome</code></li>
<li><code>units.outcome</code></li>
<li><code>outcome</code></li>
<li><code>consortium.outcome</code></li>
<li><code>year.outcome</code></li>
<li><code>pmid.outcome</code></li>
<li><code>id.outcome</code></li>
<li><code>originalname.outcome</code></li>
<li><code>proxy.outcome</code></li>
<li><code>target_snp.outcome</code></li>
<li><code>proxy_snp.outcome</code></li>
<li><code>target_a1.outcome</code></li>
<li><code>target_a2.outcome</code></li>
<li><code>proxy_a1.outcome</code></li>
<li><code>proxy_a2.outcome</code></li>
<li><code>mr_keep.outcome</code></li>
<li><code>data_source.outcome</code></li>
</ul>
<hr>
</div>
</div>
<div id="harmonise-data" class="section level1">
<h1 class="hasAnchor">
<a href="#harmonise-data" class="anchor"></a>Harmonise data</h1>
<p>The exposure data and outcome data are now obtained, but it is important to harmonise the effects. This means that the effect of a SNP on the exposure and the effect of that SNP on the outcome must each correspond to the same allele.</p>
<p>To harmonise the exposure and outcome data, do the following:</p>
<pre class="sourceCode r" id="cb34"><code class="sourceCode r"><div class="sourceLine" id="cb34-1" data-line-number="1">dat &lt;-<span class="st"> </span><span class="kw"><a href="../reference/harmonise_data.html">harmonise_data</a></span>(</div>
<div class="sourceLine" id="cb34-2" data-line-number="2">    <span class="dt">exposure_dat =</span> bmi_exp_dat, </div>
<div class="sourceLine" id="cb34-3" data-line-number="3">    <span class="dt">outcome_dat =</span> chd_out_dat</div>
<div class="sourceLine" id="cb34-4" data-line-number="4">)</div>
<div class="sourceLine" id="cb34-5" data-line-number="5"><span class="co">#&gt; Harmonising BMI (TrGPJg) and Coronary heart disease || id:7 (7)</span></div></code></pre>
<p>This creates a new data frame that has the exposure data and outcome data combined.</p>
<p>If there were 3 exposure traits and 3 outcome traits then there will be 9 sets of harmonisations being performed - harmonising the SNP effects of exposure trait 1 against outcome trait 1; exposure trait 1 against outcome trait 2; and so on.</p>
<div id="dealing-with-strand-issues" class="section level2">
<h2 class="hasAnchor">
<a href="#dealing-with-strand-issues" class="anchor"></a>Dealing with strand issues</h2>
<p>Recent GWASs typically present the effects of a SNP in reference to the allele on the forward strand. But as reference panels are updated the forward strand sometimes changes, and GWASs from a few years ago aren’t guaranteed to be using forward strand conventions.</p>
<p>Some examples are shown below:</p>
<div id="correct-unambigious" class="section level3">
<h3 class="hasAnchor">
<a href="#correct-unambigious" class="anchor"></a>Correct, unambigious</h3>
<pre><code>exposure effect = 0.5
effect allele = A
other allele = G

outcome effect = 0.05
effect allele = A
other allele = G</code></pre>
<p>Here the effect allele on the exposure and the outcome is the same</p>
</div>
<div id="incorrect-reference-unambigious" class="section level3">
<h3 class="hasAnchor">
<a href="#incorrect-reference-unambigious" class="anchor"></a>Incorrect reference, unambigious</h3>
<pre><code>exposure effect = 0.5
effect allele = A
other allele = G

outcome effect = -0.05
effect allele = C
other allele = T</code></pre>
<p>Here the outcome GWAS is presenting the effect for the alternate allele on the reverse strand. We need to flip the outcome effect to 0.05 to correspond to the same allele as the exposure GWAS on the forward strand.</p>
</div>
<div id="ambiguous" class="section level3">
<h3 class="hasAnchor">
<a href="#ambiguous" class="anchor"></a>Ambiguous</h3>
<pre><code>exposure effect = 0.5
effect allele = A
other allele = G

outcome effect = -0.05
effect allele = A
other allele = C</code></pre>
<p>Here the alleles do not correspond for the same SNP, so this SNP will be discarded from the analysis.</p>
</div>
<div id="palindromic-snp-inferrable" class="section level3">
<h3 class="hasAnchor">
<a href="#palindromic-snp-inferrable" class="anchor"></a>Palindromic SNP, inferrable</h3>
<pre><code>exposure effect = 0.5
effect allele = A
other allele = T
effect allele frequency = 0.11

outcome effect = -0.05
effect allele = A
other allele = T
effect allele frequency = 0.91</code></pre>
<p>Here the alleles correspond, but it is a palindromic SNP, such that the alleles on the forward strand are the same as on the reverse strand (A/T on forward is T/A on the reverse). However, the allele frequency of the effect allele gives us information - if the outcome effect allele (A) were on the forward strand we would expect it to have a low allele frequency, but given it has a high frequency (0.91) we infer that the outcome GWAS is presenting the effect on the reverse strand for the alternative allele. We would flip the effect to 0.05 for the outcome GWAS.</p>
</div>
<div id="palindromic-snp-not-inferrable" class="section level3">
<h3 class="hasAnchor">
<a href="#palindromic-snp-not-inferrable" class="anchor"></a>Palindromic SNP, not inferrable</h3>
<pre><code>exposure effect = 0.5
effect allele = A
other allele = T
effect allele frequency = 0.50

outcome effect = -0.05
effect allele = A
other allele = T
effect allele frequency = 0.50</code></pre>
<p>This is similar to the above, except the allele frequency no longer gives us information about the strand. We would discard this SNP. This is done for any palindromic SNPs that have minor allele frequency above 0.42.</p>
</div>
<div id="options" class="section level3">
<h3 class="hasAnchor">
<a href="#options" class="anchor"></a>Options</h3>
<p>There are three options to harmonising the data.</p>
<ol style="list-style-type: decimal">
<li>Assume all alleles are presented on the forward strand</li>
<li>Try to infer the forward strand alleles using allele frequency information</li>
<li>Correct the strand for non-palindromic SNPs, but drop all palindromic SNPs</li>
</ol>
<p>By default, the <code>harmonise_data</code> function uses option 2, but this can be modified using the <code>action</code> argument, e.g. <code><a href="../reference/harmonise_data.html">harmonise_data(exposure_dat, outcome_dat, action=3)</a></code>.</p>
</div>
</div>
<div id="drop-duplicate-exposure-outcome-summary-sets" class="section level2">
<h2 class="hasAnchor">
<a href="#drop-duplicate-exposure-outcome-summary-sets" class="anchor"></a>Drop duplicate exposure-outcome summary sets</h2>
<p>After data harmonisation, users may find that their dataset contains duplicate exposure-outcome summary sets. This can arise, for example, when a GWAS consortium has released multiple results from separate GWAS analyses for the same trait. For example, there are multiple GWAS summary datasets for body mass index and coronary heart disease:</p>
<pre class="sourceCode r" id="cb40"><code class="sourceCode r"><div class="sourceLine" id="cb40-1" data-line-number="1">ao[ao<span class="op">$</span>trait<span class="op">==</span><span class="st">"Body mass index"</span>,<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"trait"</span>,<span class="st">"id"</span>,<span class="st">"pmid"</span>,<span class="st">"author"</span>,<span class="st">"sample_size"</span>,<span class="st">"nsnp"</span>)]</div>
<div class="sourceLine" id="cb40-2" data-line-number="2"><span class="co">#&gt; # A tibble: 8 x 6</span></div>
<div class="sourceLine" id="cb40-3" data-line-number="3"><span class="co">#&gt;   trait           id        pmid author       sample_size    nsnp</span></div>
<div class="sourceLine" id="cb40-4" data-line-number="4"><span class="co">#&gt;   &lt;chr&gt;           &lt;chr&gt;    &lt;int&gt; &lt;chr&gt;              &lt;int&gt;   &lt;int&gt;</span></div>
<div class="sourceLine" id="cb40-5" data-line-number="5"><span class="co">#&gt; 1 Body mass index 1089  26961502 Wood              120286 8654252</span></div>
<div class="sourceLine" id="cb40-6" data-line-number="6"><span class="co">#&gt; 2 Body mass index 95    23754948 Randall JC         73137 2736876</span></div>
<div class="sourceLine" id="cb40-7" data-line-number="7"><span class="co">#&gt; 3 Body mass index 2     25673413 Locke AE          339224 2555511</span></div>
<div class="sourceLine" id="cb40-8" data-line-number="8"><span class="co">#&gt; 4 Body mass index 84    20935630 Speliotes EK      123865 2471517</span></div>
<div class="sourceLine" id="cb40-9" data-line-number="9"><span class="co">#&gt; 5 Body mass index 835   25673413 Locke AE          322154 2554668</span></div>
<div class="sourceLine" id="cb40-10" data-line-number="10"><span class="co">#&gt; 6 Body mass index 974   25673413 Locke AE          171977 2494613</span></div>
<div class="sourceLine" id="cb40-11" data-line-number="11"><span class="co">#&gt; 7 Body mass index 785   25673413 Locke AE          152893 2477659</span></div>
<div class="sourceLine" id="cb40-12" data-line-number="12"><span class="co">#&gt; 8 Body mass index 94    23754948 Randall JC         60586 2736876</span></div>
<div class="sourceLine" id="cb40-13" data-line-number="13">ao[ao<span class="op">$</span>trait<span class="op">==</span><span class="st">"Coronary heart disease"</span>,<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"trait"</span>,<span class="st">"id"</span>,<span class="st">"pmid"</span>,<span class="st">"author"</span>,<span class="st">"ncase"</span>,<span class="st">"ncontrol"</span>,<span class="st">"nsnp"</span>)]</div>
<div class="sourceLine" id="cb40-14" data-line-number="14"><span class="co">#&gt; # A tibble: 4 x 7</span></div>
<div class="sourceLine" id="cb40-15" data-line-number="15"><span class="co">#&gt;   trait                  id        pmid author      ncase ncontrol    nsnp</span></div>
<div class="sourceLine" id="cb40-16" data-line-number="16"><span class="co">#&gt;   &lt;chr&gt;                  &lt;chr&gt;    &lt;int&gt; &lt;chr&gt;       &lt;int&gt;    &lt;int&gt;   &lt;int&gt;</span></div>
<div class="sourceLine" id="cb40-17" data-line-number="17"><span class="co">#&gt; 1 Coronary heart disease 7     26343387 Nikpay      60801   123504 9455779</span></div>
<div class="sourceLine" id="cb40-18" data-line-number="18"><span class="co">#&gt; 2 Coronary heart disease 6     21378988 Peden       15420    15062  540233</span></div>
<div class="sourceLine" id="cb40-19" data-line-number="19"><span class="co">#&gt; 3 Coronary heart disease 9     23202125 Deloukas    63746   130681   79129</span></div>
<div class="sourceLine" id="cb40-20" data-line-number="20"><span class="co">#&gt; 4 Coronary heart disease 8     21378990 Schunkert H 22233    64762 2420361</span></div></code></pre>
<p>There are therefore multiple potential combinations of body mass index and coronary heart disease, which would likely lead to duplicate MR analyses. We recommend that users prune their datasets so that only the exposure-outcome combination with the highested expected power is retained. This can be done by selecting the exposure-outcome summary set with the largest sample size for the outcome, using the power_prune function:</p>
<pre class="sourceCode r" id="cb41"><code class="sourceCode r"><div class="sourceLine" id="cb41-1" data-line-number="1">dat&lt;-<span class="kw"><a href="../reference/power_prune.html">power_prune</a></span>(dat,<span class="dt">method=</span><span class="dv">1</span>,<span class="dt">dist.outcome=</span><span class="st">"binary"</span>)</div></code></pre>
<p>This drops the duplicate exposure-outcome sets with the smaller outcome sample size (number of cases for binary outcomes). Remaining duplicates are then dropped on the basis of the exposure sample size. However, if there are a large number of SNPs available to instrument an exposure, the outcome GWAS with the better SNP coverage may provide better power than the outcome GWAS with the larger sample size. This can occur, for example, if the larger outcome GWAS has used a targeted genotyping array. In such instances, it may be better to prune studies on the basis of instrument strength (i.e. variation in exposure explained by the instrumental SNPs) as well as sample size. This can be done by setting the method argument to 2:</p>
<pre class="sourceCode r" id="cb42"><code class="sourceCode r"><div class="sourceLine" id="cb42-1" data-line-number="1">dat&lt;-<span class="kw"><a href="../reference/power_prune.html">power_prune</a></span>(dat,<span class="dt">method=</span><span class="dv">2</span>,<span class="dt">dist.outcome=</span><span class="st">"binary"</span>)</div></code></pre>
<p>This procedure drops duplicate exposure-outcome sets on the basis of instrument strength and sample size, and assumes that the SNP-exposure effects correspond to a continuous trait with a normal distribution (i.e. exposure should not be binary). The SNP-outcome effects can correspond to either a binary or continuous trait (default behaviour is to assume a binary distribution). If the exposure is binary then method 1 should be used.</p>
<hr>
</div>
</div>
<div id="perform-mr" class="section level1">
<h1 class="hasAnchor">
<a href="#perform-mr" class="anchor"></a>Perform MR</h1>
<p>Once the exposure and outcome data are harmonised, we have effects and standard errors for each instrument SNP available for the exposure and outcome traits. We can use this information to perform Mendelian randomisation. To do this, simply run:</p>
<pre class="sourceCode r" id="cb43"><code class="sourceCode r"><div class="sourceLine" id="cb43-1" data-line-number="1">res &lt;-<span class="st"> </span><span class="kw"><a href="../reference/mr.html">mr</a></span>(dat)</div>
<div class="sourceLine" id="cb43-2" data-line-number="2"><span class="co">#&gt; Analysing 'TrGPJg' on '7'</span></div>
<div class="sourceLine" id="cb43-3" data-line-number="3">res</div>
<div class="sourceLine" id="cb43-4" data-line-number="4"><span class="co">#&gt;   id.exposure id.outcome                        outcome exposure</span></div>
<div class="sourceLine" id="cb43-5" data-line-number="5"><span class="co">#&gt; 1      TrGPJg          7 Coronary heart disease || id:7      BMI</span></div>
<div class="sourceLine" id="cb43-6" data-line-number="6"><span class="co">#&gt; 2      TrGPJg          7 Coronary heart disease || id:7      BMI</span></div>
<div class="sourceLine" id="cb43-7" data-line-number="7"><span class="co">#&gt; 3      TrGPJg          7 Coronary heart disease || id:7      BMI</span></div>
<div class="sourceLine" id="cb43-8" data-line-number="8"><span class="co">#&gt; 4      TrGPJg          7 Coronary heart disease || id:7      BMI</span></div>
<div class="sourceLine" id="cb43-9" data-line-number="9"><span class="co">#&gt; 5      TrGPJg          7 Coronary heart disease || id:7      BMI</span></div>
<div class="sourceLine" id="cb43-10" data-line-number="10"><span class="co">#&gt;                      method nsnp          b         se         pval</span></div>
<div class="sourceLine" id="cb43-11" data-line-number="11"><span class="co">#&gt; 1                  MR Egger   27 0.11385662 0.03292752 1.962323e-03</span></div>
<div class="sourceLine" id="cb43-12" data-line-number="12"><span class="co">#&gt; 2           Weighted median   27 0.07940417 0.02098263 1.541531e-04</span></div>
<div class="sourceLine" id="cb43-13" data-line-number="13"><span class="co">#&gt; 3 Inverse variance weighted   27 0.11391684 0.01611170 1.544411e-12</span></div>
<div class="sourceLine" id="cb43-14" data-line-number="14"><span class="co">#&gt; 4               Simple mode   27 0.07510331 0.03878766 6.377628e-02</span></div>
<div class="sourceLine" id="cb43-15" data-line-number="15"><span class="co">#&gt; 5             Weighted mode   27 0.08875377 0.02209427 4.471698e-04</span></div></code></pre>
<p>This returns a data frame of estimates of the causal effect of the exposure on the outcome for a range of different MR methods.</p>
<p>If there were multiple exposures against multiple outcomes in <code>dat</code>, the <code><a href="../reference/mr.html">mr()</a></code> function will perform each MR method for each combination of exposure-outcome traits.</p>
<div id="mr-methods" class="section level2">
<h2 class="hasAnchor">
<a href="#mr-methods" class="anchor"></a>MR methods</h2>
<p>The list of available MR methods can be obtained:</p>
<pre class="sourceCode r" id="cb44"><code class="sourceCode r"><div class="sourceLine" id="cb44-1" data-line-number="1">
<span class="kw"><a href="../reference/mr_method_list.html">mr_method_list</a></span>()</div>
<div class="sourceLine" id="cb44-2" data-line-number="2"><span class="co">#&gt;                              obj</span></div>
<div class="sourceLine" id="cb44-3" data-line-number="3"><span class="co">#&gt; 1                  mr_wald_ratio</span></div>
<div class="sourceLine" id="cb44-4" data-line-number="4"><span class="co">#&gt; 2               mr_two_sample_ml</span></div>
<div class="sourceLine" id="cb44-5" data-line-number="5"><span class="co">#&gt; 3            mr_egger_regression</span></div>
<div class="sourceLine" id="cb44-6" data-line-number="6"><span class="co">#&gt; 4  mr_egger_regression_bootstrap</span></div>
<div class="sourceLine" id="cb44-7" data-line-number="7"><span class="co">#&gt; 5               mr_simple_median</span></div>
<div class="sourceLine" id="cb44-8" data-line-number="8"><span class="co">#&gt; 6             mr_weighted_median</span></div>
<div class="sourceLine" id="cb44-9" data-line-number="9"><span class="co">#&gt; 7   mr_penalised_weighted_median</span></div>
<div class="sourceLine" id="cb44-10" data-line-number="10"><span class="co">#&gt; 8                         mr_ivw</span></div>
<div class="sourceLine" id="cb44-11" data-line-number="11"><span class="co">#&gt; 9                  mr_ivw_radial</span></div>
<div class="sourceLine" id="cb44-12" data-line-number="12"><span class="co">#&gt; 10                    mr_ivw_mre</span></div>
<div class="sourceLine" id="cb44-13" data-line-number="13"><span class="co">#&gt; 11                     mr_ivw_fe</span></div>
<div class="sourceLine" id="cb44-14" data-line-number="14"><span class="co">#&gt; 12                mr_simple_mode</span></div>
<div class="sourceLine" id="cb44-15" data-line-number="15"><span class="co">#&gt; 13              mr_weighted_mode</span></div>
<div class="sourceLine" id="cb44-16" data-line-number="16"><span class="co">#&gt; 14         mr_weighted_mode_nome</span></div>
<div class="sourceLine" id="cb44-17" data-line-number="17"><span class="co">#&gt; 15           mr_simple_mode_nome</span></div>
<div class="sourceLine" id="cb44-18" data-line-number="18"><span class="co">#&gt; 16                       mr_raps</span></div>
<div class="sourceLine" id="cb44-19" data-line-number="19"><span class="co">#&gt; 17                       mr_sign</span></div>
<div class="sourceLine" id="cb44-20" data-line-number="20"><span class="co">#&gt; 18                        mr_uwr</span></div>
<div class="sourceLine" id="cb44-21" data-line-number="21"><span class="co">#&gt;                                                         name PubmedID</span></div>
<div class="sourceLine" id="cb44-22" data-line-number="22"><span class="co">#&gt; 1                                                 Wald ratio         </span></div>
<div class="sourceLine" id="cb44-23" data-line-number="23"><span class="co">#&gt; 2                                         Maximum likelihood         </span></div>
<div class="sourceLine" id="cb44-24" data-line-number="24"><span class="co">#&gt; 3                                                   MR Egger 26050253</span></div>
<div class="sourceLine" id="cb44-25" data-line-number="25"><span class="co">#&gt; 4                                       MR Egger (bootstrap) 26050253</span></div>
<div class="sourceLine" id="cb44-26" data-line-number="26"><span class="co">#&gt; 5                                              Simple median         </span></div>
<div class="sourceLine" id="cb44-27" data-line-number="27"><span class="co">#&gt; 6                                            Weighted median         </span></div>
<div class="sourceLine" id="cb44-28" data-line-number="28"><span class="co">#&gt; 7                                  Penalised weighted median         </span></div>
<div class="sourceLine" id="cb44-29" data-line-number="29"><span class="co">#&gt; 8                                  Inverse variance weighted         </span></div>
<div class="sourceLine" id="cb44-30" data-line-number="30"><span class="co">#&gt; 9                                                 IVW radial         </span></div>
<div class="sourceLine" id="cb44-31" data-line-number="31"><span class="co">#&gt; 10 Inverse variance weighted (multiplicative random effects)         </span></div>
<div class="sourceLine" id="cb44-32" data-line-number="32"><span class="co">#&gt; 11                 Inverse variance weighted (fixed effects)         </span></div>
<div class="sourceLine" id="cb44-33" data-line-number="33"><span class="co">#&gt; 12                                               Simple mode         </span></div>
<div class="sourceLine" id="cb44-34" data-line-number="34"><span class="co">#&gt; 13                                             Weighted mode         </span></div>
<div class="sourceLine" id="cb44-35" data-line-number="35"><span class="co">#&gt; 14                                      Weighted mode (NOME)         </span></div>
<div class="sourceLine" id="cb44-36" data-line-number="36"><span class="co">#&gt; 15                                        Simple mode (NOME)         </span></div>
<div class="sourceLine" id="cb44-37" data-line-number="37"><span class="co">#&gt; 16                      Robust adjusted profile score (RAPS)         </span></div>
<div class="sourceLine" id="cb44-38" data-line-number="38"><span class="co">#&gt; 17                                     Sign concordance test         </span></div>
<div class="sourceLine" id="cb44-39" data-line-number="39"><span class="co">#&gt; 18                                     Unweighted regression         </span></div>
<div class="sourceLine" id="cb44-40" data-line-number="40"><span class="co">#&gt;                                                    Description</span></div>
<div class="sourceLine" id="cb44-41" data-line-number="41"><span class="co">#&gt; 1                                                             </span></div>
<div class="sourceLine" id="cb44-42" data-line-number="42"><span class="co">#&gt; 2                                                             </span></div>
<div class="sourceLine" id="cb44-43" data-line-number="43"><span class="co">#&gt; 3                                                             </span></div>
<div class="sourceLine" id="cb44-44" data-line-number="44"><span class="co">#&gt; 4                                                             </span></div>
<div class="sourceLine" id="cb44-45" data-line-number="45"><span class="co">#&gt; 5                                                             </span></div>
<div class="sourceLine" id="cb44-46" data-line-number="46"><span class="co">#&gt; 6                                                             </span></div>
<div class="sourceLine" id="cb44-47" data-line-number="47"><span class="co">#&gt; 7                                                             </span></div>
<div class="sourceLine" id="cb44-48" data-line-number="48"><span class="co">#&gt; 8                                                             </span></div>
<div class="sourceLine" id="cb44-49" data-line-number="49"><span class="co">#&gt; 9                                                             </span></div>
<div class="sourceLine" id="cb44-50" data-line-number="50"><span class="co">#&gt; 10                                                            </span></div>
<div class="sourceLine" id="cb44-51" data-line-number="51"><span class="co">#&gt; 11                                                            </span></div>
<div class="sourceLine" id="cb44-52" data-line-number="52"><span class="co">#&gt; 12                                                            </span></div>
<div class="sourceLine" id="cb44-53" data-line-number="53"><span class="co">#&gt; 13                                                            </span></div>
<div class="sourceLine" id="cb44-54" data-line-number="54"><span class="co">#&gt; 14                                                            </span></div>
<div class="sourceLine" id="cb44-55" data-line-number="55"><span class="co">#&gt; 15                                                            </span></div>
<div class="sourceLine" id="cb44-56" data-line-number="56"><span class="co">#&gt; 16                                                            </span></div>
<div class="sourceLine" id="cb44-57" data-line-number="57"><span class="co">#&gt; 17 Tests for concordance of signs between exposure and outcome</span></div>
<div class="sourceLine" id="cb44-58" data-line-number="58"><span class="co">#&gt; 18                                     Doesn't use any weights</span></div>
<div class="sourceLine" id="cb44-59" data-line-number="59"><span class="co">#&gt;    use_by_default heterogeneity_test</span></div>
<div class="sourceLine" id="cb44-60" data-line-number="60"><span class="co">#&gt; 1            TRUE              FALSE</span></div>
<div class="sourceLine" id="cb44-61" data-line-number="61"><span class="co">#&gt; 2           FALSE               TRUE</span></div>
<div class="sourceLine" id="cb44-62" data-line-number="62"><span class="co">#&gt; 3            TRUE               TRUE</span></div>
<div class="sourceLine" id="cb44-63" data-line-number="63"><span class="co">#&gt; 4           FALSE              FALSE</span></div>
<div class="sourceLine" id="cb44-64" data-line-number="64"><span class="co">#&gt; 5           FALSE              FALSE</span></div>
<div class="sourceLine" id="cb44-65" data-line-number="65"><span class="co">#&gt; 6            TRUE              FALSE</span></div>
<div class="sourceLine" id="cb44-66" data-line-number="66"><span class="co">#&gt; 7           FALSE              FALSE</span></div>
<div class="sourceLine" id="cb44-67" data-line-number="67"><span class="co">#&gt; 8            TRUE               TRUE</span></div>
<div class="sourceLine" id="cb44-68" data-line-number="68"><span class="co">#&gt; 9           FALSE               TRUE</span></div>
<div class="sourceLine" id="cb44-69" data-line-number="69"><span class="co">#&gt; 10          FALSE              FALSE</span></div>
<div class="sourceLine" id="cb44-70" data-line-number="70"><span class="co">#&gt; 11          FALSE              FALSE</span></div>
<div class="sourceLine" id="cb44-71" data-line-number="71"><span class="co">#&gt; 12           TRUE              FALSE</span></div>
<div class="sourceLine" id="cb44-72" data-line-number="72"><span class="co">#&gt; 13           TRUE              FALSE</span></div>
<div class="sourceLine" id="cb44-73" data-line-number="73"><span class="co">#&gt; 14          FALSE              FALSE</span></div>
<div class="sourceLine" id="cb44-74" data-line-number="74"><span class="co">#&gt; 15          FALSE              FALSE</span></div>
<div class="sourceLine" id="cb44-75" data-line-number="75"><span class="co">#&gt; 16          FALSE              FALSE</span></div>
<div class="sourceLine" id="cb44-76" data-line-number="76"><span class="co">#&gt; 17          FALSE              FALSE</span></div>
<div class="sourceLine" id="cb44-77" data-line-number="77"><span class="co">#&gt; 18          FALSE               TRUE</span></div></code></pre>
<p>To perform them, they can be specified in the <code><a href="../reference/mr.html">mr()</a></code> function, e.g. to only perform MR Egger regression and Inverse variance weighted methods,</p>
<pre class="sourceCode r" id="cb45"><code class="sourceCode r"><div class="sourceLine" id="cb45-1" data-line-number="1">
<span class="kw"><a href="../reference/mr.html">mr</a></span>(dat, <span class="dt">method_list=</span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"mr_egger_regression"</span>, <span class="st">"mr_ivw"</span>))</div>
<div class="sourceLine" id="cb45-2" data-line-number="2"><span class="co">#&gt; Analysing 'TrGPJg' on '7'</span></div>
<div class="sourceLine" id="cb45-3" data-line-number="3"><span class="co">#&gt;   id.exposure id.outcome                        outcome exposure</span></div>
<div class="sourceLine" id="cb45-4" data-line-number="4"><span class="co">#&gt; 1      TrGPJg          7 Coronary heart disease || id:7      BMI</span></div>
<div class="sourceLine" id="cb45-5" data-line-number="5"><span class="co">#&gt; 2      TrGPJg          7 Coronary heart disease || id:7      BMI</span></div>
<div class="sourceLine" id="cb45-6" data-line-number="6"><span class="co">#&gt;                      method nsnp         b         se         pval</span></div>
<div class="sourceLine" id="cb45-7" data-line-number="7"><span class="co">#&gt; 1                  MR Egger   27 0.1138566 0.03292752 1.962323e-03</span></div>
<div class="sourceLine" id="cb45-8" data-line-number="8"><span class="co">#&gt; 2 Inverse variance weighted   27 0.1139168 0.01611170 1.544411e-12</span></div></code></pre>
<p>By default, all the methods that are labelled <code>TRUE</code> in the <code>use_by_default</code> column are used by the <code><a href="../reference/mr.html">mr()</a></code> function.</p>
<hr>
</div>
<div id="sensitivity-analyses" class="section level2">
<h2 class="hasAnchor">
<a href="#sensitivity-analyses" class="anchor"></a>Sensitivity analyses</h2>
<div id="heterogeneity-statistics" class="section level3">
<h3 class="hasAnchor">
<a href="#heterogeneity-statistics" class="anchor"></a>Heterogeneity statistics</h3>
<p>Some of the MR methods can also perform tests for heterogeneity. To obtain those statistics:</p>
<pre class="sourceCode r" id="cb46"><code class="sourceCode r"><div class="sourceLine" id="cb46-1" data-line-number="1">
<span class="kw"><a href="../reference/mr_heterogeneity.html">mr_heterogeneity</a></span>(dat)</div>
<div class="sourceLine" id="cb46-2" data-line-number="2"><span class="co">#&gt;   id.exposure id.outcome                        outcome exposure</span></div>
<div class="sourceLine" id="cb46-3" data-line-number="3"><span class="co">#&gt; 1      TrGPJg          7 Coronary heart disease || id:7      BMI</span></div>
<div class="sourceLine" id="cb46-4" data-line-number="4"><span class="co">#&gt; 2      TrGPJg          7 Coronary heart disease || id:7      BMI</span></div>
<div class="sourceLine" id="cb46-5" data-line-number="5"><span class="co">#&gt;                      method        Q Q_df     Q_pval</span></div>
<div class="sourceLine" id="cb46-6" data-line-number="6"><span class="co">#&gt; 1                  MR Egger 39.68702   25 0.03139634</span></div>
<div class="sourceLine" id="cb46-7" data-line-number="7"><span class="co">#&gt; 2 Inverse variance weighted 39.68703   26 0.04185786</span></div></code></pre>
<p>As with the <code><a href="../reference/mr.html">mr()</a></code> function, the <code><a href="../reference/mr_heterogeneity.html">mr_heterogeneity()</a></code> function can take an argument to only perform heterogeneity tests using specified methods, e.g.</p>
<pre class="sourceCode r" id="cb47"><code class="sourceCode r"><div class="sourceLine" id="cb47-1" data-line-number="1">
<span class="kw"><a href="../reference/mr_heterogeneity.html">mr_heterogeneity</a></span>(dat, <span class="dt">method_list=</span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"mr_egger_regression"</span>, <span class="st">"mr_ivw"</span>))</div>
<div class="sourceLine" id="cb47-2" data-line-number="2"><span class="co">#&gt;   id.exposure id.outcome                        outcome exposure</span></div>
<div class="sourceLine" id="cb47-3" data-line-number="3"><span class="co">#&gt; 1      TrGPJg          7 Coronary heart disease || id:7      BMI</span></div>
<div class="sourceLine" id="cb47-4" data-line-number="4"><span class="co">#&gt; 2      TrGPJg          7 Coronary heart disease || id:7      BMI</span></div>
<div class="sourceLine" id="cb47-5" data-line-number="5"><span class="co">#&gt;                      method        Q Q_df     Q_pval</span></div>
<div class="sourceLine" id="cb47-6" data-line-number="6"><span class="co">#&gt; 1                  MR Egger 39.68702   25 0.03139634</span></div>
<div class="sourceLine" id="cb47-7" data-line-number="7"><span class="co">#&gt; 2 Inverse variance weighted 39.68703   26 0.04185786</span></div></code></pre>
</div>
<div id="horizontal-pleiotropy" class="section level3">
<h3 class="hasAnchor">
<a href="#horizontal-pleiotropy" class="anchor"></a>Horizontal pleiotropy</h3>
<p>The intercept term in MR Egger regression can be a useful indication of whether directional horizontal pleiotropy is driving the results of an MR analysis. This can be obtained as follows:</p>
<pre class="sourceCode r" id="cb48"><code class="sourceCode r"><div class="sourceLine" id="cb48-1" data-line-number="1">
<span class="kw"><a href="../reference/mr_pleiotropy_test.html">mr_pleiotropy_test</a></span>(dat)</div>
<div class="sourceLine" id="cb48-2" data-line-number="2"><span class="co">#&gt;   id.exposure id.outcome                        outcome exposure</span></div>
<div class="sourceLine" id="cb48-3" data-line-number="3"><span class="co">#&gt; 1      TrGPJg          7 Coronary heart disease || id:7      BMI</span></div>
<div class="sourceLine" id="cb48-4" data-line-number="4"><span class="co">#&gt;   egger_intercept          se     pval</span></div>
<div class="sourceLine" id="cb48-5" data-line-number="5"><span class="co">#&gt; 1    1.110681e-05 0.005263163 0.998333</span></div></code></pre>
</div>
<div id="single-snp-analysis" class="section level3">
<h3 class="hasAnchor">
<a href="#single-snp-analysis" class="anchor"></a>Single SNP analysis</h3>
<p>To obtain the MR estimates using each of the SNPs singly we can do the following:</p>
<pre class="sourceCode r" id="cb49"><code class="sourceCode r"><div class="sourceLine" id="cb49-1" data-line-number="1">res_single &lt;-<span class="st"> </span><span class="kw"><a href="../reference/mr_singlesnp.html">mr_singlesnp</a></span>(dat)</div></code></pre>
<p>This returns a data.frame of results that is similar to the output from <code><a href="../reference/mr.html">mr()</a></code> except it performs the analysis multiple times for each exposure-outcome combination - each time using a different single SNP to perform the analysis.</p>
<p>The method used to perform the single SNP MR is the Wald ratio by default, though this can be changed, e.g. to use the fixed effects meta analysis method instead:</p>
<pre class="sourceCode r" id="cb50"><code class="sourceCode r"><div class="sourceLine" id="cb50-1" data-line-number="1">res_single &lt;-<span class="st"> </span><span class="kw"><a href="../reference/mr_singlesnp.html">mr_singlesnp</a></span>(dat, <span class="dt">single_method=</span><span class="st">"mr_meta_fixed"</span>)</div></code></pre>
<p>The <code><a href="../reference/mr_singlesnp.html">mr_singlesnp()</a></code> function calculates the full MR using all available SNPs as well, and by default it uses the IVW and MR Egger methods. This can be specified as so:</p>
<pre class="sourceCode r" id="cb51"><code class="sourceCode r"><div class="sourceLine" id="cb51-1" data-line-number="1">res_single &lt;-<span class="st"> </span><span class="kw"><a href="../reference/mr_singlesnp.html">mr_singlesnp</a></span>(dat, <span class="dt">all_method=</span><span class="st">"mr_two_sample_ml"</span>)</div></code></pre>
<p>will perform only the maximum likelihood method for the combined test.</p>
</div>
<div id="leave-one-out-analysis" class="section level3">
<h3 class="hasAnchor">
<a href="#leave-one-out-analysis" class="anchor"></a>Leave-one-out analysis</h3>
<p>It is possible to perform a leave-one-out analysis, where the MR is performed again but leaving out each SNP in turn, to identify if a single SNP is driving the association.</p>
<pre class="sourceCode r" id="cb52"><code class="sourceCode r"><div class="sourceLine" id="cb52-1" data-line-number="1">res_loo &lt;-<span class="st"> </span><span class="kw"><a href="../reference/mr_leaveoneout.html">mr_leaveoneout</a></span>(dat)</div></code></pre>
<p>By default the method used is the inverse variance weighted method, but this can be changed by using the <code>method</code> argument.</p>
<hr>
</div>
</div>
<div id="plots" class="section level2">
<h2 class="hasAnchor">
<a href="#plots" class="anchor"></a>Plots</h2>
<p>There are a few ways to visualise the results, listed below</p>
<div id="scatter-plot" class="section level3">
<h3 class="hasAnchor">
<a href="#scatter-plot" class="anchor"></a>Scatter plot</h3>
<p>We can depict the relationship of the SNP effects on the exposure against the SNP effects on the outcome using a scatter plot.</p>
<pre class="sourceCode r" id="cb53"><code class="sourceCode r"><div class="sourceLine" id="cb53-1" data-line-number="1">res &lt;-<span class="st"> </span><span class="kw"><a href="../reference/mr.html">mr</a></span>(dat)</div>
<div class="sourceLine" id="cb53-2" data-line-number="2"><span class="co">#&gt; Analysing 'TrGPJg' on '7'</span></div>
<div class="sourceLine" id="cb53-3" data-line-number="3">p1 &lt;-<span class="st"> </span><span class="kw"><a href="../reference/mr_scatter_plot.html">mr_scatter_plot</a></span>(res, dat)</div>
<div class="sourceLine" id="cb53-4" data-line-number="4"><span class="co">#&gt; Warning: Ignoring unknown aesthetics: text</span></div></code></pre>
<p>A scatter plot is created for each exposure-outcome test, and stored in <code>p1</code> as a list of plots. For example, to plot the first scatter plot:</p>
<pre class="sourceCode r" id="cb54"><code class="sourceCode r"><div class="sourceLine" id="cb54-1" data-line-number="1">p1[[<span class="dv">1</span>]]</div></code></pre>
<p><img src="guide_files/figure-html/unnamed-chunk-48-1.png" width="700"></p>
<p>And to see how many plots there are:</p>
<pre class="sourceCode r" id="cb55"><code class="sourceCode r"><div class="sourceLine" id="cb55-1" data-line-number="1">
<span class="kw"><a href="https://rdrr.io/r/base/length.html">length</a></span>(p1)</div>
<div class="sourceLine" id="cb55-2" data-line-number="2"><span class="co">#&gt; [1] 1</span></div></code></pre>
<p>Lines are drawn for each method used in <code><a href="../reference/mr.html">mr(dat)</a></code>, the slope of the line corresponding to the estimated causal effect. To limit which lines are drawn, simply specify the desired methods, e.g. to only draw MR Egger and IVW:</p>
<pre class="sourceCode r" id="cb56"><code class="sourceCode r"><div class="sourceLine" id="cb56-1" data-line-number="1">res &lt;-<span class="st"> </span><span class="kw"><a href="../reference/mr.html">mr</a></span>(dat, <span class="dt">method_list=</span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"mr_egger_regression"</span>, <span class="st">"mr_ivw"</span>))</div>
<div class="sourceLine" id="cb56-2" data-line-number="2"><span class="co">#&gt; Analysing 'TrGPJg' on '7'</span></div>
<div class="sourceLine" id="cb56-3" data-line-number="3">p1 &lt;-<span class="st"> </span><span class="kw"><a href="../reference/mr_scatter_plot.html">mr_scatter_plot</a></span>(res, dat)</div>
<div class="sourceLine" id="cb56-4" data-line-number="4"><span class="co">#&gt; Warning: Ignoring unknown aesthetics: text</span></div></code></pre>
<p>It is possible to save this plot using the <code>ggsave()</code> function, e.g. to save as a pdf</p>
<pre class="sourceCode r" id="cb57"><code class="sourceCode r"><div class="sourceLine" id="cb57-1" data-line-number="1">
<span class="kw">ggsave</span>(p1[[<span class="dv">1</span>]], <span class="dt">file=</span><span class="st">"filename.pdf"</span>, <span class="dt">width=</span><span class="dv">7</span>, <span class="dt">height=</span><span class="dv">7</span>)</div></code></pre>
<p>or save as a png</p>
<pre class="sourceCode r" id="cb58"><code class="sourceCode r"><div class="sourceLine" id="cb58-1" data-line-number="1">
<span class="kw">ggsave</span>(p1[[<span class="dv">1</span>]], <span class="dt">file=</span><span class="st">"filename.png"</span>, <span class="dt">width=</span><span class="dv">7</span>, <span class="dt">height=</span><span class="dv">7</span>)</div></code></pre>
<p>See <code>?ggsave</code> for more info.</p>
</div>
<div id="forest-plot" class="section level3">
<h3 class="hasAnchor">
<a href="#forest-plot" class="anchor"></a>Forest plot</h3>
<p>Use the <code><a href="../reference/mr_forest_plot.html">mr_forest_plot()</a></code> function to compare the MR estimates using the different MR methods against the single SNP tests.</p>
<pre class="sourceCode r" id="cb59"><code class="sourceCode r"><div class="sourceLine" id="cb59-1" data-line-number="1">res_single &lt;-<span class="st"> </span><span class="kw"><a href="../reference/mr_singlesnp.html">mr_singlesnp</a></span>(dat)</div>
<div class="sourceLine" id="cb59-2" data-line-number="2">p2 &lt;-<span class="st"> </span><span class="kw"><a href="../reference/mr_forest_plot.html">mr_forest_plot</a></span>(res_single)</div>
<div class="sourceLine" id="cb59-3" data-line-number="3">p2[[<span class="dv">1</span>]]</div>
<div class="sourceLine" id="cb59-4" data-line-number="4"><span class="co">#&gt; Warning: Removed 1 rows containing missing values (geom_errorbarh).</span></div>
<div class="sourceLine" id="cb59-5" data-line-number="5"><span class="co">#&gt; Warning: Removed 1 rows containing missing values (geom_point).</span></div></code></pre>
<p><img src="guide_files/figure-html/unnamed-chunk-53-1.png" width="700"></p>
<p>Here, the plot shows the causal effect as estimated using each of the SNPs on their own, and comparing against the causal effect as estimated using the methods that use all the SNPs.</p>
<p>To get plots that use different methods, specify them in the <code><a href="../reference/mr_singlesnp.html">mr_singlesnp()</a></code> function:</p>
<pre class="sourceCode r" id="cb60"><code class="sourceCode r"><div class="sourceLine" id="cb60-1" data-line-number="1">res_single &lt;-<span class="st"> </span><span class="kw"><a href="../reference/mr_singlesnp.html">mr_singlesnp</a></span>(dat, <span class="dt">all_method=</span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"mr_ivw"</span>, <span class="st">"mr_two_sample_ml"</span>))</div>
<div class="sourceLine" id="cb60-2" data-line-number="2">p2 &lt;-<span class="st"> </span><span class="kw"><a href="../reference/mr_forest_plot.html">mr_forest_plot</a></span>(res_single)</div>
<div class="sourceLine" id="cb60-3" data-line-number="3">p2[[<span class="dv">1</span>]]</div>
<div class="sourceLine" id="cb60-4" data-line-number="4"><span class="co">#&gt; Warning: Removed 1 rows containing missing values (geom_errorbarh).</span></div>
<div class="sourceLine" id="cb60-5" data-line-number="5"><span class="co">#&gt; Warning: Removed 1 rows containing missing values (geom_point).</span></div></code></pre>
<p><img src="guide_files/figure-html/unnamed-chunk-54-1.png" width="700"></p>
</div>
<div id="leave-one-out-plot" class="section level3">
<h3 class="hasAnchor">
<a href="#leave-one-out-plot" class="anchor"></a>Leave-one-out plot</h3>
<p>Use the <code>mr_leaveoneout_plot</code> function to visualise the leave-one-out analysis:</p>
<pre class="sourceCode r" id="cb61"><code class="sourceCode r"><div class="sourceLine" id="cb61-1" data-line-number="1">res_loo &lt;-<span class="st"> </span><span class="kw"><a href="../reference/mr_leaveoneout.html">mr_leaveoneout</a></span>(dat)</div>
<div class="sourceLine" id="cb61-2" data-line-number="2">p3 &lt;-<span class="st"> </span><span class="kw"><a href="../reference/mr_leaveoneout_plot.html">mr_leaveoneout_plot</a></span>(res_loo)</div>
<div class="sourceLine" id="cb61-3" data-line-number="3">p3[[<span class="dv">1</span>]]</div>
<div class="sourceLine" id="cb61-4" data-line-number="4"><span class="co">#&gt; Warning: Removed 1 rows containing missing values (geom_errorbarh).</span></div>
<div class="sourceLine" id="cb61-5" data-line-number="5"><span class="co">#&gt; Warning: Removed 1 rows containing missing values (geom_point).</span></div></code></pre>
<p><img src="guide_files/figure-html/unnamed-chunk-55-1.png" width="700"></p>
<p>Specify the test to use e.g. <code><a href="../reference/mr_leaveoneout.html">mr_leaveoneout(dat, method = mr_egger_regression)</a></code> to use Egger regression.</p>
</div>
<div id="funnel-plot" class="section level3">
<h3 class="hasAnchor">
<a href="#funnel-plot" class="anchor"></a>Funnel plot</h3>
<p>Asymmetry in a funnel plot is useful for gauging the reliability of a particular MR analysis. Funnel plots can be produced using the single SNP results as follows: howzit</p>
<pre class="sourceCode r" id="cb62"><code class="sourceCode r"><div class="sourceLine" id="cb62-1" data-line-number="1">res_single &lt;-<span class="st"> </span><span class="kw"><a href="../reference/mr_singlesnp.html">mr_singlesnp</a></span>(dat)</div>
<div class="sourceLine" id="cb62-2" data-line-number="2">p4 &lt;-<span class="st"> </span><span class="kw"><a href="../reference/mr_funnel_plot.html">mr_funnel_plot</a></span>(res_single)</div>
<div class="sourceLine" id="cb62-3" data-line-number="3">p4[[<span class="dv">1</span>]]</div></code></pre>
<p><img src="guide_files/figure-html/unnamed-chunk-56-1.png" width="700"></p>
</div>
</div>
<div id="to-many-forest-plot" class="section level2">
<h2 class="hasAnchor">
<a href="#to-many-forest-plot" class="anchor"></a>1-to-many forest plot</h2>
<p>A 1-to-many MR analysis interrogates the effect of a single exposure on multiple outcomes or multiple exposures on a single outcome. The results of this analysis can be visualised using the 1-to-many forest plot, with or without stratification on a categorical variable. From a visual point of view, the function works best for 50 or fewer results and is not really designed to handle more than a 100 results. If your number of results is much greater than 50, it may be better to split these across two separate plots. For example, if you have 100 sets of results you could divide these equally across two plots and then combine the two plots together in another programme like powerpoint. The function assumes the results are already in the right order for plotting. As such, users are advised to sort their results according to how they would like them to appear in the plot. Users can use their own code to do this or they can use the sort.1.to.many() function.</p>
<div id="step-1-generate-1-to-many-mr-results" class="section level3">
<h3 class="hasAnchor">
<a href="#step-1-generate-1-to-many-mr-results" class="anchor"></a>Step 1: generate 1-to-many MR results</h3>
<pre class="sourceCode r" id="cb63"><code class="sourceCode r"><div class="sourceLine" id="cb63-1" data-line-number="1">exp_dat &lt;-<span class="st"> </span><span class="kw"><a href="../reference/extract_instruments.html">extract_instruments</a></span>(<span class="dt">outcomes=</span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dv">2</span>,<span class="dv">100</span>,<span class="dv">1032</span>,<span class="dv">104</span>,<span class="dv">1</span>,<span class="dv">72</span>,<span class="dv">999</span>))</div>
<div class="sourceLine" id="cb63-2" data-line-number="2">
<span class="kw"><a href="https://rdrr.io/r/base/table.html">table</a></span>(exp_dat<span class="op">$</span>exposure)</div>
<div class="sourceLine" id="cb63-3" data-line-number="3">chd_out_dat &lt;-<span class="st"> </span><span class="kw"><a href="../reference/extract_outcome_data.html">extract_outcome_data</a></span>(</div>
<div class="sourceLine" id="cb63-4" data-line-number="4">    <span class="dt">snps =</span> exp_dat<span class="op">$</span>SNP,</div>
<div class="sourceLine" id="cb63-5" data-line-number="5">    <span class="dt">outcomes =</span> <span class="dv">7</span>
</div>
<div class="sourceLine" id="cb63-6" data-line-number="6">)</div>
<div class="sourceLine" id="cb63-7" data-line-number="7"></div>
<div class="sourceLine" id="cb63-8" data-line-number="8">dat2 &lt;-<span class="st"> </span><span class="kw"><a href="../reference/harmonise_data.html">harmonise_data</a></span>(</div>
<div class="sourceLine" id="cb63-9" data-line-number="9">    <span class="dt">exposure_dat =</span> exp_dat, </div>
<div class="sourceLine" id="cb63-10" data-line-number="10">    <span class="dt">outcome_dat =</span> chd_out_dat</div>
<div class="sourceLine" id="cb63-11" data-line-number="11">)</div>
<div class="sourceLine" id="cb63-12" data-line-number="12">res&lt;-<span class="kw"><a href="../reference/mr.html">mr</a></span>(dat2)</div></code></pre>
</div>
<div id="step-2--make-the-1-to-many-forest-plot" class="section level3">
<h3 class="hasAnchor">
<a href="#step-2--make-the-1-to-many-forest-plot" class="anchor"></a>Step 2. Make the 1-to-many forest plot</h3>
<div id="example-1--effect-of-multiple-risk-factors-on-coronary-heart-disease" class="section level4">
<h4 class="hasAnchor">
<a href="#example-1--effect-of-multiple-risk-factors-on-coronary-heart-disease" class="anchor"></a>Example 1. Effect of multiple risk factors on coronary heart disease</h4>
<p>In this example we wish to plot results from an MR analysis of the effect of multiple exposures on coronary heart disease, with results sorted by decreasing effect size (largest effect at the top of the plot) and with one MR method for each unique exposure-outcome combination. We will also make the size of each point estimate proportional to its inverse variance. This is a useful way to draw attention towards the most reliable results and away from results with very wide confidence intervals. To specify the size of the point estimate, set the weight argument to the name of the column in the data with the weight information.</p>
<pre class="sourceCode r" id="cb64"><code class="sourceCode r"><div class="sourceLine" id="cb64-1" data-line-number="1">res&lt;-<span class="kw"><a href="../reference/subset_on_method.html">subset_on_method</a></span>(res) <span class="co">#default is to subset on either the IVW method (&gt;1 instrumental SNP) or Wald ratio method (1 instrumental SNP). </span>
</div>
<div class="sourceLine" id="cb64-2" data-line-number="2">res&lt;-<span class="kw"><a href="../reference/sort_1_to_many.html">sort_1_to_many</a></span>(res,<span class="dt">b=</span><span class="st">"b"</span>,<span class="dt">sort_action=</span><span class="dv">4</span>) <span class="co">#this sorts results by decreasing effect size (largest effect at top of the plot)</span>
</div>
<div class="sourceLine" id="cb64-3" data-line-number="3">res&lt;-<span class="kw"><a href="../reference/split_exposure.html">split_exposure</a></span>(res) <span class="co"># to keep the Y axis label clean we exclude the exposure ID labels from the exposure column </span>
</div>
<div class="sourceLine" id="cb64-4" data-line-number="4">res<span class="op">$</span>weight&lt;-<span class="dv">1</span><span class="op">/</span>res<span class="op">$</span>se</div>
<div class="sourceLine" id="cb64-5" data-line-number="5"></div>
<div class="sourceLine" id="cb64-6" data-line-number="6">
<span class="kw"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/Log.html">exp</a></span>(res<span class="op">$</span>b<span class="fl">-1.96</span><span class="op">*</span>res<span class="op">$</span>se)) <span class="co">#identify value for 'lo' in forest_plot_1_to_many</span>
</div>
<div class="sourceLine" id="cb64-7" data-line-number="7">
<span class="kw"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/Log.html">exp</a></span>(res<span class="op">$</span>b<span class="fl">+1.96</span><span class="op">*</span>res<span class="op">$</span>se)) <span class="co">#identify value for 'up' in forest_plot_1_to_many</span>
</div>
<div class="sourceLine" id="cb64-8" data-line-number="8"></div>
<div class="sourceLine" id="cb64-9" data-line-number="9">
<span class="kw"><a href="../reference/forest_plot_1_to_many.html">forest_plot_1_to_many</a></span>(res,<span class="dt">b=</span><span class="st">"b"</span>,<span class="dt">se=</span><span class="st">"se"</span>,</div>
<div class="sourceLine" id="cb64-10" data-line-number="10">    <span class="dt">exponentiate=</span>T,<span class="dt">ao_slc=</span>F,<span class="dt">lo=</span><span class="fl">0.3</span>,<span class="dt">up=</span><span class="fl">2.5</span>,</div>
<div class="sourceLine" id="cb64-11" data-line-number="11">    <span class="dt">TraitM=</span><span class="st">"exposure"</span>,<span class="dt">col1_width=</span><span class="dv">2</span>,<span class="dt">by=</span><span class="ot">NULL</span>,</div>
<div class="sourceLine" id="cb64-12" data-line-number="12">    <span class="dt">trans=</span><span class="st">"log2"</span>,<span class="dt">xlab=</span><span class="st">"OR for CHD per SD increase in risk factor (95% confidence interval)"</span>,<span class="dt">weight=</span><span class="st">"weight"</span>)</div></code></pre>
<p>It is also possible to add additional columns and column titles and to choose the size of the text in the columns:</p>
<pre class="sourceCode r" id="cb65"><code class="sourceCode r"><div class="sourceLine" id="cb65-1" data-line-number="1">res<span class="op">$</span>pval&lt;-<span class="kw"><a href="https://rdrr.io/r/base/formatc.html">formatC</a></span>(res<span class="op">$</span>pval, <span class="dt">format =</span> <span class="st">"e"</span>, <span class="dt">digits =</span> <span class="dv">2</span>)</div>
<div class="sourceLine" id="cb65-2" data-line-number="2">
<span class="kw"><a href="../reference/forest_plot_1_to_many.html">forest_plot_1_to_many</a></span>(res,<span class="dt">b=</span><span class="st">"b"</span>,<span class="dt">se=</span><span class="st">"se"</span>,</div>
<div class="sourceLine" id="cb65-3" data-line-number="3">    <span class="dt">exponentiate=</span>T,<span class="dt">ao_slc=</span>F,<span class="dt">lo=</span><span class="fl">0.3</span>,<span class="dt">up=</span><span class="fl">2.5</span>,</div>
<div class="sourceLine" id="cb65-4" data-line-number="4">    <span class="dt">TraitM=</span><span class="st">"exposure"</span>,<span class="dt">by=</span><span class="ot">NULL</span>,</div>
<div class="sourceLine" id="cb65-5" data-line-number="5">    <span class="dt">trans=</span><span class="st">"log2"</span>,<span class="dt">xlab=</span><span class="st">"OR for CHD per SD increase in risk factor (95% CI)"</span>,    </div>
<div class="sourceLine" id="cb65-6" data-line-number="6">    <span class="dt">weight=</span><span class="st">"weight"</span>,<span class="dt">subheading_size=</span><span class="dv">11</span>,</div>
<div class="sourceLine" id="cb65-7" data-line-number="7">    <span class="dt">col1_title=</span><span class="st">"Risk factor"</span>,</div>
<div class="sourceLine" id="cb65-8" data-line-number="8">    <span class="dt">col1_width=</span><span class="fl">2.5</span>,</div>
<div class="sourceLine" id="cb65-9" data-line-number="9">    <span class="dt">col_text_size=</span><span class="dv">4</span>,</div>
<div class="sourceLine" id="cb65-10" data-line-number="10">    <span class="dt">addcols=</span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"nsnp"</span>,<span class="st">"pval"</span>),</div>
<div class="sourceLine" id="cb65-11" data-line-number="11">    <span class="dt">addcol_widths=</span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1.0</span>,<span class="fl">1.0</span>),</div>
<div class="sourceLine" id="cb65-12" data-line-number="12">    <span class="dt">addcol_titles=</span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"No. SNPs"</span>,<span class="st">"P-val"</span>)</div>
<div class="sourceLine" id="cb65-13" data-line-number="13">    )</div></code></pre>
<p>In my own workflow I prefer to to keep the plot free of axis and column titles and to add them separately in a program like powerpoint:</p>
<pre class="sourceCode r" id="cb66"><code class="sourceCode r"><div class="sourceLine" id="cb66-1" data-line-number="1">
<span class="kw"><a href="../reference/forest_plot_1_to_many.html">forest_plot_1_to_many</a></span>(res,<span class="dt">b=</span><span class="st">"b"</span>,<span class="dt">se=</span><span class="st">"se"</span>,</div>
<div class="sourceLine" id="cb66-2" data-line-number="2">    <span class="dt">exponentiate=</span>T,<span class="dt">ao_slc=</span>F,<span class="dt">lo=</span><span class="fl">0.3</span>,<span class="dt">up=</span><span class="fl">3.0</span>,</div>
<div class="sourceLine" id="cb66-3" data-line-number="3">    <span class="dt">TraitM=</span><span class="st">"exposure"</span>,<span class="dt">col1_width=</span><span class="fl">2.0</span>,<span class="dt">by=</span><span class="ot">NULL</span>,</div>
<div class="sourceLine" id="cb66-4" data-line-number="4">    <span class="dt">trans=</span><span class="st">"log2"</span>,<span class="dt">xlab=</span><span class="st">""</span>,<span class="dt">addcols=</span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"nsnp"</span>,<span class="st">"pval"</span>),</div>
<div class="sourceLine" id="cb66-5" data-line-number="5">    <span class="dt">weight=</span><span class="st">"weight"</span>, <span class="dt">col_text_size=</span><span class="dv">4</span>,   </div>
<div class="sourceLine" id="cb66-6" data-line-number="6">    <span class="dt">addcol_widths=</span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0.5</span>,<span class="fl">1.0</span>),<span class="dt">addcol_titles=</span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">""</span>,<span class="st">""</span>))</div></code></pre>
</div>
<div id="example-2--mr-results-for-multiple-mr-methods-grouped-by-multiple-exposures" class="section level4">
<h4 class="hasAnchor">
<a href="#example-2--mr-results-for-multiple-mr-methods-grouped-by-multiple-exposures" class="anchor"></a>Example 2. MR results for multiple MR methods grouped by multiple exposures</h4>
<p>In this next example we plot the results from an analysis of the effect of multiple exposures on coronary heart disease using multiple methods, with results grouped by exposure. We also want the result for the IVW method to be given priority and to go above the other methods. We also want the exposure with the largest IVW effect size to go the top of the plot. We also set the TraitM argument to the column describing the MR method. This is because we are grouping the results on the exposures. Normally the row labels would correspond to the exposures but in this example we want the row names to correspond to the MR method.</p>
<pre class="sourceCode r" id="cb67"><code class="sourceCode r"><div class="sourceLine" id="cb67-1" data-line-number="1">res&lt;-<span class="kw"><a href="../reference/mr.html">mr</a></span>(dat2)</div>
<div class="sourceLine" id="cb67-2" data-line-number="2">res&lt;-<span class="kw"><a href="../reference/split_exposure.html">split_exposure</a></span>(res) <span class="co"># to keep the Y axis label clean we exclude the exposure ID labels from the exposure column </span>
</div>
<div class="sourceLine" id="cb67-3" data-line-number="3"></div>
<div class="sourceLine" id="cb67-4" data-line-number="4">res&lt;-<span class="kw"><a href="../reference/sort_1_to_many.html">sort_1_to_many</a></span>(res,<span class="dt">group=</span><span class="st">"exposure"</span>,<span class="dt">sort_action=</span><span class="dv">3</span>,<span class="dt">priority=</span><span class="st">"Inverse variance weighted"</span>,<span class="dt">trait_m=</span><span class="st">"method"</span>)</div>
<div class="sourceLine" id="cb67-5" data-line-number="5"></div>
<div class="sourceLine" id="cb67-6" data-line-number="6">
<span class="kw"><a href="../reference/forest_plot_1_to_many.html">forest_plot_1_to_many</a></span>(res,<span class="dt">b=</span><span class="st">"b"</span>,<span class="dt">se=</span><span class="st">"se"</span>,</div>
<div class="sourceLine" id="cb67-7" data-line-number="7">    <span class="dt">exponentiate=</span>T,<span class="dt">trans=</span><span class="st">"log2"</span>,<span class="dt">ao_slc=</span>F,<span class="dt">lo=</span><span class="fl">0.03</span>,</div>
<div class="sourceLine" id="cb67-8" data-line-number="8">    <span class="dt">up=</span><span class="dv">22</span>,<span class="dt">col1_width=</span><span class="dv">2</span>,<span class="dt">by=</span><span class="st">"exposure"</span>,<span class="dt">TraitM=</span><span class="st">"method"</span>,</div>
<div class="sourceLine" id="cb67-9" data-line-number="9">    <span class="dt">xlab=</span><span class="st">"OR for CHD per SD increase in risk factor (95% confidence interval)"</span>,</div>
<div class="sourceLine" id="cb67-10" data-line-number="10">    <span class="dt">subheading_size=</span><span class="dv">12</span>,<span class="dt">col_text_size=</span><span class="dv">4</span>)</div></code></pre>
</div>
<div id="example-3--stratify-results-on-a-grouping-variable" class="section level4">
<h4 class="hasAnchor">
<a href="#example-3--stratify-results-on-a-grouping-variable" class="anchor"></a>Example 3. Stratify results on a grouping variable</h4>
<p>In this next example we plot the same results as above but with results stratified by a grouping variable. We also select one MR method for each unique exposure-outcome combination and sort the results by decreasing effect size within each group (i.e. largest effect at the top).</p>
<pre class="sourceCode r" id="cb68"><code class="sourceCode r"><div class="sourceLine" id="cb68-1" data-line-number="1"></div>
<div class="sourceLine" id="cb68-2" data-line-number="2">res&lt;-<span class="kw"><a href="../reference/mr.html">mr</a></span>(dat2) </div>
<div class="sourceLine" id="cb68-3" data-line-number="3">res&lt;-<span class="kw"><a href="../reference/split_exposure.html">split_exposure</a></span>(res)</div>
<div class="sourceLine" id="cb68-4" data-line-number="4">res&lt;-<span class="kw"><a href="../reference/subset_on_method.html">subset_on_method</a></span>(res)</div>
<div class="sourceLine" id="cb68-5" data-line-number="5">res<span class="op">$</span>subcategory[res<span class="op">$</span>exposure <span class="op">%in%</span><span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"Adiponectin"</span>,<span class="st">"Hip circumference"</span>,<span class="st">"Waist circumference"</span>)]&lt;-<span class="st">"Group 1"</span>
</div>
<div class="sourceLine" id="cb68-6" data-line-number="6">res<span class="op">$</span>subcategory[<span class="kw"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(res<span class="op">$</span>subcategory)]&lt;-<span class="st">"Group 2"</span>
</div>
<div class="sourceLine" id="cb68-7" data-line-number="7">res<span class="op">$</span>weight&lt;-<span class="dv">1</span><span class="op">/</span>res<span class="op">$</span>se</div>
<div class="sourceLine" id="cb68-8" data-line-number="8">res&lt;-<span class="kw"><a href="../reference/sort_1_to_many.html">sort_1_to_many</a></span>(res,<span class="dt">sort_action=</span><span class="dv">1</span>,<span class="dt">group=</span><span class="st">"subcategory"</span>)</div>
<div class="sourceLine" id="cb68-9" data-line-number="9"></div>
<div class="sourceLine" id="cb68-10" data-line-number="10">
<span class="kw"><a href="../reference/forest_plot_1_to_many.html">forest_plot_1_to_many</a></span>(res,<span class="dt">b=</span><span class="st">"b"</span>,<span class="dt">se=</span><span class="st">"se"</span>,</div>
<div class="sourceLine" id="cb68-11" data-line-number="11">    <span class="dt">exponentiate=</span>T,<span class="dt">trans=</span><span class="st">"log2"</span>,<span class="dt">ao_slc=</span>F,<span class="dt">lo=</span><span class="fl">0.3</span>,</div>
<div class="sourceLine" id="cb68-12" data-line-number="12">    <span class="dt">up=</span><span class="fl">2.5</span>,<span class="dt">TraitM=</span><span class="st">"exposure"</span>,<span class="dt">col_text_size=</span><span class="dv">4</span>,<span class="dt">col1_width=</span><span class="fl">1.5</span>,<span class="dt">by=</span><span class="st">"subcategory"</span>,</div>
<div class="sourceLine" id="cb68-13" data-line-number="13">    <span class="dt">xlab=</span><span class="st">"OR for CHD per SD increase in risk factor (95% confidence interval)"</span>,</div>
<div class="sourceLine" id="cb68-14" data-line-number="14">    <span class="dt">subheading_size=</span><span class="dv">14</span>,<span class="dt">weight=</span><span class="st">"weight"</span>)</div></code></pre>
<p>In the above example we made up an arbitrary grouping variable called “subcategory” with values “Group 1” and “Group 2”. Typically, however, the grouping variable might correspond to something like a trait ontology (e.g. anthropometric and glycemic traits) or study design (e.g. MR and observational studies).</p>
</div>
<div id="example-4--effect-of-bmi-on-103-diseases" class="section level4">
<h4 class="hasAnchor">
<a href="#example-4--effect-of-bmi-on-103-diseases" class="anchor"></a>Example 4. Effect of BMI on 103 diseases</h4>
<p>The plot function works best with 50 or fewer rows and is not really designed to handle more than a 100. Visualising a single-column forest plot with 100 results is also quite difficult. If your number of results is much greater than 50, it is advisable to split the results across two different plots. In the example below we select BMI as the exposure and test this against 103 diseases in MR-Base:</p>
<pre class="sourceCode r" id="cb69"><code class="sourceCode r"><div class="sourceLine" id="cb69-1" data-line-number="1"></div>
<div class="sourceLine" id="cb69-2" data-line-number="2">exp_dat &lt;-<span class="st"> </span><span class="kw"><a href="../reference/extract_instruments.html">extract_instruments</a></span>(<span class="dt">outcomes=</span><span class="dv">2</span>)  <span class="co">#extract instruments for BMI</span>
</div>
<div class="sourceLine" id="cb69-3" data-line-number="3">ao&lt;-<span class="kw"><a href="../reference/available_outcomes.html">available_outcomes</a></span>()</div>
<div class="sourceLine" id="cb69-4" data-line-number="4">ao&lt;-ao[ao<span class="op">$</span>category<span class="op">==</span><span class="st">"Disease"</span>,] <span class="co">#identify diseases</span>
</div>
<div class="sourceLine" id="cb69-5" data-line-number="5">ao&lt;-ao[<span class="kw"><a href="https://rdrr.io/r/base/which.html">which</a></span>(ao<span class="op">$</span>ncase<span class="op">&gt;</span><span class="dv">100</span>),]</div>
<div class="sourceLine" id="cb69-6" data-line-number="6"></div>
<div class="sourceLine" id="cb69-7" data-line-number="7">dis_dat &lt;-<span class="st"> </span><span class="kw"><a href="../reference/extract_outcome_data.html">extract_outcome_data</a></span>(</div>
<div class="sourceLine" id="cb69-8" data-line-number="8">    <span class="dt">snps =</span> exp_dat<span class="op">$</span>SNP,</div>
<div class="sourceLine" id="cb69-9" data-line-number="9">    <span class="dt">outcomes =</span> ao<span class="op">$</span>id</div>
<div class="sourceLine" id="cb69-10" data-line-number="10">)</div>
<div class="sourceLine" id="cb69-11" data-line-number="11"></div>
<div class="sourceLine" id="cb69-12" data-line-number="12">dat3 &lt;-<span class="st"> </span><span class="kw"><a href="../reference/harmonise_data.html">harmonise_data</a></span>(</div>
<div class="sourceLine" id="cb69-13" data-line-number="13">    <span class="dt">exposure_dat =</span> exp_dat, </div>
<div class="sourceLine" id="cb69-14" data-line-number="14">    <span class="dt">outcome_dat =</span> dis_dat</div>
<div class="sourceLine" id="cb69-15" data-line-number="15">)</div>
<div class="sourceLine" id="cb69-16" data-line-number="16"></div>
<div class="sourceLine" id="cb69-17" data-line-number="17"></div>
<div class="sourceLine" id="cb69-18" data-line-number="18">res&lt;-<span class="kw"><a href="../reference/mr.html">mr</a></span>(dat3,<span class="dt">method_list=</span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"mr_wald_ratio"</span>,<span class="st">"mr_ivw"</span>))</div>
<div class="sourceLine" id="cb69-19" data-line-number="19">res&lt;-<span class="kw"><a href="../reference/split_outcome.html">split_outcome</a></span>(res) <span class="co"># to keep the Y axis label clean we exclude the exposure ID labels from the exposure column </span>
</div>
<div class="sourceLine" id="cb69-20" data-line-number="20"></div>
<div class="sourceLine" id="cb69-21" data-line-number="21">res&lt;-<span class="kw"><a href="../reference/sort_1_to_many.html">sort_1_to_many</a></span>(res,<span class="dt">b=</span><span class="st">"b"</span>,<span class="dt">sort_action=</span><span class="dv">4</span>) <span class="co">#this sorts results by decreasing effect size (largest effect at top of the plot)</span>
</div></code></pre>
<p>MR results for 103 diseases can be difficult to visualise in a single-column forest plot. In my own workflow I would split these across two plots and then join them together in a separate program, such as powerpoint, and do further refinements there. I typically save my plots using the pdf() graphics device. In this particular example the disease labels probably require some cleaning up (some are a bit long) or alternatively the column text size could be made smaller. It is also possible to change the colour of the plot and the shape of the point estimates. Type ?forest_plot_1_to_many for further details.</p>
<pre class="sourceCode r" id="cb70"><code class="sourceCode r"><div class="sourceLine" id="cb70-1" data-line-number="1"></div>
<div class="sourceLine" id="cb70-2" data-line-number="2">res1&lt;-res[<span class="dv">1</span><span class="op">:</span><span class="dv">52</span>,]</div>
<div class="sourceLine" id="cb70-3" data-line-number="3">res2&lt;-res[<span class="dv">53</span><span class="op">:</span><span class="dv">103</span>,]</div>
<div class="sourceLine" id="cb70-4" data-line-number="4"></div>
<div class="sourceLine" id="cb70-5" data-line-number="5">plot1&lt;-<span class="kw"><a href="../reference/forest_plot_1_to_many.html">forest_plot_1_to_many</a></span>(res1,<span class="dt">b=</span><span class="st">"b"</span>,<span class="dt">se=</span><span class="st">"se"</span>,</div>
<div class="sourceLine" id="cb70-6" data-line-number="6">    <span class="dt">exponentiate=</span>T,<span class="dt">trans=</span><span class="st">"log2"</span>,<span class="dt">ao_slc=</span>F,<span class="dt">lo=</span><span class="fl">0.004</span>,</div>
<div class="sourceLine" id="cb70-7" data-line-number="7">    <span class="dt">up=</span><span class="dv">461</span>,<span class="dt">col1_width=</span><span class="dv">2</span>,<span class="dt">TraitM=</span><span class="st">"outcome"</span>,</div>
<div class="sourceLine" id="cb70-8" data-line-number="8">    <span class="dt">col_text_size=</span><span class="dv">3</span>,<span class="dt">xlab=</span><span class="st">""</span>)</div>
<div class="sourceLine" id="cb70-9" data-line-number="9"></div>
<div class="sourceLine" id="cb70-10" data-line-number="10">plot2&lt;-<span class="kw"><a href="../reference/forest_plot_1_to_many.html">forest_plot_1_to_many</a></span>(res2,<span class="dt">b=</span><span class="st">"b"</span>,<span class="dt">se=</span><span class="st">"se"</span>,</div>
<div class="sourceLine" id="cb70-11" data-line-number="11">    <span class="dt">exponentiate=</span>T,<span class="dt">trans=</span><span class="st">"log2"</span>,<span class="dt">ao_slc=</span>F,<span class="dt">lo=</span><span class="fl">0.004</span>,</div>
<div class="sourceLine" id="cb70-12" data-line-number="12">    <span class="dt">up=</span><span class="dv">461</span>,<span class="dt">col1_width=</span><span class="dv">2</span>,<span class="dt">TraitM=</span><span class="st">"outcome"</span>,</div>
<div class="sourceLine" id="cb70-13" data-line-number="13">    <span class="dt">subheading_size=</span><span class="dv">11</span>,<span class="dt">col_text_size=</span><span class="dv">3</span>,<span class="dt">xlab=</span><span class="st">""</span>)</div>
<div class="sourceLine" id="cb70-14" data-line-number="14"></div>
<div class="sourceLine" id="cb70-15" data-line-number="15">plot1</div></code></pre>
<p><img src="guide_files/figure-html/unnamed-chunk-64-1.png" width="700"></p>
<pre class="sourceCode r" id="cb71"><code class="sourceCode r"><div class="sourceLine" id="cb71-1" data-line-number="1">plot2</div></code></pre>
<p><img src="guide_files/figure-html/unnamed-chunk-64-2.png" width="700"></p>
<pre class="sourceCode r" id="cb72"><code class="sourceCode r"><div class="sourceLine" id="cb72-1" data-line-number="1"></div>
<div class="sourceLine" id="cb72-2" data-line-number="2">
<span class="kw"><a href="https://rdrr.io/r/grDevices/pdf.html">pdf</a></span>(<span class="st">"plot1.pdf"</span>,<span class="dt">height=</span><span class="dv">10</span>,<span class="dt">width=</span><span class="dv">8</span>)</div>
<div class="sourceLine" id="cb72-3" data-line-number="3">plot1</div>
<div class="sourceLine" id="cb72-4" data-line-number="4">
<span class="kw"><a href="https://rdrr.io/r/grDevices/dev.html">dev.off</a></span>() </div>
<div class="sourceLine" id="cb72-5" data-line-number="5"><span class="co">#&gt; quartz_off_screen </span></div>
<div class="sourceLine" id="cb72-6" data-line-number="6"><span class="co">#&gt;                 2</span></div></code></pre>
<hr>
</div>
</div>
</div>
<div id="mr-raps-many-weak-instruments-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#mr-raps-many-weak-instruments-analysis" class="anchor"></a>MR.RAPS: Many weak instruments analysis</h2>
<p>MR.RAPS (Robust Adjusted Profile Score) is a recently proposed method that considers the measurement error in SNP-exposure effects, is unbiased when there are many (e.g. hundreds of) weak instruments, and is robust to systematic and idiosyncratic pleiotropy. See the <a href="https://arxiv.org/abs/1801.09652">arXiv preprint</a> for more detail about the statistical methodology.</p>
<p>MR.RAPS is implemented in the R package <em>mr.raps</em> that is available on CRAN. It can be directly called from TwoSampleMR by</p>
<pre class="sourceCode r" id="cb73"><code class="sourceCode r"><div class="sourceLine" id="cb73-1" data-line-number="1">res &lt;-<span class="st"> </span><span class="kw"><a href="../reference/mr.html">mr</a></span>(dat, <span class="dt">method_list =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"mr_raps"</span>))</div></code></pre>
<p>MR.RAPS comes with two main options: <em>over.dispersion</em> (whether the method should consider systematic pleiotropy) and <em>loss.function</em> (either “l2”, “huber”, or “tukey”). The latter two loss functions are robust to idiosyncratic pleiotropy. The default option is <em>over.dispersion = TRUE</em> and <em>loss.function = “tukey”</em>. To change these options, modify the <em>parameters</em> argument of <em>mr</em> by (for example)</p>
<pre class="sourceCode r" id="cb74"><code class="sourceCode r"><div class="sourceLine" id="cb74-1" data-line-number="1">res &lt;-<span class="st"> </span><span class="kw"><a href="../reference/mr.html">mr</a></span>(dat, <span class="dt">method_list =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"mr_raps"</span>), <span class="dt">parameters =</span> <span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="dt">over.dispersion =</span> <span class="ot">FALSE</span>, <span class="dt">loss.function =</span> <span class="st">"l2"</span>))</div></code></pre>
<hr>
</div>
<div id="reports" class="section level2">
<h2 class="hasAnchor">
<a href="#reports" class="anchor"></a>Reports</h2>
<p>A report can be generated that performs all MR analyses, sensitivity analyses, and plots, and presents them in a single self-contained html web page, word document, or pdf document.</p>
<pre class="sourceCode r" id="cb75"><code class="sourceCode r"><div class="sourceLine" id="cb75-1" data-line-number="1">
<span class="kw"><a href="../reference/mr_report.html">mr_report</a></span>(dat)</div></code></pre>
<p>By default this produces a html file in the current working directory, but see the help pages on how to modify this.</p>
<p>This function will create a separate report file for every exposure-outcome combination that is present in the <code>dat</code> object.</p>
<hr>
</div>
<div id="mr-steiger-directionality-test" class="section level2">
<h2 class="hasAnchor">
<a href="#mr-steiger-directionality-test" class="anchor"></a>MR Steiger directionality test</h2>
<p>This is an implementation of the method described here:</p>
<p><a href="http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1007081">Hemani G, Tilling K, Davey Smith G.<br><strong>Orienting the causal relationship between imprecisely measured traits using GWAS summary data.</strong><br> PLoS Genetics. 2017. 13(11).</a></p>
<p>In MR it is assumed that the instruments influence the exposure first and then the outcome through the exposure. But sometimes this is difficult to evaluate, for example is a cis-acting SNP influencing gene expression levels or DNA methylation levels first? The causal direction between the hypothesised exposure and outcomes can be tested using the Steiger test [reference to go here]. For example:</p>
<pre class="sourceCode r" id="cb76"><code class="sourceCode r"><div class="sourceLine" id="cb76-1" data-line-number="1">out &lt;-<span class="st"> </span><span class="kw"><a href="../reference/directionality_test.html">directionality_test</a></span>(dat)</div>
<div class="sourceLine" id="cb76-2" data-line-number="2"><span class="co">#&gt; r.exposure and/or r.outcome not present.</span></div>
<div class="sourceLine" id="cb76-3" data-line-number="3"><span class="co">#&gt; Calculating approximate SNP-exposure and/or SNP-outcome correlations, assuming all are quantitative traits. Please pre-calculate r.exposure and/or r.outcome using get_r_from_lor() for any binary traits</span></div>
<div class="sourceLine" id="cb76-4" data-line-number="4"><span class="co">#&gt; Estimating correlation for quantitative trait.</span></div>
<div class="sourceLine" id="cb76-5" data-line-number="5"><span class="co">#&gt; This method is an approximation, and may be numerically unstable.</span></div>
<div class="sourceLine" id="cb76-6" data-line-number="6"><span class="co">#&gt; Ideally you should estimate r directly from independent replication samples.</span></div>
<div class="sourceLine" id="cb76-7" data-line-number="7"><span class="co">#&gt; Use get_r_from_lor for binary traits.</span></div>
<div class="sourceLine" id="cb76-8" data-line-number="8"><span class="co">#&gt; Estimating correlation for quantitative trait.</span></div>
<div class="sourceLine" id="cb76-9" data-line-number="9"><span class="co">#&gt; This method is an approximation, and may be numerically unstable.</span></div>
<div class="sourceLine" id="cb76-10" data-line-number="10"><span class="co">#&gt; Ideally you should estimate r directly from independent replication samples.</span></div>
<div class="sourceLine" id="cb76-11" data-line-number="11"><span class="co">#&gt; Use get_r_from_lor for binary traits.</span></div>
<div class="sourceLine" id="cb76-12" data-line-number="12">
<span class="kw">kable</span>(out)</div></code></pre>
<table class="table">
<thead><tr class="header">
<th align="left">id.exposure</th>
<th align="left">id.outcome</th>
<th align="left">exposure</th>
<th align="left">outcome</th>
<th align="right">snp_r2.exposure</th>
<th align="right">snp_r2.outcome</th>
<th align="left">correct_causal_direction</th>
<th align="right">steiger_pval</th>
</tr></thead>
<tbody><tr class="odd">
<td align="left">TrGPJg</td>
<td align="left">7</td>
<td align="left">BMI</td>
<td align="left">Coronary heart disease || id:7</td>
<td align="right">0.0111681</td>
<td align="right">0.0006293</td>
<td align="left">TRUE</td>
<td align="right">0</td>
</tr></tbody>
</table>
<p>It calculates the variance explained in the exposure and the outcome by the instrumenting SNPs, and tests if the variance in the outcome is less than the exposure.</p>
<p>This test is, like many others, liable to give inaccurate causal directions under some measurement error parameters in the exposure and the outcome (e.g. if the outcome has much lower measurement precision then its proportion of variance explained will be underestimated). Sensitivity can be applied to evaluate the extent to which the inferred causal direction is liable to measurement error, in two ways.</p>
<ol style="list-style-type: decimal">
<li>Provide estimates of measurement error for the exposure and the outcome, and obtain an adjusted estimate of the causal direction</li>
<li>For all possible values of measurement error, identify the proportion of the parameter space which supports the inferred causal direction</li>
</ol>
<p>These tests are obtained using:</p>
<pre class="sourceCode r" id="cb77"><code class="sourceCode r"><div class="sourceLine" id="cb77-1" data-line-number="1"></div>
<div class="sourceLine" id="cb77-2" data-line-number="2">
<span class="kw"><a href="../reference/mr_steiger.html">mr_steiger</a></span>(</div>
<div class="sourceLine" id="cb77-3" data-line-number="3">    <span class="dt">p_exp =</span> dat<span class="op">$</span>pval.exposure, </div>
<div class="sourceLine" id="cb77-4" data-line-number="4">    <span class="dt">p_out =</span> dat<span class="op">$</span>pval.outcome, </div>
<div class="sourceLine" id="cb77-5" data-line-number="5">    <span class="dt">n_exp =</span> dat<span class="op">$</span>samplesize.exposure, </div>
<div class="sourceLine" id="cb77-6" data-line-number="6">    <span class="dt">n_out =</span> dat<span class="op">$</span>samplesize.outcome, </div>
<div class="sourceLine" id="cb77-7" data-line-number="7">    <span class="dt">r_xxo =</span> <span class="dv">1</span>, </div>
<div class="sourceLine" id="cb77-8" data-line-number="8">    <span class="dt">r_yyo =</span> <span class="dv">1</span>,</div>
<div class="sourceLine" id="cb77-9" data-line-number="9">    <span class="dt">r_exp=</span><span class="dv">0</span>
</div>
<div class="sourceLine" id="cb77-10" data-line-number="10">)</div></code></pre>
<hr>
</div>
<div id="multivariable-mr" class="section level2">
<h2 class="hasAnchor">
<a href="#multivariable-mr" class="anchor"></a>Multivariable MR</h2>
<p>When SNPs instrument multiple potential exposures, for example in the case of different lipid fractions, one method for overcoming this problem is to estimate the influence of each lipid conditioning on the effects of the SNPs on the other lipids. Multivariable MR can be performed using the R package as follows.</p>
<p>The GWAS IDs for HDL, LDL and total cholesterol are 299, 300 and 302. The GWAS ID for coronary heart disease (CHD) is 7. In this example we will estimate the multivariable effects of HDL, LDL and total cholesterol on CHD.</p>
<pre class="sourceCode r" id="cb78"><code class="sourceCode r"><div class="sourceLine" id="cb78-1" data-line-number="1">id_exposure &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dv">299</span>, <span class="dv">300</span>, <span class="dv">302</span>)</div>
<div class="sourceLine" id="cb78-2" data-line-number="2">id_outcome &lt;-<span class="st"> </span><span class="dv">7</span>
</div></code></pre>
<p>First obtain the instruments for each lipid fraction. This entails obtaining a combined set of SNPs including all instruments, and getting those SNPs for each lipid fraction. Therefore, if there are e.g. 20 instruments for each of 3 lipid fractions, but combined there are 30 unique SNPs, then we need to extract each of the 30 SNPs from each lipid fraction (exposure).</p>
<pre class="sourceCode r" id="cb79"><code class="sourceCode r"><div class="sourceLine" id="cb79-1" data-line-number="1">exposure_dat &lt;-<span class="st"> </span><span class="kw"><a href="../reference/mv_extract_exposures.html">mv_extract_exposures</a></span>(id_exposure)</div></code></pre>
<p>Next, also extract those SNPs from the outcome.</p>
<pre class="sourceCode r" id="cb80"><code class="sourceCode r"><div class="sourceLine" id="cb80-1" data-line-number="1">outcome_dat &lt;-<span class="st"> </span><span class="kw"><a href="../reference/extract_outcome_data.html">extract_outcome_data</a></span>(exposure_dat<span class="op">$</span>SNP, id_outcome)</div></code></pre>
<p>Once the data has been obtained, harmonise so that all are on the same reference allele.</p>
<pre class="sourceCode r" id="cb81"><code class="sourceCode r"><div class="sourceLine" id="cb81-1" data-line-number="1">mvdat &lt;-<span class="st"> </span><span class="kw"><a href="../reference/mv_harmonise_data.html">mv_harmonise_data</a></span>(exposure_dat, outcome_dat)</div></code></pre>
<p>Finally, perform the multivariable MR analysis</p>
<pre class="sourceCode r" id="cb82"><code class="sourceCode r"><div class="sourceLine" id="cb82-1" data-line-number="1">res &lt;-<span class="st"> </span><span class="kw"><a href="../reference/mv_multiple.html">mv_multiple</a></span>(mvdat)</div></code></pre>
<p>This generates a table of results.</p>
<div id="note-about-mv-methods" class="section level3">
<h3 class="hasAnchor">
<a href="#note-about-mv-methods" class="anchor"></a>Note about MV methods</h3>
<p>There are several different ways in which this analysis can be formulated. e.g. consider 3 exposures against one outcome, one could:</p>
<ol style="list-style-type: decimal">
<li>Fit all exposures together or fit one exposure at a time against the residuals of the outcome that has been adjusted for the other outcomes. The former is recommended by default in this R package through the <code>mv_multiple</code> function but the latter was how MV MR was originally described by Burgess et al 2015 and can be done with <code>mv_residual</code>.</li>
<li>Fitting all instruments for all exposures (default) or only fitting the instruments for each exposure sequentially</li>
<li>Forcing the slopes through the origin (default) or allowing an intercept term.</li>
</ol>
<p>With these three different parameters there are eight different ways to do MV analysis. We recommend the default settings as described above.</p>
</div>
<div id="note-about-visualisation" class="section level3">
<h3 class="hasAnchor">
<a href="#note-about-visualisation" class="anchor"></a>Note about visualisation</h3>
<p>Plots can be generated using the <code>plots=TRUE</code> argument for <code>mv_multiple</code> and <code>mv_residual</code>.</p>
<p>The current plots being generated are not necessarily adequate because while they show the slope through the raw points, they do not demonstrate that the raw points might be effectively different between plots because they are conditional on the other exposures.</p>
<hr>
</div>
</div>
<div id="mr-estimates-when-instruments-are-correlated" class="section level2">
<h2 class="hasAnchor">
<a href="#mr-estimates-when-instruments-are-correlated" class="anchor"></a>MR estimates when instruments are correlated</h2>
<p>In the examples shown so far it is assumed that instruments are independent (i.e. they are not in linkage disequilibrium, LD). This is to avoid ‘double counting’ effects. An alternative approach is to estimate the MR effects accounting for the correlation between variants.</p>
<p>The TwoSampleMR package has not implemented this yet, but the <a href="https://cran.r-project.org/web/packages/MendelianRandomization/index.html">MendelianRandomization</a> R package by Olena Yavorska and Stephen Burgess does have this functionality. We can use the TwoSampleMR package to extract, format and harmonise data, and then convert to the format required by the MendelianRandomization package. The MR-Base server has the individual level genetic data for ~500 Europeans in 1000 genomes data, and can obtain the LD matrix for a set of SNPs using these data. For example:</p>
<pre class="sourceCode r" id="cb83"><code class="sourceCode r"><div class="sourceLine" id="cb83-1" data-line-number="1">snplist &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"rs234"</span>, <span class="st">"rs1205"</span>)</div>
<div class="sourceLine" id="cb83-2" data-line-number="2">
<span class="kw"><a href="../reference/ld_matrix.html">ld_matrix</a></span>(snplist)</div></code></pre>
<p>Here <code>ld_matrix</code> returns the LD correlation values (not R^2) for each pair of variants present in the 1000 genomes data set.</p>
<pre class="sourceCode r" id="cb84"><code class="sourceCode r"><div class="sourceLine" id="cb84-1" data-line-number="1">dat &lt;-<span class="st"> </span><span class="kw"><a href="../reference/harmonise_data.html">harmonise_data</a></span>(</div>
<div class="sourceLine" id="cb84-2" data-line-number="2">    <span class="dt">exposure_dat =</span> bmi_exp_dat, </div>
<div class="sourceLine" id="cb84-3" data-line-number="3">    <span class="dt">outcome_dat =</span> chd_out_dat</div>
<div class="sourceLine" id="cb84-4" data-line-number="4">)</div></code></pre>
<p>Convert to the <code>MRInput</code> format for the MendelianRandomization package:</p>
<pre class="sourceCode r" id="cb85"><code class="sourceCode r"><div class="sourceLine" id="cb85-1" data-line-number="1">dat2 &lt;-<span class="st"> </span><span class="kw"><a href="../reference/dat_to_MRInput.html">dat_to_MRInput</a></span>(dat)</div></code></pre>
<p>This produces a list of <code>MRInput</code> objects that can be used with the MendelianRandomization functions, e.g.</p>
<pre class="sourceCode r" id="cb86"><code class="sourceCode r"><div class="sourceLine" id="cb86-1" data-line-number="1">MendelianRandomization<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/MendelianRandomization/man/mr_ivw.html">mr_ivw</a></span>(dat2[[<span class="dv">1</span>]])</div></code></pre>
<p>Alternatively, convert to the <code>MRInput</code> format but also obtaining the LD matrix for the instruments</p>
<pre class="sourceCode r" id="cb87"><code class="sourceCode r"><div class="sourceLine" id="cb87-1" data-line-number="1">dat2 &lt;-<span class="st"> </span><span class="kw"><a href="../reference/dat_to_MRInput.html">dat_to_MRInput</a></span>(dat, <span class="dt">get_correlation=</span><span class="ot">TRUE</span>)</div>
<div class="sourceLine" id="cb87-2" data-line-number="2">MendelianRandomization<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/MendelianRandomization/man/mr_ivw.html">mr_ivw</a></span>(dat2[[<span class="dv">1</span>]], <span class="dt">correl=</span><span class="ot">TRUE</span>)</div></code></pre>
<hr>
</div>
<div id="mr-moe-using-a-mixture-of-experts-machine-learning-approach" class="section level2">
<h2 class="hasAnchor">
<a href="#mr-moe-using-a-mixture-of-experts-machine-learning-approach" class="anchor"></a>MR-MoE: Using a mixture of experts machine learning approach</h2>
<p>We recently developed MR-MoE, a method to choose the most appropriate amongst several MR tests using a machine learning algorithm. Note that the method is still under review, but full details are described here: <a href="http://www.biorxiv.org/content/early/2017/08/23/173682">biorxiv.org/content/early/2017/08/23/173682</a>.</p>
<p>MR-MoE operates by taking a set of harmonised data, inferring some characteristics about the dataset, and using those characteristics to predict how well each of the different MR methods will perform on the dataset, in terms of maximising power while minimising false discovery rates.</p>
<p>In order to run the analysis you must download an RData object that contains the trained random forests that are used to predict the efficacy of each method. This can be downloaded from here:</p>
<p><a href="https://www.dropbox.com/s/5la7y38od95swcf/rf.rdata?dl=0">dropbox.com/s/5la7y38od95swcf/rf.rdata?dl=0</a></p>
<p><strong>Caution: this is a large file (approx 3Gb)</strong></p>
<p>Once downloaded, read in the object and use the <code>mr_moe</code> function to perform the analysis. An example is shown here, estimating the causal effect of BMI on coronary heart disease:</p>
<pre class="sourceCode r" id="cb88"><code class="sourceCode r"><div class="sourceLine" id="cb88-1" data-line-number="1"><span class="co"># Extact instruments for BMI</span></div>
<div class="sourceLine" id="cb88-2" data-line-number="2">expsure_dat &lt;-<span class="st"> </span><span class="kw"><a href="../reference/extract_instruments.html">extract_instruments</a></span>(<span class="dv">2</span>)</div>
<div class="sourceLine" id="cb88-3" data-line-number="3"></div>
<div class="sourceLine" id="cb88-4" data-line-number="4"><span class="co"># Get corresponding effects for CHD</span></div>
<div class="sourceLine" id="cb88-5" data-line-number="5">outcome_dat &lt;-<span class="st"> </span><span class="kw"><a href="../reference/extract_outcome_data.html">extract_outcome_data</a></span>(exposure_dat<span class="op">$</span>SNP, <span class="dv">7</span>)</div>
<div class="sourceLine" id="cb88-6" data-line-number="6"></div>
<div class="sourceLine" id="cb88-7" data-line-number="7"><span class="co"># Harmonise</span></div>
<div class="sourceLine" id="cb88-8" data-line-number="8">dat &lt;-<span class="st"> </span><span class="kw"><a href="../reference/harmonise_data.html">harmonise_data</a></span>(exposure_dat, outcome_dat)</div>
<div class="sourceLine" id="cb88-9" data-line-number="9"></div>
<div class="sourceLine" id="cb88-10" data-line-number="10"><span class="co"># Load the downloaded RData object. This loads the rf object</span></div>
<div class="sourceLine" id="cb88-11" data-line-number="11">
<span class="kw"><a href="https://rdrr.io/r/base/load.html">load</a></span>(<span class="st">"rf.rdata"</span>)</div>
<div class="sourceLine" id="cb88-12" data-line-number="12"></div>
<div class="sourceLine" id="cb88-13" data-line-number="13"><span class="co"># Obtain estimates from all methods, and generate data metrics</span></div>
<div class="sourceLine" id="cb88-14" data-line-number="14">res &lt;-<span class="st"> </span><span class="kw"><a href="../reference/mr_wrapper.html">mr_wrapper</a></span>(dat)</div>
<div class="sourceLine" id="cb88-15" data-line-number="15"></div>
<div class="sourceLine" id="cb88-16" data-line-number="16"><span class="co"># MR-MoE - predict the performance of each method</span></div>
<div class="sourceLine" id="cb88-17" data-line-number="17">res_moe &lt;-<span class="st"> </span><span class="kw"><a href="../reference/mr_moe.html">mr_moe</a></span>(res, rf)</div></code></pre>
<p>The function does the following:</p>
<ol style="list-style-type: decimal">
<li>Performs MR using each of 11 MR methods</li>
<li>Applies Steiger filtering or heterogeneity filtering or both to remove SNPs that do not have substantially larger R^2 with the exposure than the outcome. Note - for binary traits ensure number of cases, number of controls, and allele frequencies are available for each SNP. For continuous traits make sure the p-value and sample size is available. The function infers if a trait is binary or continuous based on the units.exposure and units.outcome columns - binary traits must have those values set to ‘log odds’</li>
<li>Performs the 14 MR methods again but using the subset of SNPs that survive Steiger filtering</li>
<li>Generates meta data about the summary data to predict the most reliable of the 28 methods applied.</li>
</ol>
<p>For every exposure / outcome combination in the <code>dat</code> object, the MR-MoE method is applied. The function returns a list which is as long as the number of exposure / outcome combinations. In this case, it will be of length 1, containing the result for BMI on CHD.</p>
<p>The result object itself is a list with the following elements:</p>
<ul>
<li>
<code>estimates</code> (results from each MR)</li>
<li>
<code>heterogeneity</code> (results from heterogeneity for different filtering approaches)</li>
<li>
<code>directional_pleiotropy</code> (egger intercepts)</li>
<li>
<code>info</code> (metrics used to generate MOE)</li>
</ul>
<p>Looking at the <code>estimates</code>, we see that there is a column called <code>MOE</code> which is the predicted AUROC curve performance of each method.</p>
<hr>
</div>
</div>
<div id="post-mr-results-management" class="section level1">
<h1 class="hasAnchor">
<a href="#post-mr-results-management" class="anchor"></a>Post MR results management</h1>
<p>The TwoSampleMR package also provides the following functions for managing or editing MR results.</p>
<div id="split-outcome-names" class="section level2">
<h2 class="hasAnchor">
<a href="#split-outcome-names" class="anchor"></a>Split outcome names</h2>
<p>The outcome column in the output of mr() combines the original outcome name with the outcome trait ID.</p>
<pre class="sourceCode r" id="cb89"><code class="sourceCode r"><div class="sourceLine" id="cb89-1" data-line-number="1">
<span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(res)</div>
<div class="sourceLine" id="cb89-2" data-line-number="2"><span class="co">#&gt;   id.exposure id.outcome                        outcome exposure</span></div>
<div class="sourceLine" id="cb89-3" data-line-number="3"><span class="co">#&gt; 1      TrGPJg          7 Coronary heart disease || id:7      BMI</span></div>
<div class="sourceLine" id="cb89-4" data-line-number="4"><span class="co">#&gt; 2      TrGPJg          7 Coronary heart disease || id:7      BMI</span></div>
<div class="sourceLine" id="cb89-5" data-line-number="5"><span class="co">#&gt;                      method nsnp         b         se         pval</span></div>
<div class="sourceLine" id="cb89-6" data-line-number="6"><span class="co">#&gt; 1                  MR Egger   27 0.1138566 0.03292752 1.962323e-03</span></div>
<div class="sourceLine" id="cb89-7" data-line-number="7"><span class="co">#&gt; 2 Inverse variance weighted   27 0.1139168 0.01611170 1.544411e-12</span></div></code></pre>
<p>The outcome column can be split into separate columns for the id and outcome name using the split_outcome function:</p>
<pre class="sourceCode r" id="cb90"><code class="sourceCode r"><div class="sourceLine" id="cb90-1" data-line-number="1">res&lt;-<span class="kw"><a href="../reference/mr.html">mr</a></span>(dat)</div>
<div class="sourceLine" id="cb90-2" data-line-number="2"><span class="co">#&gt; Analysing 'TrGPJg' on '7'</span></div>
<div class="sourceLine" id="cb90-3" data-line-number="3">
<span class="kw"><a href="../reference/split_outcome.html">split_outcome</a></span>(res)</div>
<div class="sourceLine" id="cb90-4" data-line-number="4"><span class="co">#&gt;   id.exposure id.outcome                outcome exposure</span></div>
<div class="sourceLine" id="cb90-5" data-line-number="5"><span class="co">#&gt; 1      TrGPJg          7 Coronary heart disease      BMI</span></div>
<div class="sourceLine" id="cb90-6" data-line-number="6"><span class="co">#&gt; 2      TrGPJg          7 Coronary heart disease      BMI</span></div>
<div class="sourceLine" id="cb90-7" data-line-number="7"><span class="co">#&gt; 3      TrGPJg          7 Coronary heart disease      BMI</span></div>
<div class="sourceLine" id="cb90-8" data-line-number="8"><span class="co">#&gt; 4      TrGPJg          7 Coronary heart disease      BMI</span></div>
<div class="sourceLine" id="cb90-9" data-line-number="9"><span class="co">#&gt; 5      TrGPJg          7 Coronary heart disease      BMI</span></div>
<div class="sourceLine" id="cb90-10" data-line-number="10"><span class="co">#&gt;                      method nsnp          b         se         pval</span></div>
<div class="sourceLine" id="cb90-11" data-line-number="11"><span class="co">#&gt; 1                  MR Egger   27 0.11385662 0.03292752 1.962323e-03</span></div>
<div class="sourceLine" id="cb90-12" data-line-number="12"><span class="co">#&gt; 2           Weighted median   27 0.07940417 0.01951586 4.727617e-05</span></div>
<div class="sourceLine" id="cb90-13" data-line-number="13"><span class="co">#&gt; 3 Inverse variance weighted   27 0.11391684 0.01611170 1.544411e-12</span></div>
<div class="sourceLine" id="cb90-14" data-line-number="14"><span class="co">#&gt; 4               Simple mode   27 0.07510331 0.03992376 7.119272e-02</span></div>
<div class="sourceLine" id="cb90-15" data-line-number="15"><span class="co">#&gt; 5             Weighted mode   27 0.08875377 0.02176798 3.820924e-04</span></div></code></pre>
</div>
<div id="split-exposure-names" class="section level2">
<h2 class="hasAnchor">
<a href="#split-exposure-names" class="anchor"></a>Split exposure names</h2>
<p>Similarly to the outcome column, the exposure column in the output of mr() combines the original exposure name with the exposure trait ID. This can be split into separate columns for the id and exposure name using the split_exposure function.</p>
</div>
<div id="generate-odds-ratios-with-95-confidence-intervals" class="section level2">
<h2 class="hasAnchor">
<a href="#generate-odds-ratios-with-95-confidence-intervals" class="anchor"></a>Generate odds ratios with 95% confidence intervals</h2>
<p>Users can convert log odds ratios into odds ratios with 95% confidence intervals using:</p>
<pre class="sourceCode r" id="cb91"><code class="sourceCode r"><div class="sourceLine" id="cb91-1" data-line-number="1">
<span class="kw"><a href="../reference/generate_odds_ratios.html">generate_odds_ratios</a></span>(res)</div>
<div class="sourceLine" id="cb91-2" data-line-number="2"><span class="co">#&gt;   id.exposure id.outcome                        outcome exposure</span></div>
<div class="sourceLine" id="cb91-3" data-line-number="3"><span class="co">#&gt; 1      TrGPJg          7 Coronary heart disease || id:7      BMI</span></div>
<div class="sourceLine" id="cb91-4" data-line-number="4"><span class="co">#&gt; 2      TrGPJg          7 Coronary heart disease || id:7      BMI</span></div>
<div class="sourceLine" id="cb91-5" data-line-number="5"><span class="co">#&gt; 3      TrGPJg          7 Coronary heart disease || id:7      BMI</span></div>
<div class="sourceLine" id="cb91-6" data-line-number="6"><span class="co">#&gt; 4      TrGPJg          7 Coronary heart disease || id:7      BMI</span></div>
<div class="sourceLine" id="cb91-7" data-line-number="7"><span class="co">#&gt; 5      TrGPJg          7 Coronary heart disease || id:7      BMI</span></div>
<div class="sourceLine" id="cb91-8" data-line-number="8"><span class="co">#&gt;                      method nsnp          b         se         pval</span></div>
<div class="sourceLine" id="cb91-9" data-line-number="9"><span class="co">#&gt; 1                  MR Egger   27 0.11385662 0.03292752 1.962323e-03</span></div>
<div class="sourceLine" id="cb91-10" data-line-number="10"><span class="co">#&gt; 2           Weighted median   27 0.07940417 0.01951586 4.727617e-05</span></div>
<div class="sourceLine" id="cb91-11" data-line-number="11"><span class="co">#&gt; 3 Inverse variance weighted   27 0.11391684 0.01611170 1.544411e-12</span></div>
<div class="sourceLine" id="cb91-12" data-line-number="12"><span class="co">#&gt; 4               Simple mode   27 0.07510331 0.03992376 7.119272e-02</span></div>
<div class="sourceLine" id="cb91-13" data-line-number="13"><span class="co">#&gt; 5             Weighted mode   27 0.08875377 0.02176798 3.820924e-04</span></div>
<div class="sourceLine" id="cb91-14" data-line-number="14"><span class="co">#&gt;          lo_ci     up_ci       or  or_lci95 or_uci95</span></div>
<div class="sourceLine" id="cb91-15" data-line-number="15"><span class="co">#&gt; 1  0.049318685 0.1783946 1.120591 1.0505551 1.195297</span></div>
<div class="sourceLine" id="cb91-16" data-line-number="16"><span class="co">#&gt; 2  0.041153091 0.1176553 1.082642 1.0420116 1.124856</span></div>
<div class="sourceLine" id="cb91-17" data-line-number="17"><span class="co">#&gt; 3  0.082337906 0.1454958 1.120659 1.0858227 1.156613</span></div>
<div class="sourceLine" id="cb91-18" data-line-number="18"><span class="co">#&gt; 4 -0.003147263 0.1533539 1.077996 0.9968577 1.165737</span></div>
<div class="sourceLine" id="cb91-19" data-line-number="19"><span class="co">#&gt; 5  0.046088536 0.1314190 1.092812 1.0471671 1.140446</span></div></code></pre>
</div>
<div id="subset-on-method" class="section level2">
<h2 class="hasAnchor">
<a href="#subset-on-method" class="anchor"></a>Subset on method</h2>
<p>It is sometimes useful to subset results on MR method, so that there is one unique result for each exposure-outcome combination:</p>
<pre class="sourceCode r" id="cb92"><code class="sourceCode r"><div class="sourceLine" id="cb92-1" data-line-number="1">
<span class="kw"><a href="../reference/subset_on_method.html">subset_on_method</a></span>(res)</div>
<div class="sourceLine" id="cb92-2" data-line-number="2"><span class="co">#&gt;   id.exposure id.outcome                        outcome exposure</span></div>
<div class="sourceLine" id="cb92-3" data-line-number="3"><span class="co">#&gt; 3      TrGPJg          7 Coronary heart disease || id:7      BMI</span></div>
<div class="sourceLine" id="cb92-4" data-line-number="4"><span class="co">#&gt;                      method nsnp         b        se         pval</span></div>
<div class="sourceLine" id="cb92-5" data-line-number="5"><span class="co">#&gt; 3 Inverse variance weighted   27 0.1139168 0.0161117 1.544411e-12</span></div></code></pre>
<p>The default is to subset on the IVW method when &gt;1 SNP is available and to use the Wald ratio method when a single SNP is available. Users can specify which multi-SNP method to subset on.</p>
</div>
<div id="combine-all-results" class="section level2">
<h2 class="hasAnchor">
<a href="#combine-all-results" class="anchor"></a>Combine all results</h2>
<p>It is often useful to combine all results and study level characterists into a single dataframe or table, e.g. for sharing results with collaborators or when the user wishes to present all results in a single table or figure. This can be done using the combine_all_mrresults() function:</p>
<pre class="sourceCode r" id="cb93"><code class="sourceCode r"><div class="sourceLine" id="cb93-1" data-line-number="1">res&lt;-<span class="kw"><a href="../reference/mr.html">mr</a></span>(dat)</div>
<div class="sourceLine" id="cb93-2" data-line-number="2"><span class="co">#&gt; Analysing 'TrGPJg' on '7'</span></div>
<div class="sourceLine" id="cb93-3" data-line-number="3">het&lt;-<span class="kw"><a href="../reference/mr_heterogeneity.html">mr_heterogeneity</a></span>(dat)</div>
<div class="sourceLine" id="cb93-4" data-line-number="4">plt&lt;-<span class="kw"><a href="../reference/mr_pleiotropy_test.html">mr_pleiotropy_test</a></span>(dat)</div>
<div class="sourceLine" id="cb93-5" data-line-number="5">sin&lt;-<span class="kw"><a href="../reference/mr_singlesnp.html">mr_singlesnp</a></span>(dat)</div>
<div class="sourceLine" id="cb93-6" data-line-number="6">all_res&lt;-<span class="kw"><a href="../reference/combine_all_mrresults.html">combine_all_mrresults</a></span>(res,het,plt,sin,<span class="dt">ao_slc=</span>T,<span class="dt">Exp=</span>T,<span class="dt">split.exposure=</span>F,<span class="dt">split.outcome=</span>T)</div>
<div class="sourceLine" id="cb93-7" data-line-number="7"><span class="co">#&gt; Warning: 'available_outcomes' is deprecated.</span></div>
<div class="sourceLine" id="cb93-8" data-line-number="8"><span class="co">#&gt; Use 'ieugwasr::gwasinfo()' instead.</span></div>
<div class="sourceLine" id="cb93-9" data-line-number="9"><span class="co">#&gt; See help("Deprecated")</span></div>
<div class="sourceLine" id="cb93-10" data-line-number="10">
<span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(all_res[,<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"Method"</span>,<span class="st">"outcome"</span>,<span class="st">"exposure"</span>,<span class="st">"nsnp"</span>,<span class="st">"b"</span>,<span class="st">"se"</span>,<span class="st">"pval"</span>,<span class="st">"intercept"</span>,<span class="st">"intercept_se"</span>,<span class="st">"intercept_pval"</span>,<span class="st">"Q"</span>,<span class="st">"Q_df"</span>,<span class="st">"Q_pval"</span>,<span class="st">"consortium"</span>,<span class="st">"ncase"</span>,<span class="st">"ncontrol"</span>,<span class="st">"pmid"</span>,<span class="st">"population"</span>)])</div>
<div class="sourceLine" id="cb93-11" data-line-number="11"><span class="co">#&gt;                      Method                outcome exposure nsnp</span></div>
<div class="sourceLine" id="cb93-12" data-line-number="12"><span class="co">#&gt; 1 Inverse variance weighted Coronary heart disease      BMI   27</span></div>
<div class="sourceLine" id="cb93-13" data-line-number="13"><span class="co">#&gt; 2                  MR Egger Coronary heart disease      BMI   27</span></div>
<div class="sourceLine" id="cb93-14" data-line-number="14"><span class="co">#&gt; 3               Simple mode Coronary heart disease      BMI   27</span></div>
<div class="sourceLine" id="cb93-15" data-line-number="15"><span class="co">#&gt; 4                Wald ratio Coronary heart disease      BMI    1</span></div>
<div class="sourceLine" id="cb93-16" data-line-number="16"><span class="co">#&gt; 5                Wald ratio Coronary heart disease      BMI    1</span></div>
<div class="sourceLine" id="cb93-17" data-line-number="17"><span class="co">#&gt; 6                Wald ratio Coronary heart disease      BMI    1</span></div>
<div class="sourceLine" id="cb93-18" data-line-number="18"><span class="co">#&gt;            b         se         pval    intercept intercept_se</span></div>
<div class="sourceLine" id="cb93-19" data-line-number="19"><span class="co">#&gt; 1 0.11391684 0.01611170 1.544411e-12           NA           NA</span></div>
<div class="sourceLine" id="cb93-20" data-line-number="20"><span class="co">#&gt; 2 0.11385662 0.03292752 1.962323e-03 1.110681e-05  0.005263163</span></div>
<div class="sourceLine" id="cb93-21" data-line-number="21"><span class="co">#&gt; 3 0.07510331 0.03723528 5.413216e-02           NA           NA</span></div>
<div class="sourceLine" id="cb93-22" data-line-number="22"><span class="co">#&gt; 4 0.16062308 0.09240846 8.217808e-02           NA           NA</span></div>
<div class="sourceLine" id="cb93-23" data-line-number="23"><span class="co">#&gt; 5 0.37717647 0.12765706 3.130673e-03           NA           NA</span></div>
<div class="sourceLine" id="cb93-24" data-line-number="24"><span class="co">#&gt; 6 0.14005789 0.05720105 1.434447e-02           NA           NA</span></div>
<div class="sourceLine" id="cb93-25" data-line-number="25"><span class="co">#&gt;   intercept_pval        Q Q_df     Q_pval        consortium ncase ncontrol</span></div>
<div class="sourceLine" id="cb93-26" data-line-number="26"><span class="co">#&gt; 1             NA 39.68703   26 0.04185786 CARDIoGRAMplusC4D 60801   123504</span></div>
<div class="sourceLine" id="cb93-27" data-line-number="27"><span class="co">#&gt; 2       0.998333 39.68702   25 0.03139634 CARDIoGRAMplusC4D 60801   123504</span></div>
<div class="sourceLine" id="cb93-28" data-line-number="28"><span class="co">#&gt; 3             NA       NA   NA         NA CARDIoGRAMplusC4D 60801   123504</span></div>
<div class="sourceLine" id="cb93-29" data-line-number="29"><span class="co">#&gt; 4             NA       NA   NA         NA CARDIoGRAMplusC4D 60801   123504</span></div>
<div class="sourceLine" id="cb93-30" data-line-number="30"><span class="co">#&gt; 5             NA       NA   NA         NA CARDIoGRAMplusC4D 60801   123504</span></div>
<div class="sourceLine" id="cb93-31" data-line-number="31"><span class="co">#&gt; 6             NA       NA   NA         NA CARDIoGRAMplusC4D 60801   123504</span></div>
<div class="sourceLine" id="cb93-32" data-line-number="32"><span class="co">#&gt;       pmid population</span></div>
<div class="sourceLine" id="cb93-33" data-line-number="33"><span class="co">#&gt; 1 26343387      Mixed</span></div>
<div class="sourceLine" id="cb93-34" data-line-number="34"><span class="co">#&gt; 2 26343387      Mixed</span></div>
<div class="sourceLine" id="cb93-35" data-line-number="35"><span class="co">#&gt; 3 26343387      Mixed</span></div>
<div class="sourceLine" id="cb93-36" data-line-number="36"><span class="co">#&gt; 4 26343387      Mixed</span></div>
<div class="sourceLine" id="cb93-37" data-line-number="37"><span class="co">#&gt; 5 26343387      Mixed</span></div>
<div class="sourceLine" id="cb93-38" data-line-number="38"><span class="co">#&gt; 6 26343387      Mixed</span></div></code></pre>
<p>This combines all results from mr(), mr_heterogeneity(), mr_pleiotropy_test() and mr_singlesnp() into a single dataframe. It also merges the results with outcome study level characteristics from the available_outcomes() function, including sample size characteristics. If requested, it also exponentiates results (e.g. if the user wants log odds ratio converted into odds ratios with 95 percent confidence intervals).</p>
</div>
</div>
<div id="references" class="section level1">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<p><br></p>
<div id="refs" class="references">
<div id="ref-Bowden2015">
<p>Bowden, Jack, George Davey Smith, and Stephen Burgess. 2015. “Mendelian randomization with invalid instruments: effect estimation and bias detection through Egger regression.” <em>International Journal of Epidemiology</em> In press.</p>
</div>
<div id="ref-DaveySmith2003">
<p>Davey Smith, G., and S. Ebrahim. 2003. “’Mendelian randomization’: can genetic epidemiology contribute to understanding environmental determinants of disease?” <em>International Journal of Epidemiology</em> 32 (1):1–22. <a href="https://doi.org/10.1093/ije/dyg070" class="uri">https://doi.org/10.1093/ije/dyg070</a>.</p>
</div>
<div id="ref-DaveySmithHemani2014">
<p>Davey Smith, George, and Gibran Hemani. 2014. “Mendelian randomization: genetic anchors for causal inference in epidemiological studies.” <em>Human Molecular Genetics</em> 23 (R1). Oxford Univ Press:R89—–R98. <a href="https://doi.org/10.1093/hmg/ddu328" class="uri">https://doi.org/10.1093/hmg/ddu328</a>.</p>
</div>
<div id="ref-Pierce2013">
<p>Pierce, Brandon L, and Stephen Burgess. 2013. “Efficient design for Mendelian randomization studies: subsample and 2-sample instrumental variable estimators.” <em>American Journal of Epidemiology</em> 178 (7):1177–84. <a href="https://doi.org/10.1093/aje/kwt084" class="uri">https://doi.org/10.1093/aje/kwt084</a>.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#introduction">Introduction</a><ul class="nav nav-pills nav-stacked">
<li><a href="#installation">Installation</a></li>
      <li><a href="#overview">Overview</a></li>
      <li><a href="#authentication">Authentication</a></li>
      </ul>
</li>
      <li>
<a href="#exposure-data">Exposure data</a><ul class="nav nav-pills nav-stacked">
<li><a href="#reading-in-from-a-file">Reading in from a file</a></li>
      <li><a href="#using-an-existing-data-frame">Using an existing data frame</a></li>
      <li><a href="#obtaining-instruments-from-existing-catalogues">Obtaining instruments from existing catalogues</a></li>
      <li><a href="#clumping">Clumping</a></li>
      </ul>
</li>
      <li>
<a href="#outcome-data">Outcome data</a><ul class="nav nav-pills nav-stacked">
<li><a href="#available-studies-in-mr-base">Available studies in MR Base</a></li>
      <li><a href="#extracting-particular-snps-from-particular-studies">Extracting particular SNPs from particular studies</a></li>
      <li><a href="#ld-proxies">LD proxies</a></li>
      <li><a href="#using-local-gwas-summary-data">Using local GWAS summary data</a></li>
      <li><a href="#outcome-data-format">Outcome data format</a></li>
      </ul>
</li>
      <li>
<a href="#harmonise-data">Harmonise data</a><ul class="nav nav-pills nav-stacked">
<li><a href="#dealing-with-strand-issues">Dealing with strand issues</a></li>
      <li><a href="#drop-duplicate-exposure-outcome-summary-sets">Drop duplicate exposure-outcome summary sets</a></li>
      </ul>
</li>
      <li>
<a href="#perform-mr">Perform MR</a><ul class="nav nav-pills nav-stacked">
<li><a href="#mr-methods">MR methods</a></li>
      <li><a href="#sensitivity-analyses">Sensitivity analyses</a></li>
      <li><a href="#plots">Plots</a></li>
      <li><a href="#to-many-forest-plot">1-to-many forest plot</a></li>
      <li><a href="#mr-raps-many-weak-instruments-analysis">MR.RAPS: Many weak instruments analysis</a></li>
      <li><a href="#reports">Reports</a></li>
      <li><a href="#mr-steiger-directionality-test">MR Steiger directionality test</a></li>
      <li><a href="#multivariable-mr">Multivariable MR</a></li>
      <li><a href="#mr-estimates-when-instruments-are-correlated">MR estimates when instruments are correlated</a></li>
      <li><a href="#mr-moe-using-a-mixture-of-experts-machine-learning-approach">MR-MoE: Using a mixture of experts machine learning approach</a></li>
      </ul>
</li>
      <li>
<a href="#post-mr-results-management">Post MR results management</a><ul class="nav nav-pills nav-stacked">
<li><a href="#split-outcome-names">Split outcome names</a></li>
      <li><a href="#split-exposure-names">Split exposure names</a></li>
      <li><a href="#generate-odds-ratios-with-95-confidence-intervals">Generate odds ratios with 95% confidence intervals</a></li>
      <li><a href="#subset-on-method">Subset on method</a></li>
      <li><a href="#combine-all-results">Combine all results</a></li>
      </ul>
</li>
      <li><a href="#references">References</a></li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by Gibran Hemani, Philip Haycock, Jie Zheng, Tom Gaunt, Ben Elsworth.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
